{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import openai\n",
    "# import deeplake \n",
    "from dotenv import load_dotenv\n",
    "\n",
    "from langchain.document_loaders import TextLoader\n",
    "from langchain.embeddings.openai import OpenAIEmbeddings\n",
    "from langchain.vectorstores import DeepLake\n",
    "from langchain.text_splitter import CharacterTextSplitter\n",
    "from langchain.document_loaders import PyPDFLoader\n",
    "\n",
    "from langchain.vectorstores import Chroma\n",
    "from langchain.chains import RetrievalQA\n",
    "from langchain.chat_models import ChatOpenAI\n",
    "from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "load_dotenv(os.getcwd()+'/keys.env')\n",
    "activeloop_token = os.getenv('ACTIVELOOP_TOKEN')\n",
    "deeplake_username = os.getenv('DEEPLAKE_USERNAME')\n",
    "openai.api_key = os.environ.get('OPENAI_API_KEY')\n",
    "\n",
    "pdfdirname = os.getcwd()+'/../LangChainQueryTexts/ThesisPapers'\n",
    "pdfdirname = os.path.abspath(pdfdirname)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_docs(root_dir):\n",
    "    docs = []\n",
    "    for dirpath, dirnames, filenames in os.walk(root_dir):\n",
    "        for file in filenames:\n",
    "            print(file)\n",
    "            try:\n",
    "                loader = PyPDFLoader(os.path.join(\n",
    "                    dirpath, file))\n",
    "                docs.extend(loader.load_and_split())\n",
    "            except Exception as e:\n",
    "                print(e)\n",
    "                pass\n",
    "    return docs\n",
    "\n",
    "\n",
    "def split_docs(docs):\n",
    "    text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=0)\n",
    "    return text_splitter.split_documents(docs)\n",
    "\n",
    "\n",
    "def main(root_dir, deep_lake_path):\n",
    "    docs = load_docs(root_dir)\n",
    "    texts = split_docs(docs)\n",
    "    embeddings = OpenAIEmbeddings()\n",
    "    \n",
    "    db = DeepLake(dataset_path=deep_lake_path, embedding_function=embeddings)\n",
    "    \n",
    "    db.add_documents(texts)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load, split, and embed text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GeneticAlgorithm.pdf\n",
      "Sensitivity_analysis_of_autonomous_oscillations_ap.pdf\n",
      "SelfAssemblyOscillations.pdf\n",
      "Osman.pdf\n",
      "TunableOscillator.pdf\n",
      "DesignPrinciplesOscillators.pdf\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 \\nhttps://doi.org/10.1186/s13036-019-0205-0\\nMETHODOLOGY OpenAccess\\nComputationalanalysisofviable\\nparameterregionsinmodelsofsynthetic\\nbiologicalsystems\\nŽigaPušnik,MihaMraz,NikolajZimicandMihaMoškon*\\nAbstract\\nBackground: Gene regulatorynetworkswithdifferenttopologicaland/ordynamicalpropertiesmightexhibitsimilar\\nbehavior.Systemthatislessperceptivefortheperturbationsofitsinternalandexternalfactorsshouldbepreferred.\\nMethodsforsensitivityandrobustnessassessmenthavealreadybeendevelopedandcanberoughlydividedinto\\nlocalandglobalapproaches.Localmethodsfocusonlyonthelocalareaaroundnominalparametervalues.Thiscan\\nbeproblematicwhenparametersexhibitsthedesiredbehavioroveralargerangeofparameterperturbationsor\\nwhenparametervaluesareunknown.Globalmethods,ontheotherhand,investigatethewholespaceofparameter\\nvaluesandmostlyrelyondifferentsamplingtechniques.Thiscanbecomputationallyinefficient.Toaddressthese\\nshortcomings’glocal’approachesweredevelopedthatapplyglobalandlocalapproachesinaneffectiveandrigorous\\nmanner.\\nResults: Herein,wepresentacomputationalapproachfor’glocal’analysisofviableparameterregionsinbiological\\nmodels.Themethodologyisbasedontheexplorationofhigh-dimensionalviableparameterspaceswithglobaland\\nlocalsampling,clusteringanddimensionalityreductiontechniques.Theproposedmethodologyallowsusto\\nefficientlyinvestigatetheviableparameterspaceregions,evaluatetheregionswhichexhibitthelargestrobustness,\\nandtogathernewinsightsregardingthesizeandconnectivityoftheviableparameterregions.Weevaluatethe\\nproposedmethodologyonthreedifferentsyntheticgeneregulatorynetworkmodels,i.e.therepressilatormodel,the\\nmodeloftheAC-DCcircuitandthemodeloftheedge-triggeredmaster-slaveDflip-flop.\\nConclusions: Theproposedmethodologyprovidesarigorousassessmentoftheshapeandsizeofviableparameter\\nregionsbasedon(1)themathematicaldescriptionofthebiologicalsystemofinterest,(2)constraintsthatdefine\\nfeasibleparameterregionsand(3)costfunctionthatdefinesthedesiredorobservedbehaviorofthesystem.These\\ninsightscanbeusedtoassesstherobustnessofbiologicalsystems,eveninthecasewhenparametervaluesare\\nunknownandmoreimportantly,evenwhentherearemultiplepoorlyconnectedviableparameterregionsinthe\\nsolutionspace.Moreover,themethodologycanbeefficientlyappliedtotheanalysisofbiologicalsystemsthatexhibit\\nmultiplemodesofthetargetedbehavior.\\nKeywords: Biologicalmodel,Repressilator,AC-DCcircuit,BiologicalDflip-flop,Computationalanalysis,Genetic\\nalgorithms,Principalcomponents,Viableparameterregions,Robustness\\n*Correspondence: miha.moskon@fri.uni-lj.si\\nUniversityofLjubljana,FacultyofComputerandInformationScience,Ve ˇcna\\npot113,1000Ljubljana,Slovenia\\n©TheAuthor(s).2019 OpenAccess ThisarticleisdistributedunderthetermsoftheCreativeCommonsAttribution4.0\\nInternationalLicense( http://creativecommons.org/licenses/by/4.0/ ),whichpermitsunrestricteduse,distribution,and\\nreproductioninanymedium,providedyougiveappropriatecredittotheoriginalauthor(s)andthesource,providealinktothe\\nCreativeCommonslicense,andindicateifchangesweremade. TheCreativeCommonsPublicDomainDedicationwaiver\\n(http://creativecommons.org/publicdomain/zero/1.0/ )appliestothedatamadeavailableinthisarticle,unlessotherwisestated.', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 0}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page2of21\\nBackground\\nBiological oscillators govern various biological processes,\\nsuch as cellular respiration, cardiac functions, and circa-\\ndian rhythms [ 1–3]. In the terms of synthetic biology, the\\nresearchofoscillatorysystemsismotivatedby(1)abetter\\nunderstanding of known biological systems [ 2,4–6], and\\n(2) by the development of systems that could potentially\\nbeusedinpracticalapplications[ 7,8].Animplementation\\nof the first synthetic repressilator by Elowitz and Leibler\\n[4] was, together with the synthetic toggle switch by\\nGardneretal.[ 9],animportantbreakthroughinsynthetic\\nbiology. Since then the focus has shifted from simpler to\\nmore complex biological systems [ 10]. For this reason,\\nthe development of robust and fast-response systems is\\nof vital importance. For example, Fink et al. [ 7]d e s i g n e d\\nan artificial fast-response system in mammalian cells that\\ncan respond to chemical signals in minutes rather than\\nhours. In the terms of optimization, this can be trans-\\nlated to multiobjective optimization [11]. Otero-Muras\\nand Banga [ 12]r e c e n t l yp r o p o s e dam u l t i o b j e c t i v eo p t i -\\nmization framework for synthetic biology based on the\\nPareto optimality. This framework is however limited to\\nthe library of synthetic parts, which can be a potential\\nlimitation. Moreover, when designing complex biological\\nsystems, the desired modes of behavior should not be\\nthe only criteria. One should also take into an account\\nthe system’s robustness ,i . e .i t ss t a b i l i t yi nt e r m so fc o r -\\nrect behavior for a large range of different perturbations\\nof extrinsic and intrinsic factors. If two systems exhibit\\nthesamerequireddynamics,thenthemorerobustsystem\\nshouldbepreferred[ 13].Thisallowsforthedevelopment\\nofmoreefficientandstablebiologicalsystems.Inorderto\\ndetermine the robustness of the system, one must be able\\ntoefficientlyexploreandcharacterizeitsparameterspace,\\nfor which mathematical modeling is usually applied [ 14].\\nTo find the optimal parameters that exhibit the desired\\nbehavior, different heuristic approaches, such as genetic\\nalgorithms (GAs) can be used [ 14–16]. While GAs have\\nnumerous applications, they usually provide only a single\\nnear-optimal solution. However, these approaches do not\\ngiveusaninsightintotheshapeofthesolutionspaceand\\ntherobustnessoftheacquiredsolution.Otherapproaches\\nare focused on the efficient investigation of the whole\\nparameter regions for which the system displays some\\npredefinedbehavior.Theseregionsdefineso-calledviable\\nparameter space. Identification of the viable parameter\\nspace allows for a more thorough analysis in the context\\nof system’s robustness, sensitivity and possible modes of\\nbehavior [ 13,17,18]. Schillings et al. [ 18]u s e da d a p t i v e\\nSmolyak interpolation that relies on sparse polynomial\\napproximations to characterize the solution space of bio-\\nchemicalnetworks.Theassumptionhereisthatthefunc-\\ntion we want to interpolate is sufficiently smooth, which\\nis not always the case. This problem can to some extentbeaddressedwiththeadaptiveinterpolation.Lietal.[ 19]\\nintroduced structural and correlative sensitivity analysis\\n(SCSA) which belongs to the family of global sensitivity\\nmethods based on the decomposition of variance. Since\\nthe viable parameter spaces only represent a small frac-\\ntion of all feasible solutions, we are more interested in\\nviable regions, and not on the solution space as a whole.\\nHafner et al. [ 13] developed a ’glocal’ robustness analy-\\nsis and model discrimination method that can be used\\nfor the analysis of circadian as well as other oscillators.\\nThis approach allows us to efficiently explore the model’s\\nparameter space and assess its robustness. One of its\\nmajor drawbacks is that it is not applicable to biologi-\\ncal systems with high dimensional and poorly connected\\nviable parameter regions. Two regions are poorly con-\\nnected if one cannot traverse the solution space from', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 1}),\n",
       " Document(page_content='cal systems with high dimensional and poorly connected\\nviable parameter regions. Two regions are poorly con-\\nnected if one cannot traverse the solution space from\\none viable region to the other with the arbitrarily small\\nsteps, while constantly preserving the viability of the cur-\\nrentsolution.Thisisnotproblematicfortheevolutionary\\ndeveloped systems, where the viable solution space is\\nusually connected because natural systems have evolved\\nthrough small, gradual changes of individual biochemical\\nparameters. And while this may be true for the naturally\\noccurring motifs, it is not necessarily the case for the\\nsyntheticallydevelopedgeneregulatorynetworks(GRNs).\\nWhen designing synthetic GRNs, one could choose dif-\\nferent parts, e.g., transcription factors (TFs) with similar\\nbehavioranddifferentkineticproperties,suchasbinding-\\nsiteaffinitiesanddegradationrates.Thisproblemwasalso\\naddressedbyZamora-Silleroetal.in[ 17],wheretheypro-\\nposedanefficientellipsoidbasedsampling.Thelimitation\\nofthisapproachisthattheincreasingdimensionalitymay\\nexponentiallyincreasethenumberofiterationsneededto\\nidentifyallviableparameterregions.Thiscanoccurifthe\\nviablesolutionspaceislooselyconnected.\\nHerein,wepresentanimproved’glocal’approachforthe\\ncomputationalanalysisofviableparameterspacesinhigh-\\ndimensionaldynamicalmodelsofbiologicalsystems.The\\nmethodology is based on the robustness estimation and\\nmodel analysis methodology described by Hafner et al.\\n[13]andisduetoexhaustivesamplingwithGAsandclus-\\nteringnotonlylimitedtomodelswithconnectedsolution\\nspaces. The methodology consists of multiple steps, i.e.\\n(1) the estimation of viable parameter regions with GAs,\\n(2) efficient exploration of viable regions with local sam-\\npling, and (3) robustness estimation for each of the viable\\nregions. Our methodology differs from the one intro-\\nduced by Hafner et al. [ 13]i nt w om a i na s p e c t s .F i r s t l y ,\\nweemployGAsfortheinitialestimationofviableparam-\\neter regions, whereas Hafner et al. relies on the literature\\na v a i l a b l ed a t a.T h ep r o b l e mi st h a tt h en u m be ro fa l r e a d y\\npublishedviableparametervaluescanbequitelimitedfor\\na particular system, and can guide the exploration in the\\nwrong direction. The second important difference is that', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 1}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page3of21\\nour approach can account for and discriminate between\\nmultiple poorly connected viable regions, whereas the\\n’glocal’ methodology by Hafner et al. is limited to a sin-\\ngle viable region. Moreover, our methodology can be\\napplied to the systems that exhibit multiple modes of\\nbehavior, such as alternative current (AC)-direct current\\n(DC)circuit[ 20].Weevaluatetheproposedmethodology\\non the repressilator model, on the model of the AC-\\nDC circuit that can switch between the oscillatory and\\nbistable behavior as described in [ 20]a n do nt h em o d e l\\nof the biological edge-triggered D flip-flop in a master-\\nslaveconfigurationproposedbyMagdevskaetal.[ 14].We\\nperform the analysis of viable regions of the repressila-\\ntor model with different cost functions. We analyze four\\ndifferent versions of the D flip-flop model, which differ\\nin the functional forms describing the protein degrada-\\ntion (Michaelian versus linear functions) and transcrip-\\ntion factor binding at promoter level (competitive versus\\nindependent).Wevalidatetheresultsobtainedwithdeter-\\nministic simulations with the additional stochastic sim-\\nulations. These are performed on the randomly selected\\nsamples from the viable regions of each model. The pro-\\nposed methodology is efficient and thorough, and can be\\napplied for the model-to-model comparison in terms of\\ntheir robustness. Finally, it is not limited only to systems\\nthat exhibit oscillatory dynamics, but can be applied to\\ncomplexbiologicalsystemswitharbitrarydynamics.\\nMethods\\nThe proposed approach consists of multiple consecutive\\nsteps, namely (1) global estimation of viable parameter\\nregions, (2) efficient local sampling, and (3) robustness\\nanalysis (see Fig. 1). We apply our approach on mod-\\nels of repressilator, AC-DC circuit, and D flip-flop in a\\nmaster-slave configuration. Topologies of these models\\naredisplayedinFig. 2.\\nGlobalestimationofviableparameterregions\\nNominal parameter values can be extracted from the lit-\\nerature. We are more interested in the analysis of the\\nwhole solution space, rather than a single viable solu-\\ntion. Furthermore, in general, nominal values are often\\nunknown or only partially known. The viable parame-\\nter space is composed of parameter regions in solution\\nspace for which the system exhibits a predefined behav-\\nior. More formally, the parameter space can be defined as\\naCartesianproduct\\n/Theta1p=p\\n⨉\\ni=1/Theta1i,( 1 )\\nwhere /Theta1prepresents the whole parameter space, pis the\\nnumber of parameters and /Theta1irepresents a range of fea-\\nsible parameter values for the i-th parameter. Ranges of\\nbiochemical parameters used in our examples are shownin Table1. Each point in the parameter space represents\\na candidate for a viable solution. A parameter point θis\\nviableiffthegivenconditionismet\\nE(θ) <E0,( 2 )\\nwhereE(θ)is a cost value of candidate point and E0is\\na predefined threshold, that divides candidates based on\\ntheir qualitative behavior, e.g., does a candidate exhibit\\noscillatory behavior or not. The cost function must be\\ndesigned in a way to promote candidates that exhibit the\\ndesired behavior and in some cases to promote addi-\\ntional biological criteria such as biomass maximization.\\nForedge-triggeredDflip-flopinamaster-slaveconfigura-\\ntion, we defined a cost function in the frequency domain\\nas a mean squared error (MSE) between an ideal and the\\nobservedsignalofthebiologicalsystem\\nE(θ)=n/summationdisplay\\ni=1(ˆhi−hi)2,( 3 )\\nwherenis the maximal number of considered harmon-\\nics of observed response for candidate θ,a n d ˆhi,hiare\\ni-thharmonicsoftheobservedandidealresponse,respec-\\ntively. We have chosen the frequency domain since it\\nbetterdescribesthesignalinthetermofitsharmonicsand\\nthussimplifiestheanalysis.Anadditionalusefulproperty\\nof this cost function is its invariance to the phase of the\\noscillatingsignal.Nonetheless,thiscostfunctionisappro-\\npriateonlywhenweareabletodefinetheexactdynamics', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 2}),\n",
       " Document(page_content='of this cost function is its invariance to the phase of the\\noscillatingsignal.Nonetheless,thiscostfunctionisappro-\\npriateonlywhenweareabletodefinetheexactdynamics\\nwe would like to achieve for the response of a model. If\\nwe are interested in the system behavior in a more qual-\\nitative way, e.g., does a system exhibits stable oscillations\\nor not, then the cost function should be defined to cover\\nall possible configurations that dictate the correct behav-\\nior. For this reason, we defined the second cost function\\nin the frequency domain, which optimizes the difference\\nbetweentheneighboringpeaksandapeakprominenceof\\naresponsesignalinafrequencydomain\\nE(θ)=−1\\nPP/summationdisplay\\ni=1σi−P/summationdisplay\\ni=2(γi−γi−1),( 4 )\\nwhere γiis thei-th peak, Pi st h en u m b e ro fp e a k s ,a n d\\nσiis standard deviation for the i-th peak in the neigh-\\nboring window of size 3. Ideal sine signals have only\\none prominent peak. To avoid rewarding the signals with\\nmany peaks, we consider only average standard deviation\\nper peak. The minimal size 3 for the window was cho-\\nseninordertopromotehighandnarrowpeaks.Thiscost\\nfunction is defined more loosely in order to not overlook\\nthelargerangeofpossiblerigidoscillatingsignalsregard-\\nless of their amplitude and period, while still favoring\\nundamped oscillations with high amplitudes and a stable', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 2}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page4of21\\nFig.1Visualizationoftheproposedmethodology.Themethodologyisdepictedinthefollowingconsecutivesteps(1)globalestimationofviable\\nregionswithGA,(2)efficientexplorationofviableregionswithlocalsampling,and(3)robustnessanalysisforthediscoveredviableregions\\nFig.2Theschematicsofbiologicalrepressilator,AC-DCcircuitandproposedDflip-flopinamaster-slaveconfiguration.Figure( a)displaysthe\\nschematicdiagramofarepressilator.Proteins X,YandZinanegativefeedbackloopinhibitadjacentproteins.Figure( b)displaysthedesignof\\nAC-DCcircuit.Mutualinhibitionof XandYcanproducebistablebehavior.Figure( c)displaystheschematicofproposedbiologicalDflip-flopina\\nmaster-slaveconfiguration,where CLKisthesynchronizationsignaland distheinput.Proteins aandacrepresentthemastersegmentofthe\\nflip-flop,whereas qandqcrepresenttheslavesegment', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 3}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page5of21\\nTable1Possiblerangesofparametervaluesweusedinour\\nmodels\\nParameter Span Unit\\nTranscription 10−2-50 h−1\\nTranslation 10−2-50 h−1\\nProteinproduction 10−1-50 h−1\\nProteindegradation 10−3-50 h−1\\nmRNAdegradation 10−1-100 h−1\\nDissociationconstant 10−2-250 nM\\nMichaelisconstant 10−2-250 nM\\nProteaseconcentration 10-1000 nM\\nDilutionrate 0.6 h−1\\nHillcoefficient 1-5 -\\nReactionvolume( VR·NA)1 nM−1\\nValuesobtainedfrom[ 40,45,46]arearoughestimateofrealkineticratesfor\\nprokaryotesandeukaryotescombined.Nevertheless,weexcludedthepossibilityof\\nsomeextremecases,likeextremelylong-livedorunstableproteins. VRdescribes\\nreactionspacevolumeand NAAvogadroconstant.\\nperiods. We applied both cost functions to the repres-\\nsilator model to analyze the size and shape of its viable\\nparameterregions.\\nInthefirststepoftheproposedmethodology,theviable\\nregions are estimated by the GA. GAs are inspired by\\nnatural evolution and are often used to solve hard opti-\\nmization problems [ 15]. Subjects within the population\\ngradually evolve by the means of genetic operators, i.e.\\nmutation, reproduction, and selection. Our initial popu-\\nlation consisted of 5000 randomly generated candidates.\\nEachcandidate θwasrepresentedasavectorofbiochem-\\nical parameters mutated with predefined probability. In\\nthe literature mutation probabilities most often range on\\nthe order of 0.01 per position. While this is much higher\\nthan in biology, mutation rates should be chosen on the\\npropertiesandthedifficultyoftheproblemweareaiming\\nto solve [ 21]. In our case, every biochemical parame-\\nter was multiplied with a random value between 0.8 and\\n1.2 with the probability 0.75 (high mutation probability\\nwas set in order to promote greater exploration of solu-\\ntion space). Every parameter can, therefore, increase or\\ndecreaseineachiterationofGA.Reproductionwasimple-\\nmented using the two-point crossover. Unlike mutation,\\nwhich serves as a fine-tuning mechanism, the crossover\\nintroduces a certain amount of variability in the popula-\\ntion, which makes the problem less susceptible to local\\nextrema. At the end of every iteration of genetic algo-\\nrithm, subjects are evaluated with the appropriate cost\\nfunction, and only the fraction of the individuals are cho-\\nsen for the next generation by tournament selection. We\\nused the tournament size of a tenth of the entire popula-\\ntion.IncontrarytothetraditionaluseofGAs,wesampled\\nall viable subjects, from which the initial viable set ν(0)iscomposed. The exploration of solution space stops when\\nthe maximal number of generations is reached. To obtain\\nonly the approximate estimation of viable regions, the\\ntotal number of generations should not be too high. We\\nterminatedourGAafter10generations.\\nEfficientlocalsampling\\nSince the viable parameter space exploration with GAs\\nis biased towards the evolvability of the problem defined\\nwith the selection of a cost function, GAs are not appro-\\npriate for the final estimation of the solution space. To\\nobtain a more accurate estimate of the viable regions, the\\nsolution space is efficiently and thoroughly explored in\\nan iterative manner as described by Hafner et al. [ 13].\\nWe apply the Gaussian sampling in the direction of the\\nprincipalcomponentsoftheexploredsolutionspace\\nS(i)=/braceleftBig\\nE[ν(i−1)]+λ(i−1)ξj|j=1,...,N/bracerightBig\\n,( 5 )\\nwhereS(i)isasetcomposedofcandidatesforviablesolu-\\ntions ini-th iteration of size N(in our case N=105),\\nE[ν(i−1)] is a mean of the viable candidate solutions in\\nthe set ν(i−1)obtained from the previous iteration of the\\nsamplingprocess, ξjisthej-thGaussiansamplealongthe\\nprincipal components of viable set ν(i−1),a n d λ(i−1)is\\nthe variance scaling factor. Note that the scaled variance\\nshouldalwaysbegreaterthantheinitialvarianceofprinci-\\npalcomponentsinordertocoverthewholeviablesolution\\nspace. We set the initial value of λ(0)to 4 and decreased\\nit linearly to 2 in the last, in our case tenth iteration. In', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 4}),\n",
       " Document(page_content='palcomponentsinordertocoverthewholeviablesolution\\nspace. We set the initial value of λ(0)to 4 and decreased\\nit linearly to 2 in the last, in our case tenth iteration. In\\nthiswaywefocusonlyonthepotentialsolutionareasand\\navoidunnecessarysampling.Attheendofeveryiteration\\nnew viable set ν(i)is obtained by evaluating candidates in\\nsetS(i).Formoredetailspleasereferto[ 13].\\nTo distinguish between poorly connected regions we\\nused the K-means clustering algorithm. This allows for\\nmore efficient and accurate characterization of loosely\\nconnected viable regions. Since the K-means algorithm\\nexpects the number of means, i.e. clusters, one must esti-\\nmate the correct number of clusters the data is divided\\ninto. We tackled this problem with the gap statistic [ 22].\\nLet us define Wkas the within-cluster sum of squares\\naround the cluster means. Gap statistic is a cluster analy-\\nsis algorithm that compares log(Wk)with its expectation\\nunder the appropriate reference null distribution of the\\ndataE∗[log(Wk)]\\nG(k)=E∗[log(Wk)]−log(Wk).( 6 )\\nFor example consider clustering nuniformly distributed\\npointsin pdimensions.Expectedvalueof log(Wk)isthen\\nE∗/bracketleftbig\\nlog(Wk)/bracketrightbig\\n=log(pn/12)−(2/p)log(k)+A,( 7 )\\nwherekis a number of clusters and Ais a constant. If\\nin reality our data consists of Kwell separated clusters,', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 4}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page6of21\\nthenthelog(Wk)willdecreasefasterthanitsexpectedrate\\nE∗[log(Wk)]f o rk≤Kand slower for k>K.H e n c e ,\\nwhenk=Kthe gap will be the largest. Similarly, we\\nset the number of clusters in our data according to the\\nlargest gap obtained by gap statistic. In [ 22], the authors\\nproposed two different choices for the reference distribu-\\ntion,namely(1)generateeachreferencefeatureuniformly\\nover the range of the observed values, and (2) generate\\nthe reference features from a uniform distribution over a\\nbounding hyper-box Baligned with the principal compo-\\nnents of the data. We have chosen the second approach\\nsince it is invariant to the rotation of the data. Note\\nthat the same technique was used by Hafner et al. [ 13]\\nfor the purposes of Monte Carlo integration. For more\\ninformationaboutgapstatisticsee[ 22].\\nWhenaviableareaisrecognizedasaseparateregion,it\\nis extensively explored regardless of any other regions by\\naniterativeproceduredescribedinEq.( 5).This,however,\\nposes a threat that explored viable regions overlap, which\\nmakesitharderfor model-to-model comparison.Alterna-\\ntively, we can compare only the most robusts regions of\\nbothmodelsorcombineallregionsintoasingleset.\\nDue to its computational complexity, clustering is\\nappliedonlywhenoneofthethreecriteriaismet,namely\\n(1)maximalnumberofiterationsisreached,(2)thenum-\\nber of viable points in next iteration is n-times smaller\\nthan number of points in the previous iteration (in our\\ncasen=10), or (3) the convergence of a region C\\nexceedsapredefinedthreshold C0.Ourmotivationbehind\\npoint (2) is that if samples are truly well separated, then\\nthe number of viable samples obtained by local sam-\\np l i n gw i l lb es i g n i f i c a n t l ys m a l l e ri fw es a m p l ep o i n t s\\nas a single region as opposed to separate sampling for\\neveryviableregion.Regardingthethirdpoint,wedefined\\nconvergenceforregion jastheFrobeniusnormofthedif-\\nference between current iteration principal components\\nPC(νj)(i)andthepreviousiterationprincipalcomponents\\nPC(νj)(i−1). The reason for the selection of this norm is\\nthatthemainprincipalcomponentsdirectlyinfluencethe\\ndirection of sampling in consecutive iterations. We thus\\ndefinedconvergenceas\\nC(i)\\nj=/vextendsingle/vextendsingle/vextendsingle/vextendsingle/vextendsingle/vextendsinglePC(νj)(i)−PC(νj)(i−1)/vextendsingle/vextendsingle/vextendsingle/vextendsingle/vextendsingle/vextendsingle\\nF.( 8 )\\nThis allows us to avoid unnecessary clustering at the\\nbeginning of the sampling process when the number of\\nviable candidates does not suffice for representative clus-\\nteringresults.Thisiterativeprocessstopswhenallregions\\naresufficientlyexplored,i.e.principalcomponentsdonot\\nchange over next iterations or the maximal number of\\niterationsisreached.Robustnessanalysis\\nA model of the robust system can cope with a large range\\nof changes of internal or external factors, e.g., temper-\\nature, that can directly influence its kinetic parameters.\\nThemorerobustmodelwillingeneralhaveagreatervol-\\nume of viable parameter regions and will be able to cope\\nwith perturbations of its parameters. To assess the model\\nrobustness, we estimate the volume of each viable region\\nVwithMonteCarlointegration[ 13,23,24]\\nVol(V)≈(|ν|/|S|)∗Vol(B),( 9 )\\nwhereVol(B)is the volume of a bounding hyper-box B\\naligned with principal components of the data, |S|is the\\nnumber of uniformly generated samples within Band|ν|\\ni st h en u m b e ro fv i a b l es a m p l e sw i t h i n S.B e c a u s ep r i n -\\ncipal components are perpendicular, the volume of Bcan\\nbe calculated as a product across lengths of hyper-box\\nedges. Sampling within Btherefore drastically reduces\\nthe number of samples required to estimate the viable\\nvolumeinthesameprecisionasopposedtouniformsam-\\npling of the whole parameter space. Note that instead of\\nboundingboxonecouldalsocalculatetheconvexhulland\\ncalculateitsvolume.Thisiscomputationallymoreexpen-', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 5}),\n",
       " Document(page_content='pling of the whole parameter space. Note that instead of\\nboundingboxonecouldalsocalculatetheconvexhulland\\ncalculateitsvolume.Thisiscomputationallymoreexpen-\\nsive compared to the bounding box approach, especially\\nwhen considering that we already calculated the princi-\\npal components in the previous step of our methodology.\\nTo makemodel-to-model comparisonwecannotdisregard\\nthat models can have a different number of dimensions.\\nForthisreason,wenormalizedtheviablevolumetoobtain\\na relative volume, which we calculated as a ratio between\\nthe viable volume and the volume of the total solution\\nspace\\nVol/prime(V)=Vol(V)/Vol(/Theta1p). (10)\\nOne question remains. What is the required number of\\nsamples in Monte Carlo simulations to assess the viable\\nvolume within the error δfor a confidence β?E v e r yu n i -\\nformlygeneratedsample θwithintheboundingbox Bcan\\nbeviewedasaBernoullirandomvariable\\nU=/braceleftBigg\\n1i fθisviable,\\n0o t h e r w i s e .(11)\\nThemeanof Uisthen μ=Vol(V)\\nVol(B).Thestandarddeviation\\nσis not known. Since Uis a Bernoulli random variable,\\nthemaximalvariance σ2is1/4.Thecentrallimittheorem\\ncanthenbeapplied', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 5}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page7of21\\nP/parenleftbigg/vextendsingle/vextendsingle/vextendsingle/vextendsingle|ν|\\n|S|∗Vol(B)−Vol(V)/vextendsingle/vextendsingle/vextendsingle/vextendsingle≤δ/parenrightbigg\\n=\\nP/parenleftbigg/vextendsingle/vextendsingle/vextendsingle/vextendsingle|ν|\\n|S|−μ/vextendsingle/vextendsingle/vextendsingle/vextendsingle≤δ\\nVol(B)/parenrightbigg\\n≈\\nP/parenleftbigg\\n|Z|≤δ∗√|S|\\nσ∗Vol(B)/parenrightbigg\\n≈\\n2P/parenleftbigg\\nZ≤2δ∗√|S|\\nVol(B)/parenrightbigg\\n−1, (12)\\nwhereZis a normally distributed random variable with\\nzero mean and unit variance. To get 2 P/parenleftBig\\nZ≤2δ∗√|S|\\nVol(B)/parenrightBig\\n−\\n1≥β,thenumberofsamples |S|mustbeatleast\\n|S|≥/vextendsingle/vextendsingle/vextendsingle/vextendsingle/vextendsingle/vextendsingle/Phi1−1/parenleftBig\\nβ−1\\n2/parenrightBig\\n∗Vol(B)\\n2δ/vextendsingle/vextendsingle/vextendsingle/vextendsingle/vextendsingle/vextendsingle2\\n, (13)\\nwhere /Phi1−1is inverse cumulative distribution function of\\nZ.Foraconfidencelevelof0.95,weget |S|≥/vextendsingle/vextendsingle/vextendsingle0.98∗Vol(B)\\nδ/vextendsingle/vextendsingle/vextendsingle2\\n.\\nIn other words, if we want to assess the viable volume\\nwithinonepercentofthetotalboundingboxvolumewith\\nthe confidence level of 0.95, Sshould contain at least 104\\nsamples.\\nResults\\nRepressilator\\nRepressilator is a simple GRN composed of an odd num-\\nber of repressors connected in a negative feedback loop.\\nThe simplest repressilator is a Goodwin oscillator, which\\nconsists of a single repressor [ 25]. We are interested in\\nthe repressilator with three repressors, first described byElowitzandLeibler[ 4].TheGRNtopologyoftherepressi-\\nlatorisdisplayedinFig. 2(a).Thedynamicsofrepressila-\\ntorcanbedescribedbythefollowingordinarydifferential\\nequations(ODEs)\\ndmX\\ndt=−δmmX+α\\n1+/parenleftBig\\nZ\\nKd/parenrightBign+α0, (14)\\ndmY\\ndt=−δmmY+α\\n1+/parenleftbigX\\nKd/parenrightbign+α0, (15)\\ndmZ\\ndt=−δmmZ+α\\n1+/parenleftbigY\\nKd/parenrightbign+α0, (16)\\ndX\\ndt= βmX−δpX, (17)\\ndY\\ndt= βmY−δpY, (18)\\ndZ\\ndt= βmZ−δpZ, (19)\\nwheremX,mYandmZare mRNA concentrations of\\nrepressors X,YandZ,r e s p e c t i v e l y .P a r a m e t e r δmis\\nmRNA degradation rate, δpprotein degradation rate, α\\ntranscription rate, α0leakage rate, βtranslation rate, and\\nKddissociation constant. Hill coefficient ndescribes the\\nstrength of cooperative binding between transcription\\nfactors. The deterministic simulation of repressilator is\\nshowninFig. 3.\\nTo analyze the effect of selecting different cost func-\\ntion on the size and shape of viable parameter regions,\\nwe studied two different scenarios. In the first scenario\\nwe applied cost function from the Eq. ( 3) to obtain a sig-\\nnal with amplitude of 300 nMand a period of 12 h.W e\\nconsidered a parameter point viable if the harmonics of\\nits response did not deviate, on average, for more than\\nFig.3Resultsofadeterministicsimulationperformedontherepressilatormodel.Blue,redandgreenrepresenttheconcentrationofproteins X,Y\\nandZ,respectively.Thedurationofsimulationis48 hwiththeinitialconditions X=150nM,Y=0nM,Z=0nM,andα=49.61h−1,α0=1.43\\nh−1,n=4.4,β=21.83h−1,δm=1.72h−1,δp=0.78h−1andKd=123.12nM.Proteinconcentrationoscillatewiththeperiodof12 handthe\\namplitudeof300 nM', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 6}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page8of21\\n10nMfrom the harmonics of the ideal signal. The first,\\nsecond and last iteration of exploration of viable regions\\nfor scenario 1 are displayed in Fig. 4(a). Parameter space\\nexploration is projected to the first two principal compo-\\nnents.Wecanobservethattheviablespaceiswelldefined\\nand consists of a single region, which is also confirmed\\nby a gap statistic that predicted one optimal cluster. Our\\nresultsareconsistentwiththefindingsofotherresearches\\n[13,17]. For the second scenario, we used Eq. ( 4)a sa\\ncostfunctionandconsideredparameterpointviableifthe\\namplitude of its response was roughly between 200 nM\\nand400nM,anditscostvaluebelow −200.Weestimated\\nthe amplitude of the signal with the size of its leading\\nharmonic. The exploration of viable regions for the sec-\\nond scenario is displayed in Fig. 4(b). Here, similarly to\\nthe first scenario, the solution space is well defined and\\nconnected. The obtained solution spaces can be directly\\ncomparedonthebasisoftheirvolume.Theviablevolume\\nfor scenario 2 is by our observations approximately 20-\\nfold larger than for scenario 1. Since the second scenario\\nhas a more loosely defined cost function, this coincides\\nwith our expectations. We can compare the volumes in\\nFig.5,i nw h i c hw ep r o j e c t e dt w ov i a b l er e g i o n st ot h e i r\\nmain principal components. Figure 6represents the box-\\nplots of parameter values for both cases. We can observe\\nthattherangeofvaluestheparametersspan,isinthesame\\nrange for both regions, except for the parameter δp.I n\\nthesecondscenario, δphassignificantlygreaterspancom-\\nparedtothefirstone,whichisalsothemainreasonbehind\\nthedifferenceintheviablevolumes.\\nDeterministic simulation approaches are only able to\\ndescribe average response of the system and do notdirectly account for the noise influences. We addition-\\nally validated the obtained results with the execution of\\nstochastic simulations using stochastic simulation algo-\\nrithm (SSA) [ 26,27] and its quasi steady-state approxi-\\nmation (QSSA) [ 28]. Latter allowed us to directly project\\nthe deterministic reaction system to its stochastic equiv-\\nalent. We randomly selected nine points from the viable\\nsolutionspaceandthreepointsoutsidetheviablesolution\\nspace to compare the results of stochastic and determin-\\nistic simulations (see Fig. 7). We repeated each stochastic\\nsimulation for a hundred times and measured the aver-\\nageamplitudesandperiods,whicharevisualizedinFig. 8.\\nPoints sampled from the feasible parameter regions per-\\ntainedtheoscillatorybehavioralsointhestochasticsimu-\\nlations.Moreover,periodswerewithinthesamerangesas\\nin the deterministic simulations. Larger amplitudes were\\nobservedinstochasticsimulationresults,presumablydue\\nto intrinsic noise, which is not regarded in determinis-\\nticsimulations.Pointswhichreflectedstationarybehavior\\nin deterministic simulations pertained this dynamics in\\nstochasticsimulationsaswell.\\nAC-DCcircuitexhibitsbistabilityandoscillations\\nThe AC-DC circuit described by Panovska-Griffiths et al.\\n[20] presents the combination of the toggle switch and\\nthe repressilator circuit. This GRN pattern has a natu-\\nral role in the determination of the spatial organization\\nof cell type generation and aids in the tissue development\\nof the ventral regions of the vertebrate neural tube [ 20].\\nThe topology of this circuit allows switch-like as well as\\noscillatory behavior by changing the concentration of the\\nsignalingmolecule S.Moreover,Perez-Carrascoetal.[ 29]\\nFig.4Explorationofviableregionsoftherepressilatormodels.Bluedotsrepresentviablesolutions,reddotsrepresentcandidatesforviable\\nsolutions.BluedotsinIteration1correspondtotheviablesolutionsobtainedwithGA.Figure( a)displaystheexplorationofviableregionsbasedon\\nthefirstcostfunction,Figure( b)displaystheexplorationofviableregionsbasedonthesecond,morelooselydefinedcostfunction', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 7}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page9of21\\nFig.5Theprojectionofrepressilatorviableregionsonthetwomainprincipalcomponentsfordifferentcostfunctions.Bluedotsrepresentsamples\\nobtainedwiththefirstcostfunction(Eq. 3),greendotsrepresentsamplesobtainedwiththesecondcostfunction(Eq. 4).Whilebothviableregions\\noverlap,theviableregionobtainedwiththefirstcostfunctionisclearlysmaller\\nFig.6Boxplotsofviableparameterregionsforthedeterministicmodelofrepressilator.Theparametersthatweobservedare α,α0,β,δm,δp,nand\\nKd.Wecanseethattheparametersspanapproximatelyoverthesamerangeofvalues,excepttheparameter δp,whichhassignificantlysmaller\\nrangeforviableregion1comparedtotheviableregion2.Viableregion1correspondstoscenario1,region2correspondstoscenario2', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 8}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page10of21\\nFig.7Comparisonofstochasticanddeterministicsimulationsperformedontherepressilatormodel.Inthefirstthreerowstheparameterswere\\nrandomlysampledfromtheviablesolutionspace.Thelastrowpresentstheresultsofthesimulationswheretheparametersweresampledoutside\\ntheviablesolutionspace.Solidlinespresenttheresultsofstochasticsimulationsanddashedlinespresenttheresultsofdeterministicsimulati ons.\\nEachofthelinespresentstheevolutionofprotein Xconcentrations\\nFig.8Boxplotsvisualizingtherepressilatoramplitudesandperiodsobtainedwithstochasticsimulations.Figure( a)presentstheaverageoscillation\\namplitudesandFigure( b)theaverageoscillationperiods.SamplenumberscorrespondtothenumbersusedinFig. 7.Onlyviablesamplesare\\ndisplayedhere.Ahundredsimulationswereperformedforeachsample.Averageoscillationamplitudesandperiodsweremeasuredineachofthe\\nsimulations.Boldblacklinespresenttheoscillationamplitudesandperiodsobservedindeterministicsimulations', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 9}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page11of21\\nshowedthatthecoexistenceofoscillatoryandstablegene\\nexpression can in the dependence of the intrinsic noise\\ngiverisetothecoherenceofoscillations.Weareinterested\\nin the multi-modal behavior of the AC-DC circuit. For\\nthe sake of simplicity, we excluded the signaling molecule\\nfrom the model and focused on two modes of behavior,\\ni.e.bistableandoscillatorybehavior.Theschematicofthe\\nAC-DC circuit is displayed in Fig. 2(b). The dynamics\\nof the AC-DC circuit can be described by the following\\nequations\\ndmX\\ndt=−δmmX+α\\n1+/parenleftBig\\nZ\\nKda/parenrightBign\\n+/parenleftBig\\nY\\nKdb/parenrightBign, (20)\\ndmY\\ndt=− δmmY+α\\n1+/parenleftBig\\nX\\nKdc/parenrightBign, (21)\\ndmZ\\ndt=− δmmZ+α\\n1+/parenleftBig\\nY\\nKdd/parenrightBign, (22)\\ndX\\ndt= βmX−δpX, (23)\\ndY\\ndt= βmY−δpY, (24)\\ndZ\\ndt= βmZ−δpZ, (25)\\nwheremX,mYandmZaremRNAconcentrationsofpro-\\nteinsX,YandZ,r e s p e c t i v e l y .P a r a m e t e r δmis mRNA\\ndegradationrate, δpproteindegradationrate, αtranscrip-\\ntion rate, βtranslation rate, and nis a Hill coefficient.\\nKda,Kdb,Kdc,Kddare dissociation constants. Notice the\\nsimilarity to the repressilator model, whereas the third\\nand the second term in the denominator of Eqs. ( 20,21)correspond to the toggle switch dynamics. The results of\\ndeterministic simulations for both modes of behavior are\\ndisplayed in Fig. 9. The same model can exhibit different\\nmodes of behavior with the same initial conditions and\\ndifferentkineticparametervalues.\\nWe regarded a parameter point viable if its response\\nexhibited oscillations or bistability. We tested the bista-\\nbility with two scenarios. First, the initial concentration\\nof protein Xwas low and Ywas high and vice versa in\\nthe second scenario. Throughout the simulation, the pro-\\ntein with the initial high concentration should stabilize\\nat 400nMand the protein with initial low concentra-\\ntion should stabilize at 0 nM. To optimize the bistable\\nbehavior we resorted to the cost function (Eq. 3), where\\nwe directly compared the response in the time domain\\ninstead of in the frequency domain. If the observed\\nresponse did not deviate, on average, from the ideal sig-\\nnal for more than 4 nM, we considered a parameter\\npointviable.\\nEquivalently,weconsideredaparameterpointviableifit\\nexhibited oscillations. We applied the cost function from\\nEq. (3). We considered only the first ten harmonics and\\ntreated a point in parameter space viable if the differ-\\nence between the harmonics of its response and the ideal\\nresponse did not exceed 15 nMper harmonic on average.\\nWesettheamplitudeoftheidealsignalto200 nMandthe\\nperiod to 12 h. We thus defined two cost functions, one\\nf o rea c hsc e n a ri oa n dr a nG At wi c e .I nt h el oca ls a m p l i n g\\nstep, we concatenated all viable solutions together and\\ntreated every sample from the parameter space equally,\\nregardless of its behavior, i.e. oscillations or bistability.\\nBecausethenumberofviablesamplesinthefirstiteration\\nFig.9ResultsofadeterministicsimulationsperformedontheAC-DCcircuitmodel.Blue,redandgreenrepresenttheconcentrationofproteins X,Y\\nandZ,respectively.Thedurationofsimulationis48 hwiththeinitialconditions X=37nM,Y=280nM,Z=280nM.Figure(a)showsthebistable\\ndynamicsforapointintheparameterspace α=5.91h−1,n=4.16,β=47.35h−1,δm=0.63h−1,δp=1.12h−1,Kda=165.03nM,Kdb=111.2\\nnM,Kdc=101.5nM,Kdd=0.48nM.Theinitialconcentrationofprotein Xislowand Yishigh.Throughtimetheconcentrationsofproteins XandY\\nbalancedat0and400 nM,respectively.Figure( b)showstheoscillatingdynamicsfortheparameterpoint α=26.26h−1,n=3.82,β=18.14h−1,\\nδm=0.92h−1,δp=1.29h−1,Kda=151.42nM,Kdb=197.61nM,Kdc=41.88nM,Kdd=11.7nM.Proteins XandZhavethesameamplitudeof\\n200nM,whiletheamplitudeofprotein Yisaround80 nM.Allproteinshavethesameperiodof12hours', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 10}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page12of21\\noflocalsamplingwastolow(10-timessmaller),clustering\\nwas performed and two distinct regions were obtained.\\nAfterthat,eachregionwasexploredseparatelyfor10iter-\\nations and the exploration was terminated regardless of\\nthe gapstatistic predictions.Figure 10displaysthecourse\\nof the exploration of viable regions. Our approach cor-\\nrectly identified both viable regions and explored them\\nindependently.\\nWe also assessed the robustness to the perturbation\\nof parameters for both regions. A viable region with the\\noscillatorybehaviorhasanapproximately500-foldgreater\\nvolume that the viable regions with bistable dynamics.\\nThiscanbeconfirmedwiththeboxplotsoftheparameters\\nspans in both viable regions (see Fig. 11). Transcription\\n(α), translation (β),m R N Ad e g r a d a t i o n (δm)and disso-\\nciation constant (Kdd)have significantly smaller span for\\nthe bistable region than for the region with the oscilla-\\ntorybehavior.OurresultsindicatethattheAC-DCmodel\\ntendstotheoscillationsmorethantothebistability.\\nWe validated the obtained results with the execution\\nof stochastic simulations in a similar way as in the\\nrepressilator model. We randomly selected six random\\npoints which reflected oscillatory behavior and six ran-\\ndompointswhichreflectedbistablebehaviorinthedeter-\\nministic simulations. All of the selected samples reflected\\nthe predefined dynamics in stochastic simulations as well\\n(seeFig.12).Werepeatedeachstochasticsimulationfora\\nhundred times and measured the average amplitudes and\\nperiodsforthesamplesthatreflectedoscillatorybehavior,\\nandaveragedistancebetweenthestablestatesforthesamples\\nthat reflected bistability. The obtained results are visual-\\nized in Fig. 13. As in the case of the repressilator model,\\nlarger amplitudes were observed in stochastic simulation\\nresults,whiletheperiodswereapproximatelythesame.\\nEdge-triggereddflip-flopinamaster-slaveconfiguration\\nIn electronics, D (delay) flip-flop is a memory circuit\\nthat exhibits two stable states and can thus store 1 bitof information. D flip-flop delays the input bit for one\\nc l o c kc y c l ea n di st r i g g e r e db yh i g ho rl o wc l o c kl e v e l s .\\nThe clock signal (CLK) is a synchronization signal fre-\\nquentlyusedinelectronics.Thesensitivityonhighorlow\\nlevels of CLK can be problematic when the input signal\\nis not exactly synchronized with CLK, which can result\\ninunpredictablebehavior.Themaster-slaveconfiguration\\nsolves this problem by connecting two flip-flops succes-\\nsively. The master flip-flop is triggered on the low CLK\\nlevelsandtheslaveflip-floponthehighCLKlevels.TheD\\nflip-flop in a master-slave configuration is therefore trig-\\ngeredsolelyonthepositiveedgeoftheCLKsignal.Inbio-\\nlogicalterms,theCLKrepresentsasynchronizationsignal\\nthat governs periodic cellular processes. A determinis-\\ntic model of biological edge-triggered D flip-flop based\\non ODEs was already established and studied by Magde-\\nvska et al. [ 14]. Recently, Andrews et al. [ 30] demon-\\nstratedtheimplementationofsequentiallogicincellswith\\ngenetic Negated OR (NOR) gates. Among others, they\\nimplemented and successfully verified the correct behav-\\ni o ro fag a t e dDf l i p - f l o pi n Escherichia coli (E. coli).\\nTheir flip-flop correctly switched and maintained states\\nfor more than 2 days, and accurately responded to a\\nsynchronization signal. Nonetheless, there are two dis-\\ntinctionsbetweentheirimplementationandoneproposed\\nby Magdevska et al. Flip-flop by Andrews et al. is con-\\nstructed with NOR gates, and responds only on the high\\nlevels of the synchronization signal, whereas Magdevska\\netal.proposedtheedge-triggeredflip-flop,whichiscom-\\nposed of two bistable switches regulated with delay, i.e.\\nd, and synchronization, i.e. CLK signals. In the latter,\\nwhend=qc, flip-flop oscillates between a high and low\\ns t a t ew i t ht w i c et h ep e r i o do fC L Ks i g n a l .I nt h i sw a y ,\\nflip-flop acts as a 1-bit counter. We used this property', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 11}),\n",
       " Document(page_content='whend=qc, flip-flop oscillates between a high and low\\ns t a t ew i t ht w i c et h ep e r i o do fC L Ks i g n a l .I nt h i sw a y ,\\nflip-flop acts as a 1-bit counter. We used this property\\nto obtain the desired dynamics with GAs. We applied\\nour methodology on the flip-flop topology described\\nin [14], which can be presented with the following\\nODE-basedmodel\\nFig.10ExplorationofviableregionsfortheAC-DCcircuit.Blueandgreendotsrepresentviablesolutions,reddotsrepresentcandidatesforviable\\nsolutions.BluedotsinIteration1and10correspondtoviablesolutionswithbistabledynamics,whilegreendotscorrespondtosolutionswith\\noscillatorydynamics', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 11}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page13of21\\nFig.11BoxplotsofvalueseachparametercanspaninbothviableregionsforthedeterministicmodelofAC-DCcircuit.Theobservedparameters\\nareα,n,β,δm,δp,Kda,Kdb,KdcandKdd.Wecanseethattheparametersspangenerallyoverdifferentareasforbothviableregions.Region1is\\nbistable,region2oscillates\\nda\\ndt=α1/parenleftBig\\nd\\nKd/parenrightBign\\n1+/parenleftBig\\nd\\nKd/parenrightBign\\n+/parenleftBig\\nCLK\\nKd/parenrightBign\\n+/Omega11/parenleftBig\\nd\\nKd/parenrightBign/parenleftBig\\nCLK\\nKd/parenrightBign+α21\\n1+/parenleftbigac\\nKd/parenrightbign−\\nafdeg(/Omega12,δ1,δdil,E,KM,a+ac+q+qc)\\n(26)\\ndac\\ndt=α11\\n1+/parenleftBig\\nd\\nKd/parenrightBign\\n+/parenleftBig\\nCLK\\nKd/parenrightBign\\n+/Omega11/parenleftBig\\nd\\nKd/parenrightBign/parenleftBig\\nCLK\\nKd/parenrightBign+α21\\n1+/parenleftbiga\\nKd/parenrightbign−\\nacfdeg(/Omega12,δ1,δdil,E,KM,a+ac+q+qc)\\n(27)\\ndq\\ndt=α3/parenleftbiga\\nKd/parenrightbign/parenleftBig\\nCLK\\nKd/parenrightBign\\n1+/parenleftbiga\\nKd/parenrightbign+/parenleftBig\\nCLK\\nKd/parenrightBign\\n+/parenleftbiga\\nKd/parenrightbign/parenleftBig\\nCLK\\nKd/parenrightBign+α41\\n1+/parenleftbigqc\\nKd/parenrightbign−\\nqfdeg(/Omega12,δ2,δdil,E,KM,a+ac+q+qc),(28)\\ndqc\\ndt=α3/parenleftbigac\\nKd/parenrightbign/parenleftBig\\nCLK\\nKd/parenrightBign\\n1+/parenleftbigac\\nKd/parenrightbign+/parenleftBig\\nCLK\\nKd/parenrightBign\\n+/parenleftbigac\\nKd/parenrightbign/parenleftBig\\nCLK\\nKd/parenrightBign+α41\\n1+(q\\nKd)n−\\nqcfdeg(/Omega12,δ2,δdil,E,KM,a+ac+q+qc),(29)\\nwhere\\nfdeg(/Omega1,δ,δdil,E,KM,P)=/braceleftbiggδE\\nKM+P+δdil;if/Omega1=0\\nδ; if/Omega1=1.(30)Parameters α1,α2,α3andα4are the expression rates of\\nproteinsa,ac,qandqc,KdandKMare dissociation and\\nMichaelis constants, nis Hill coefficient, δ1,δ2are degra-\\ndation rates of the observed proteins, δdilis dilution rate\\nandEis the total protease concentration. We introduce\\nparameters /Omega11and/Omega12to study the effects of different\\nfunctional forms describing the binding of transcription\\nfactors to the promoters and the effects of different mod-\\nels of protein degradation. Namely, when /Omega11equals 0 we\\npresume competitive binding of transcription factors to\\nthepromotersregulatingtheexpressionof aandac.When\\n/Omega11equals 1 independent binding is presumed. Parameter\\n/Omega12definesthedegradationmodelforwhichweuseeither\\nlinear or Michaelian form (see Eq. 30). The Michaelian\\nmodel of degradation was derived in the same manner\\nas described in [ 31]a n d[32]. The schematic of a general\\nbiological D flip-flop is shown in Fig. 2(c). Results of a\\ndeterministicsimulationfortheflip-flopmodelwithinde-\\npendent binding of transcription factors (/Omega11=1)and\\nlineardegradationterm (/Omega12=1)areshowninFig. 14.\\nParameters /Omega11and/Omega12definedifferentfunctionalforms\\nof the same flip-flop model. We were interested in the\\ncomparison of the solution spaces for each of the func-\\ntional forms. We evaluated the viable solution space and', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 12}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page14of21\\nFig.12ComparisonofstochasticanddeterministicsimulationsperformedontheAC-DCcircuitmodel.Inthefirsttworowstheparameterswere\\nrandomlysampledfromtheviablesolutionspacedefiningoscillatorybehavior(region2).Inthelasttworowstheparameterswererandomly\\nsampledfromtheviablesolutionspacedefiningbistablebehavior(region1).Solidlinespresenttheresultsofstochasticsimulationsanddashed\\nlinespresenttheresultsofdeterministicsimulations.Eachofthelinespresentstheevolutionofprotein X(green)and Y(red)concentrations\\nits volume for each of the four combinations, i.e compet-\\nitive versus independent binding and Michaelian versus\\nlinear protein degradation. To obtain the viable set of\\nparameters,weadoptedthecostfunctionfromEq.( 3)and\\ndeemed the parameter point non-viable if the difference\\nbetween the first ten harmonics of its response and the\\nideal signal exceeded, on average, 10 nMs per harmonic.\\nTheamplitudeoftheidealsignalwassetto50 nMandthe\\nperiodwassetto48 h.F i gu r e15presents the exploration\\nofviableregionsforeachofthefourfunctionalformcom-\\nbinations. Our approach identified single well-connected\\nsolutionspaceineachcase.\\nTo assess the robustness of different functional form\\ncombinations we evaluated the relative volumes of their\\nsolution spaces, which are presented in Table 2.T h e\\nresultsindicatethatMichaelianproteindegradationform\\nincreases the solutions space and thus the robustness of\\nthe proposed topology, but not as much as noncompet-\\nitive transcription factor binding at promoter level. Wewanted to confirm this hypothesis with the investigation\\nof parameter spans in each of the solution spaces. These\\nare presented in Fig. 16. The spans of protein degrada-\\ntion rates ( δ1andδ2) are larger when the Michaelian\\nprotein degradation model is presumed. When compar-\\ningthemodelswithMichaelianproteindegradationform,\\nnoncompetitive binding increases or pertains the span\\nof feasible parameter ranges. The same holds for the\\nlinear protein degradation form, except in the case of\\ndissociation constant, where larger span is observed for\\ncompetitivescenario.\\nWe additionally validated the obtained results with\\nthe execution of stochastic simulations. We randomly\\nselected three feasible random points for each functional\\nformcombination.Majorityofthesolutionspertainedthe\\nresponse observed in deterministic simulations, however,\\nthenoiseaffectedthesesolutionstoagreaterdegreethan\\nintherepressilatororAC-DCcircuitmodels(seeFig. 17).\\nOn the other hand, when execution large number of', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 13}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page15of21\\nFig.13BoxplotsvisualizingthestochasticresponseoftheAC-DCcircuit,namelyamplitudesandperiodsintheoscillatorymodeanddistance\\nbetweenthestablestatesinthebistablemode.Figures( a)and(b)correspondtotheoscillatorymodeofthecircuit.Figure( a)presentstheaverage\\noscillationamplitudesandFigure( b)theaverageoscillationperiods.Figure( c)correspondstothebistablemodeofthecircuitandpresentsthe\\naveragedistancesbetweenthestablestatesineachofthestochasticsimulations.SamplenumberscorrespondtothenumbersusedinFigure 12.A\\nhundredsimulationswereperformedforeachsample.Averageoscillationamplitudesandperiodsoraveragedistancesbetweenthestablestates\\nweremeasuredineachofthesimulations.Boldblacklinespresenttheresultsobservedindeterministicsimulationsforthecorrespondingsamples\\nstochastic simulations, approximately the same response\\nis observed as in deterministic simulations in all but one\\ntestedsample(seeFig. 18).\\nDiscussion\\nWedevelopedthecomputationalpipelinethatcanbeused\\nformodel-to-model comparisonintermsoftherobustness\\nto the perturbation of their kinetic parameters. Our work\\nis based on the already established ’glocal’ method intro-\\nducedbyHafneretal.[ 13].Inthefirststep,ourapproachroughlyestimatesglobalviablesolutionspacethroughthe\\noptimization with the proposed genetic algorithm. Next,\\nthe viable solution space is more thoroughly explored\\nwith efficient local sampling. Because the viable solu-\\ntionspacecanbehyzz‘potheticallyunconnected,wetook\\na step further and proposed clustering to perform fine-\\ngrained exploration. The size and shape of viable regions\\ncan then be utilized for the assessment of model robust-\\nness. We successfully applied and validated our method-\\nology on three distinct models that exhibit oscillatory\\nFig.14ResultsofadeterministicsimulationperformedonthemodeloftheDflip-flopinamaster-slaveconfigurationwithindependentbindingof\\ntranscriptionfactors (/Omega11=1)andlineardegradationterm (/Omega12=1).Black,black-dashed,greenandbluelinesrepresenttheconcentrationof\\nproteinsa,ac,qandqc,respectively.Greenlinerepresentthe CLKsignalwiththe24 hperiod.Theparametersofthemodelare α1=34.73h−1,\\nα2=49.36h−1,α3=32.73h−1,α4=49.54h−1,δ1=1.93h−1,δ2=0.69h−1,Kd=4.44nMandn=4.35', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 14}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page16of21\\nFig.15Theexplorationofviableregionsfortheedge-triggeredDflip-flop. /Omega11definescompetitive(0)ornoncompetitive(1)bindingsitesat\\npromoterlevel,and /Omega12Michaelian(0)orlinear(1)proteindegradationform.Bluedotsrepresenttheviablesolutions,reddotsrepresentcandidates\\nforviablesolutions\\nand/or bistable behavior. Our approach utilizes exhaus-\\ntive search of solution space, first by GAs and then with\\na prudent selection of samples, which is performed with\\nlocal sampling in the direction of main principal compo-\\nnents.Wedemonstratedtheapplicabilityoftheproposed\\napproach on three different deterministic ODE-based\\nmodels.\\nOne must also be aware of the potential drawbacks of\\nthe proposed approach. The first limitation is that the\\ngap statistic and clustering are not perfect. There is not\\nTable2Approximatedrelativevolumesofthefeasiblesolution\\nspacesintheDflip-flopmodel\\n/Omega11 /Omega12 Relativevolume (Vol/prime)\\n003 . 3 ·10−6\\n011 . 2 ·10−6\\n109 . 5 ·10−5\\n112 . 5 ·10−5\\nThefunctionalformsusedinthemodelaredefinedwithparameters /Omega11and/Omega12.\\nNamely,parameter /Omega11definescompetitive(0)ornoncompetitive(1)bindingsites\\natpromoterlevel.Parameter /Omega12definesMichaelian(0)orlinear(1)protein\\ndegradationform.a strict consensus of what constitutes a separate cluster\\nand different approacheswill cluster the same population\\nof samples differently. The drawback of gap statistic is\\nthatinordertotestthenullhypothesis,onemustassume\\nthe distribution of the data points. We have shown, that\\nif samples are uniformly distributed gap statistic always\\npredicts the optimal number of clusters, however, this is\\nnotalwaysthecasefornon-uniformlydistributeddata.To\\naddress this problem one could take methods for deter-\\nmining the number of clusters into account only partially\\nas a guide and select the number of clusters based on\\nobservations. By our experience gap statistic consistently\\noverestimated the number of clusters in the model of the\\nAC-DC circuit and D flip-flop model. We approached\\nthis conservatively by setting the maximum number of\\nclusters in the clustering step to 2 and by faster termi-\\nnation of the algorithm. The first viable region obtained\\nby GAs can be clustered only once. All consecutive\\nregionsarethendisregardedregardlessofthegapstatistic\\nprediction.\\nThesecondpossibledrawbackisthatGAsandtheprob-\\nability based sampling introduces non-determinism into\\nthe sampling process, i.e. repetitive runs with the same', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 15}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page17of21\\nFig.16BoxplotsofvalueseachparametercanspanintheviableregionsforthedeterministicmodelofDflip-flop. /Omega11definescompetitive(0)or\\nnoncompetitive(1)bindingsitesatpromoterlevel,and /Omega12Michaelian(0)orlinear(1)proteindegradationform.Parameters KMandEarenotused\\ninthelinearproteindegradationmodel\\nconfiguration can return different results. We mitigated\\nthis with high population size and with a high probability\\nof mutation and crossover in GA optimization. Similarly,\\nwe set the variance scaling factor λrelatively high at the\\nbeginning of our sampling process, when we were not as\\nconfident in the explored solution space, and gradually\\ndecreasedittowardstheend.\\nWe have shown that the size and shape of a viable solu-\\ntion space are directly dependent on the definition of\\ncost function and the threshold selection, which can be\\ndefined subjectively. We demonstrated the consequences\\nof different interpretations of viable solution with the\\ninvestigation of viable parameter regions on the same\\nrepressilator model for two different cost functions. The\\nfirstcostfunctionwasdefinedinastrictmanner,whereas\\nthe second one was defined more loosely. The viable\\nparameter region of the repressilator model was greater\\nwhenweemployedtheexplorationofviableregionsbased\\nonalooselydefinedcostfunction.Thiscoincideswithour\\nexpectationsandtosomedegreevalidatesthecorrectnessof our approach. In both cases, the parameter spans\\nwere roughly the same except for the degradation rate\\nδp(see Fig. 6), which in the end contributed to the dif-\\nference in the viable volume. Both cases have the same\\nHill coefficient ranges, 2 ≤n≤5. Higher values of\\nHillcoefficientcorrespondtothepositivecooperativityof\\ntranscription factors and a nonlinear response. The first\\nimplementation of a repressilator by Elowitz and Leibler\\n[4] achieved non-linearity with the selection of tran-\\nscription factors that function as oligomers, i.e. the TetR\\nrepressorproteinisadimer[ 33],andtheLacIrepressoris\\natetramer[ 34].\\nTo illustrate the exploration of viable regions with mul-\\ntiple well defined and unconnected clusters we analyzed\\nthe AC-DC model for two different modes of behavior.\\nOurmethodprovedtobeeffectiveinthisscenarioaswell.\\nBased on our observation, the AC-DC circuit tends more\\ntowards the oscillatory behavior than to bistability. We\\ncan assume this because the viable parameter volume is\\nsignificantly larger for the parameter solutions that yield', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 16}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page18of21\\nFig.17ComparisonofstochasticanddeterministicsimulationsperformedontheDflip-flopmodelwithdifferentfunctionalformsdescribing\\nproteindegradationandtranscriptionfactorbindingatpromoterlevel. /Omega11definescompetitive(0)ornoncompetitive(1)bindingsitesatpromoter\\nlevel,and /Omega12Michaelian(0)orlinear(1)proteindegradationform.Solidlinespresenttheresultsofstochasticsimulationsanddashedlinespresent\\ntheresultsofdeterministicsimulations.Eachofthelinespresentstheevolutionofprotein q(blue)and qc(red)concentrations\\noscillatory dynamics in comparison to the bistable solu-\\ntions. However, for the sake of simplicity, we excluded\\nthe signaling molecule Sfrom the model and thus dis-\\nregarded the switch like dynamics. It has already been\\nshown that by adjusting the strength and/or the con-\\ncentration of the signaling molecule the AC-DC circuit\\ncan exhibit both modes of behavior for the same sets\\nof parameters [ 29]. Our simulations clearly indicate that\\nin order to achieve either bistability or oscillations, the\\nsystem should reflect non-linear transcription response.\\nThis can be described with larger values of the Hill coef-\\nficient [35]. Other researchers have already addressed\\nthis problem. For example, Lebar et al. [ 36]d e s i g n e da\\nbistablegeneticswitchbasedondesignableDNA-binding\\ndomains of transcription-activator-like effectors (TALEs)\\n[37]. Since TALEs bind non-cooperatively as monomers,\\na simple mutual repressor-based toggle switch does not\\nsupport bistability. In order to introduce non-linearity\\nand achieve bistability, Lebar et al. designed a bistable\\nswitch with an additional positive feedback loop of TALE\\nrepressors.We investigated the robustness of the proposed D\\nflip-flop model in dependence of different functional\\nforms describing the protein degradation (Michaelian\\nversus linear functions) and the transcription factor-\\npromoter binding process (competitive versus indepen-\\ndent binding). The flip-flop model proposed in [ 14]w a s\\nextended into four different models upon which the pro-\\nposed methodology was executed. We used the obtained\\nresults to assess the relative volumes of feasible solutions\\nspaces in each of the models. These were used to per-\\nform the model-to-model comparison and to assess their\\nrobustness.WeshowedthatMichaeliandegradationform\\nincreases the chances of obtaining oscillations as already\\ndescribed in [ 38], but not as much as noncompetitive\\nbinding sites at promoter level [ 39]. When performing\\nstochasticsimulationsweobservedlargernoisesensitivity\\nas in the repressilator or AC-DC ciruit models. Solu-\\ntions more resilient to intrinsic noise could potentially\\nbe obtained with the selection of different cost function\\ndefining feasibility of the solution, i.e. with larger oscil-\\nlation amplitudes. The required amplitudes used in the', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 17}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page19of21\\nFig.18BoxplotsvisualizingthestochasticresponseoftheDflip-flopmodel,namelyamplitudesandperiodsofobservedoscillations. /Omega11defines\\ncompetitive(0)ornoncompetitive(1)bindingsitesatpromoterlevel,and /Omega12Michaelian(0)orlinear(1)proteindegradationform.Figuresin\\ncolumn(a)presenttheaverageoscillationamplitudesandfiguresincolumn( b)theaverageoscillationperiods.Samplenumberscorrespondtothe\\nnumbersusedinFig. 17.Ahundredsimulationswereperformedforeachsample.Averageoscillationamplitudesandperiodsweremeasuredin\\neachofthesimulations.Boldblacklinespresenttheresultsobservedindeterministicsimulationsforthecorrespondingsamples\\nD flip-flop model were set to 50 nMwhile amplitudes\\nbetween 200 and 400 nMwere regarded as feasible in the\\nrepressilator and AC-DC circuit models. Our results still\\nindicate that the D flip-flop is robust to perturbations of\\nitskineticparametersandthatthepossibilityofitsimple-\\nmentation in the biological host is promising. Andrews\\net al. [30] demonstrated the feasibility of D flip-flop in\\nE. coli. Still, their flip-flop is triggered on high signal\\nlevels and not on an edge of the synchronization sig-\\nnal. This makes the circuit hard to control, since the\\nhigh level of the synchronization signal needs to be long\\nenoughtotriggerthetransitionfromonestatetoanother\\nand at the same time short enough to prevent multiple\\nswitches.Ourresultsindicatethattheviablesolutionspaceofbio-\\nlogicaloscillatorsisgenerallywelldefinedandconnected,\\nwhich has been already confirmed by other researchers\\n[13,17].Thisisexpectedforthenaturallyoccurringmotifs\\nthat exhibit oscillations since they possibly evolved with\\nrandom mutations that contributed to the small grad-\\nual changes of kinetic parameters [ 40]. However, it is\\ninteresting to observe similar properties in the synthetic\\ncircuits. This can be partially explained with the fact that\\nt h ed e s i g ni st os o m ed e g r e ei n s p i r e db yt h es y s t e m s\\nwe can observe in nature. Except for the AC-DC circuit,\\nour models displayed single connected viable regions.\\nThe existence of multiple unconnected viable regions for\\nthe AC-DC circuit can be contributed to its capability', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 18}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page20of21\\nof multimodal behavior. The values parameters span are\\ngenerallydifferentforbothmodesofbehavior.\\nFor a biological system to be robust, it must be able\\nto withstand the fluctuations of biochemical parameters\\nduetoexternalfactors,intrinsicnoise,andsingle-cellvari-\\nability. Our methodology can give one the insight into\\nthe shape and size of the viable parameter regions, and\\ninto the overall robustness of a system. Our approach\\nhas, therefore, two main applications. Firstly, by knowing\\nthe effect of parameters on system behavior, one could\\nfine-tune the problematic parameters and use synthetic\\nconstructs, such as degradation tags to speed-up pro-\\nteindegradation[ 41],ordesignpartswithhigherbinding\\naffinities.Forexample,Finketal.[ 7]designedcoiledcoils\\nto increase the affinity between split proteases and thus\\nincreased the response of a system. However, approaches\\nto experimentally tune the value of a given parameter are\\nquitelimited,especiallyinapredictiveway.Secondly,one\\ncould use the proposed methodology to compare differ-\\nentsystemswithsimilarbehavioranddifferenttopologies.\\nFor example, simple bistable switch could be compared\\nwith the bistable switch with positive feedback loops\\nproposed by Lebar et al. [ 36]. The results of these com-\\nparisons could guide the researcher in the selection of\\nmore robust topologies and finally in the process of the\\nimplementationofreliablebiologicalcircuits.\\nWevalidatedourmethodologyonbiologicalGRNmod-\\nels that primarily exhibit oscillatory or bistable behavior.\\nHowever, it is not hard to see how one could adapt this\\napproach to cover other modes of behavior as well. By\\nmodifying the genetic algorithm and adjusting the cost\\nfunction, one can adapt our approach to a variety of\\ndynamical models and not only models of GRNs. We\\ndemonstrated the application of the proposed method-\\nology solely on the results obtained with deterministic\\nsimulations. These models describe the average response\\nofthesystemwithoutthenoiseinfluences[ 42].Thenoise\\ninfluencescanbetosomedegreeindirectlyanalyzedwith\\nthe analysis of parameter variability effects on the deter-\\nministic dynamics of the system [ 43]. In many cases it is\\nmoresuitabletousestochasticmodelingapproaches,such\\nasSSA[26].Theseapproachesdirectlydescribetheinher-\\nent stochasticity of biochemical reactions. In our case\\nstudiesweonlyusedthestochasticsimulationstovalidate\\nthe correctness of the results obtained with determinis-\\ntic approaches. However, these approaches could as well\\nbe used to generate the data upon which the proposed\\nmethodology would be applied. Moreover, the methodol-\\nogy could be used in a combination with any other either\\nexperimental or computational approach that is able to\\ngeneratetheresponseofthesystematthesampledparam-\\neter values. However, there are some potential drawbacks\\nin the straightforward application of these approaches.\\nDeterministic models will always yield the same responsefor a given set of initial conditions and parameter val-\\nues. Contrary, stochastic or experimental procedures will\\nalways differ to some degree even with the same con-\\nditions [27]. This means that multiple repetitions of the\\nsame experiment will be needed to achieve statistical sig-\\nnificance, which in turn increases the time complexity of\\nourmethodology.Mostoftheapproachesthatpresentan\\nalternativetodeterministicmodelsinherentlyincreasethe\\ntime complexity to generate the results even for a single\\nsimulation run. In order to at least partially circumvent\\nthis problem, one of the many parallelized adaptations\\nof SSA with lower computational complexity could be\\napplied[44].\\nConclusion\\nIn this paper we proposed a novel approach that can be\\nusedtoassesstheviableparameterregionsforanarbitrary\\nGRN model, and which can be applied in the design of\\nsyntheticbiologicalsystems.Identifiedparameterregions\\nallow us to compare different models in terms of their', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 19}),\n",
       " Document(page_content='GRN model, and which can be applied in the design of\\nsyntheticbiologicalsystems.Identifiedparameterregions\\nallow us to compare different models in terms of their\\nrobustness and also to identify the critical segments of\\nthe selected system. This can be exploited for the design\\nof more reliable and robust systems. For example, if the\\ndegradation rates of observed proteins are constrained to\\na small interval, one can then specifically focus on this\\nsegment by exploring the effects of different degradation\\ntags [41] and thus fine-tune the problematic parameters.\\nMoreover, our approach can be used as a foundation\\nfor other analyzes. For example, bifurcation or sensitivity\\nanalysiscanbedonemoreefficientlyandwithhigherpre-\\ncisionifoneisconfidentinthesizeandshapeoftheviable\\nsolutionspace.\\nAbbreviations\\nAC-DC:Alternativecurrent-directcurrent;CLK:Clock;GA:Geneticalgorithm;\\nGRN:Generegulatorynetwork;MSE:Meansquarederror;QSSA:Quasi\\nsteady-stateapproximation;SCSA:Structuralandcorrelativesensitivity\\nanalysis;SSA:Stochasticsimulationalgorithm;TF:Transcriptionfactor\\nAcknowledgements\\nResultspresentedhereareinthescopeofthePh.D.thesisthatisbeing\\npreparedbyŽigaPušnik.\\nAuthors’contributions\\nŽPdesignedthemethod,performedtheexperiments,andanalyzedthe\\nresults.ŽPandMModevisedthestudyandwrotethemanuscript.MMo\\nsupervisedthestudy.MMo,MMrandNZprovidedcriticalfeedbackand\\nhelpedshapetheresearch,analysisandmanuscript.Allauthorsreadand\\napprovedthefinalmanuscript.\\nFunding\\nTheresearchwaspartiallysupportedbythescientific-researchprogramme\\nPervasiveComputing(P2-0359)financedbytheSlovenianResearchAgencyin\\ntheyearsfrom2017to2023,bythebasicresearchprojectCholesteRORin\\nmetabolicliverdiseases(J1-9176)financedbytheSlovenianResearchAgency\\nintheyearsfrom2018to2021,andbytheinfrastructuregrantELIXIR\\nsupportedbyMRICUL(grantnumberI0-0022).\\nAvailabilityofdataandmaterials\\nAllsourcecodewithaccompanyingresultsisavailablefordownloadat:\\nhttps://github.com/zigapusnik/AnalysisOfViableParameterRegions .', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 19}),\n",
       " Document(page_content='Pušniketal. JournalofBiologicalEngineering           (2019) 13:75 Page21of21\\nEthicsapprovalandconsenttoparticipate\\nNotapplicable.\\nConsentforpublication\\nNotapplicable.\\nCompetinginterests\\nTheauthorsdeclarethattheyhavenocompetinginterests.\\nReceived:24April2019 Accepted:5September2019\\nReferences\\n1. LeloupJ-C, GonzeD, GoldbeterA.Limitcyclemodelsforcircadian\\nrhythmsbasedontranscriptionalregulationinDrosophilaand\\nNeurospora.JBiolRhythms.1999;14(6):433–48.\\n2. ChabotJR, PedrazaJM, LuitelP, VanOudenaardenA.Stochasticgene\\nexpressionout-of-steady-stateinthecyanobacterialcircadianclock.\\nNature.2007;450(7173):1249–52.\\n3. DaninoT, Mondragón-PalominoO, TsimringL, HastyJ.Asynchronized\\nquorumofgeneticclocks.Nature.2010;463(7279):326–30.\\n4. ElowitzMB, LeiblerS.Asyntheticoscillatorynetworkoftranscriptional\\nregulators.Nature.2000;403(6767):335–38.\\n5. ReppertSM, WeaverDR.Coordinationofcircadiantiminginmammals.\\nNature.2002;418(6901):935–41.\\n6. TaylorAF, TinsleyMR, WangF, HuangZ, ShowalterK.Dynamical\\nquorumsensingandsynchronizationinlargepopulationsofchemical\\noscillators.Science.2009;323(5914):614–7.\\n7. FinkT, Lonzari ´cJ, PraznikA, PlaperT, MerljakE, LebenK, JeralaN, Lebar\\nT, StrmšekŽ, LapentaF,etal.Designoffastproteolysis-basedsignaling\\nandlogiccircuitsinmammaliancells.NatChemBiol.2019;15(2):115–22.\\n8. TiggesM, Marquez-LagoTT, StellingJ, FusseneggerM.Atunable\\nsyntheticmammalianoscillator.Nature.2009;457(7227):309–12.\\n9. GardnerTS, CantorCR, CollinsJJ.Constructionofagenetictoggleswitch\\ninEscherichiacoli.Nature.2000;403(6767):339–42.\\n10. CameronDE, BashorCJ, CollinsJJ.Abriefhistoryofsyntheticbiology.\\nNatRevMicrobiol.2014;12(5):381–90.\\n11. KalyanmoyD, PratapA, AgarwalS, MeyarivanT.Afastandelitist\\nmultiobjectivegeneticalgorithm:NSGA-II.IEEETransEvolComput.\\n2002;6(2):182–97.\\n12. Otero-MurasI, BangaJR.Automateddesignframeworkforsynthetic\\nbiologyexploitingParetooptimality.ACSSyntheticBiol.2017;6(7):\\n1180–93.\\n13. HafnerM, KoepplH, HaslerM, WagnerA.’Glocal’robustnessanalysisand\\nmodeldiscriminationforcircadianoscillators.PLoSComputBiol.\\n2009;5(10):e1000534.\\n14. MagdevskaL, PušnikŽ, MrazM, ZimicN, MoškonM.Computational\\ndesignofsynchronoussequentialstructuresinbiologicalsystems.J\\nComputSci.2017;18:24–31.\\n15. KozaJR.GeneticProgramming:ontheProgrammingofComputersby\\nMeansofNaturalSelection.Cambridge,MassachusettsandLondon,\\nEngland:MITpress;1992.\\n16. StražarM, MrazM, ZimicN, MoškonM.Anadaptivegeneticalgorithm\\nforparameterestimationofbiologicaloscillatormodelstoachievetarget\\nquantitativesystemresponse.NaturalComput.2014;13(1):119–27.\\n17. Zamora-SilleroE, HafnerM, IbigA, StellingJ, WagnerA.Efficient\\ncharacterizationofhigh-dimensionalparameterspacesforsystems\\nbiology.BMCSystBiol.2011;5(1):142.\\n18. SchillingsC, SunnåkerM, StellingJ, SchwabC.Efficientcharacterization\\nofparametricuncertaintyofcomplex(bio)chemicalnetworks.PLoS\\nComputBiol.2015;11(8):e1004457.\\n19. LiG, RabitzH, YelvingtonPE, OluwoleOO, BaconF, KolbCE,\\nSchoendorfJ.Globalsensitivityanalysisforsystemswithindependent\\nand/orcorrelatedinputs.JPhysChemA.2010;114(19):6022–32.\\n20. Panovska-GriffithsJ, PageKM, BriscoeJ.Ageneregulatorymotifthat\\ngeneratesoscillatoryormultiwayswitchoutputs.JRoyalSocInt.\\n2013;10(79):20120826.\\n21. FloreanoD, MattiussiC.Bio-inspiredArtificialIntelligence:Theories,\\nMethods,andTechnologies.Cambridge,MassachusettsandLondon,\\nEngland:MITpress;2008.22. TibshiraniR, WaltherG, HastieT.Estimatingthenumberofclustersina\\ndatasetviathegapstatistic.JRoyalStatSoc:SerB(StatMethodol).\\n2001;63(2):411–23.\\n23. RobertC, CasellaG.MonteCarloStatisticalMethods.NewYork:Springer\\nScience&BusinessMedia;2010.\\n24. NealD.DeterminingsamplesizesforMonteCarlointegration.ColMathJ.\\n1993;24(3):254–9.\\n25. GoodwinBC.Oscillatorybehaviorinenzymaticcontrolprocesses.Adv\\nEnzymeRegulation.1965;3:425–37.\\n26. GillespieDT.Exactstochasticsimulationofcoupledchemicalreactions.J\\nPhysChem.1977;81(25):2340–61.\\n27. ElSamadH, KhammashM, PetzoldL, GillespieD.Stochasticmodelling\\nofgeneregulatorynetworks.IntJRobustNonlinearControl.2005;15(15):', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 20}),\n",
       " Document(page_content='PhysChem.1977;81(25):2340–61.\\n27. ElSamadH, KhammashM, PetzoldL, GillespieD.Stochasticmodelling\\nofgeneregulatorynetworks.IntJRobustNonlinearControl.2005;15(15):\\n691–711.\\n28. KimJK, Josi ´cK, BennettMR.Thevalidityofquasi-steady-state\\napproximationsindiscretestochasticsimulations.BiophysJ.2014;107(3):\\n783–93.\\n29. Perez-CarrascoR, BarnesCP, SchaerliY, IsalanM, BriscoeJ, PageKM.\\nCombiningatoggleswitchandarepressilatorwithintheAC-DCcircuit\\ngeneratesdistinctdynamicalbehaviors.CellSyst.2018;6(4):521–530.\\n30. AndrewsLB, NielsenAA, VoigtCA.Cellularcheckpointcontrolusing\\nprogrammablesequentiallogic.Science.2018;361(6408):eaap8987.\\n31. CooksonNA, MatherWH, DaninoT, Mondragón-PalominoO, Williams\\nRJ, TsimringLS, HastyJ.Queueingupforenzymaticprocessing:\\ncorrelatedsignalingthroughcoupleddegradation.MoleSystBiol.\\n2011;7(1):1–9.\\n32. TomazouM, BarahonaM, PolizziKM, StanG-B.Computationalre-design\\nofsyntheticgeneticoscillatorsforindependentamplitudeandfrequency\\nmodulation.CellSyst.2018;6(4):508–20.\\n33. RamosJL, Martínez-BuenoM, Molina-HenaresAJ, TeránW, WatanabeK,\\nZhangX, GallegosMT, BrennanR, TobesR.TheTetRfamilyof\\ntranscriptionalrepressors.MicrobiolMoleBiolRev.2005;69(2):326–56.\\n34. LewisM.Thelacrepressor.ComptesRendusBiologies.2005;328(6):\\n521–48.\\n35. PurcellO, SaveryNJ, GriersonCS, diBernardoM.Acomparativeanalysis\\nofsyntheticgeneticoscillators.JRoyalSocInt.2010;7(52):1503–24.\\n36. LebarT, BezeljakU, GolobA, JeralaM, KaduncL, PiršB, StražarM,\\nVuˇckoD, Zupan ˇciˇcU,BenˇcinaM,etal.Abistablegeneticswitchbased\\nondesignableDNA-bindingdomains.NatureCommun.2014;5:5007.\\n37. GargA, LohmuellerJJ, SilverPA, ArmelTZ.EngineeringsyntheticTAL\\neffectorswithorthogonaltargetsites.NucleicAcidsRes.2012;40(15):\\n7584–95.\\n38. WongWW, TsaiTY, LiaoJC.Single-cellzeroth-orderproteindegradation\\nenhancestherobustnessofsyntheticoscillator.MoleSystBiol.2007;3(1):\\n1–8.\\n39. MunteanuA, ConstanteM, IsalanM, SoléRV.Avoidingtranscription\\nfactorcompetitionatpromoterlevelincreasesthechancesofobtaining\\noscillation.BMCSystBiol.2010;4(1):66.\\n40. AlonU.AnIntroductiontoSystemsBiology:DesignPrinciplesof\\nBiologicalCircuits.London:ChapmanandHall/CR;2006.\\n41. McGinnessKE, BakerTA, SauerRT.Engineeringcontrollableprotein\\ndegradation.MoleCell.2006;22(5):701–7.\\n42. KaernM, BlakeWJ, CollinsJJ.Theengineeringofgeneregulatory\\nnetworks.AnnRevBiomedEng.2003;5(1):179–206.\\n43. SinghA, SoltaniM.Quantifyingintrinsicandextrinsicvariabilityin\\nstochasticgeneexpressionmodels.PLoSOne.2014;8(12):e84301.\\n44. PetroniM, ZimicN, MrazM, MoškonM.Stochasticsimulationalgorithm\\nforgeneregulatorynetworkswithmultiplebindingsites.JComputBiol.\\n2015;22(3):218–26.\\n45. BioNumbersDatabase.,GonzeD.FreeUniversityofBrussels,Belgium.\\n2019.http://homepages.ulb.ac.be/~dgonze/BIONUMBERS/bionumbers.\\nhtml.Accessed8Aug2019.\\n46. MiloR.,JorgensenP.,MoranU.,WeberG.,SpringerM.BioNumbers–the\\ndatabaseofkeynumbersinmolecularandcellbiology.NucleicAcidsRes.\\n2010;38(Databaseissue):D750–3. https://bionumbers.hms.harvard.edu/ .\\nAccessed8Aug2019.\\nPublisher’sNote\\nSpringerNatureremainsneutralwithregardtojurisdictionalclaimsin\\npublishedmapsandinstitutionalaffiliations.', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/GeneticAlgorithm.pdf', 'page': 20}),\n",
       " Document(page_content='Sensitivit yanalysis ofautonomous oscillations:\\napplication tobiochemical systems\\nBrian P.Ingalls\\nDepartmen tofApplied Mathematics\\nUniversityofWaterlo o\\nWaterlo o,Ontario\\nCanada N2L3G1\\nbingalls@math.uwaterloo.ca\\nAbstract\\nThisworkaddresses sensitivit yanalysis ofau-\\ntonomously oscillating biochemical systems by\\nextending previous treatmen tswhichconsidered\\nspecialcasesorempirical approximations tosys-\\ntemresponse.Building onresultsfromtheengi-\\nneeringliterature, ananalysisispresentedwhich\\naddresses sensitivit yofkeyfeatures oftheoscil-\\nlatorytrajectory: namelytheperiodandlocal\\nmaximumorminimumvaluesofspeciesconcen-\\ntrationsandreaction rates.Adiscussion ofsensi-\\ntivityinvariantsgeneralizes resultsfromsteady-\\nstatesensitivit yanalysis tothiscontext.\\n1Introduction\\nSensitivit yanalysisplaysanimportantroleinthe\\nstudyofbiochemicalsystems. Mathematical ap-\\nproachestosuchanalysis havebeenintroduced\\nthrough the¯eldsofMetabolicControlAnalysis\\n(MCA)[11,6]andBiochemicalSystems Theory\\n(BST)[18].Theseanalyses haveprovenuseful\\nbothbyprovidinginsightintothenatureand\\nfunction ofbiochemical systems aswellasby\\nservingasatooltopredicttheresultofinter-\\nventions,e.g.formetabolicengineering orphar-\\nmaceutic purposes.\\nInthestudyofbiochemicalnetworks,itisoften\\nthecasethatsteadystatebehaviourisofprimary\\ninterest(notleastbecausethisisthesimplest\\nbehaviourtoinvestigate experimentally).Asa\\nconsequence, thesensitivit yanalysis ofsystems\\natsteadystatehasprovenadequate formany\\ninvestigations.\\nHowever,thedynamic behaviourofbiochemical\\nsystems isincreasingly comingunderinvestiga-tion.Thisisdueinparttoadvancesinex-\\nperimentaltechniqueswhichmakeobservationof\\ntimeseriesmorereadilyattainable, andinpartto\\nanincreasing awarenessthatacomplete under-\\nstanding ofmanyofthemechanisms withinthe\\ncellwillonlybereachedoncetheirdynamic be-\\nhaviourhasbeendescribed.Asaresult,anum-\\nberofapproachestotheextension ofclassical\\n(steady-state) sensitivit yanalysishaveappeared.\\nGeneral de¯nitions oftime-varyingsensitivit y\\nfunctions were¯rstgiveninthiscontextin[1]\\n(seealso[10,12]).Thisextension ofthestan-\\ndardanalysis allowstreatmen tofsystemswhose\\nbehaviourisprimarily dynamic. Whilethesere-\\nsultshaveproventobeuseful,thereisoneformof\\ndynamic behaviourforwhichtheycannotprovide\\nsatisfactory descriptions, namelyautonomously\\noscillating systems. Suchsystemunderlymany\\noftheperiodicphenomena whichhavebeeniden-\\nti¯edinbiology(e.g.glycolytic oscillations, the\\ncellcycle,circadian rhythms,periodicneuronal\\nsignals). See[7]forareview.\\nIthaslongbeenrecognized ([2]andreferences\\ntherein)thatthesensitivit yofvariousfeaturesof\\noscillating systems (e.g.periodandamplitude)\\ncouldbeestimated experimentally(orthrough\\nsimulation), eveniftherewasnogeneraltheory\\ntoprovideameansofanalytic computation. A\\n¯rststeptowardssuchatheorywaspresented\\nin[13],inwhichasatisfactory treatmen tofsys-\\ntemsexhibitingforcedoscillations isgiven.This\\ntheorywascomplemen tedbytheresultsin[4]\\nwhichaddresssensitivit yofFouriercoe±cients.\\nAutonomously oscillating systems weretreated\\nbythisapproachin[16].Theseworksprovidevi-\\ntalcontributions tothestudyofperiodicsystems.\\nHowever,therehasyettoappearageneraltreat-\\nmentofsensitivit yofautonomously oscillating\\np.1', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Sensitivity_analysis_of_autonomous_oscillations_ap.pdf', 'page': 0}),\n",
       " Document(page_content='biochemical systems. Thispapercomplemen ts\\ntheexisting workbyprovidingsuchanexposi-\\ntion.\\nSensitivit yanalysis isacrucialaspectofthe\\nstudyofcontrolsystems, andhasbeenexten-\\nsivelytreatedwithintheengineering community.\\nIn[3],Bur¶eandRosenvasserpresentamethod\\nforderiving sensitivit yoffeatures oflimitcy-\\ncletrajectories (namely periodandamplitude)\\ntochangesinparameters. Thesameresultswere\\nderivedindependentlyin[5,14].Asthecurrent\\npaperwilldemonstrate, thismethodcanberead-\\nilyappliedtobiochemicalsystemsandtheresult-\\ninganalytic description canbeusedtoprovide\\nadditional insightinthissetting.\\n2Preliminaries\\nTheanalysis willtreatageneralnetworkofn\\nchemical species(s=(s1;s2;:::;sn))involved\\ninmreactions ina¯xedvolume.Thesystem\\nwillbemodelledasdependingonasinglescalar\\nparameter p.Thesystemisdescribedbythen\\nbymstoichiometry matrixNandthem-vector\\nvaluedreaction ratefunctionv=v(s;p).The\\ndynamics aregivenby\\nd\\ndts(t)=Nv(s(t);p)forallt¸0:(1)\\nTheparameter pmightrepresentanyofthepa-\\nrameters ofthemodel(e.g.aVmaxorKmvalue)or\\nanyparticular direction intheparameter space.\\nThefunctionvisassumed continuouslydi®eren-\\ntiable.\\nStoichiometric systems maycontainredundan t\\nstates(e.g.duetoconservedmoieties), whichcor-\\nrespondtolineardependencies amongtherows\\nofN.Insuchcases,thenumberofstatescanbe\\nreducedforthepurposesofanalysis andcompu-\\ntation,asdiscussed in[15].Inordertosimplify\\nthepresentationinwhatfollows,wewillassume\\nthatnosuchreduction iscalledfor.\\nInwhatfollowswewillassumethatforeach\\nvalueoftheparameter pintherangeofinter-\\nest,thesystem(1)exhibitsaperiodictrajectory\\nsper(t;p),withperiodT(p),i.e.\\nsper(t+T(p);p)=sper(t;p)8t¸0:(2)\\nMoreover,itwillbeassumed thateachsuchtra-\\njectoryisastablelimitcycle.3SensitivityAnalysis\\nAsdescribedin[1]and[12],thegeneral(time-\\nvarying)concentrationresponsecoe±cient (or\\nconcentrationsensitivit yfunction) isde¯nedas\\nfollows.\\nDe¯nition 3.1Givenanominal parameter\\nvaluep0andacorrespondingperiodictrajectory\\nsper(¢;p0),de¯netheresponsecoe±cient asthe\\nn-vectorvaluedfunctionRs(¢)givenby\\nRs(t):=@sper(t;p)\\n@pjp=p08t¸0:(3)\\nTheevolutionofthissensitivit yfunction isde-\\nscribedbythe¯rstorderlineardi®erentialequa-\\ntion\\nd\\ndtRs(t)=N@v(t)\\n@sRs(t)+N@v(t)\\n@p8t¸0:(4)\\nThepartialderivatives@v(t)\\n@sand@v(t)\\n@pareevalu-\\natedalongthetrajectorysper(t;p0).\\nTheresponseofsystem(1)toperturbations in\\nparameters whicha®ectonlytheinitialcondi-\\ntionsisstraightforward,aswasdiscussed in[12].\\nHenceforth wewillassumethatachangeinthe\\nparameter phasnoe®ectattimet=0,andso\\nwewilltakeinitialconditionRs(0)=0inequa-\\ntion(4).\\nInstandard sensitivit yanalysis, theresponseco-\\ne±cientsareuseddirectlytodetermine thesys-\\ntembehaviourunderperturbations. However,as\\ndiscussed in[13],thesensitivit yfunctions forau-\\ntonomously oscillating systemsgrowunbounded\\nintimeandhenceareoflittledirectuseinad-\\ndressing asymptotic behaviour.Nonetheless, as\\ndescribedin[3],thesesensitivit yfunctions canbe\\nusetodescribetheasymptotic responseofcertain\\npropertiesoftheoscillations, asoutlined below.\\nBeforeaddressing sensitivit y,wemakearemark\\nregarding equation (4).Observethatwheneval-\\nuatedalongtheperiodictrajectorysper(t;p)both\\nN@v(t)\\n@sandN@v(t)\\n@pareT(p)-periodic.Thehomo-\\ngeneouspartofequation (4),whosefundamen tal\\nmatrixH(¢)isthesolution of\\nd\\ndtH(t)=N@v(t)\\n@sH(t)H(0)=I;(5)\\ncanthusbeaddressed bythetoolsofFloquet\\nTheory(see,e.g.[8]).Theresultisasolution of\\ntheform\\nH(t)=D(t)eBt(6)\\np.2', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Sensitivity_analysis_of_autonomous_oscillations_ap.pdf', 'page': 1}),\n",
       " Document(page_content='wherethematrixD(¢)isT(p)-periodic.The\\neigenvaluesofthematrixBarecalledthechar-\\nacteristicexponentsofthesystem. Underour\\nassumption thatthesystem(1)admitstheT(p)-\\nperiodicsolutionsper(¢;p),itfollowsthatthedif-\\nferentialequation in(5)likewiseadmitsaT(p)-\\nperiodicsolution. Comparing with(6),onecon-\\ncludesthatthematrixBmusthaveatleastone\\neigenvaluewithmodulusone.\\nWenowmaketheassumption thatalloftheother\\ncharacteristic exponentsareintheunitcircle,\\nwhichisconsisten twiththefactthattheperiodic\\ntrajectorysper(¢;p)isalimitcycle.Underthis\\nassumption, thefundamen talmatrixH(¢)canbe\\nexpressed as\\nH(t)=Hper(t)+Htrans(t)8t¸0;\\nwherelimt!1Htrans(t)=0andHper(¢)isT(p)-\\nperiodic.\\nAsmentionedearlier,theunboundednatureof\\nthesensitivit yfunction meansthatitcannotbe\\nuseddirectlytodescribetheasymptotic system\\nresponse.However,thevariation inthesensitiv-\\nityfunction overtheperiodT(p0)isasolutionof\\nequation (5),andhenceisbetterbehaved.This\\nvariationprovesuseful,asshownbelow.\\n3.1SensitivityofOscillatory Characteris-\\ntics\\nInaddressing thesensitivit yofautonomously os-\\ncillating systems, itistheasymptotic response\\nwhichisofprimary interest. Inthecaseof\\nasymptotically stablesystems, theresponsecoef-\\n¯cientsasde¯nedabovetendtothesteady-state\\nsensitivit ycoe±cientsastimegrowlarge.Forau-\\ntonomously oscillating systems, thisinterpreta-\\ntioncannotbeexploited sincetheresponsecoe±-\\ncientsdiverge.However,theirvaluesdoconverge\\ntoadescription oftheasymptotic behaviourof\\ntheextremepointsontheperiodictrajectory,as\\nshownbelow.\\nSensitivityofLocalExtrema\\nTheoscillatory behaviourofsystemsisoftende-\\nscribedintermsoflocalextrema, e.g.maximal\\norminimal levelsofchemical species.Theex-\\ntremelevelsof°uxthrough apathwayarealso\\nofinterestinaddressing periodicsystems.\\nHavingidenti¯edsuchextrema asofinterest,one\\ncanthenaskhowtheymaybea®ectedbyapa-\\nrameter variation. Asatisfactory answercanbe\\ngivenprovidedonecharacterizes thepointbythe\\ntimeatwhichitoccurs.Thatis,having¯xed\\natimet0ofinterestontheperiodictrajectorysper(¢;p0),onecandescribetheresponseofthe\\nsystemattimet0toachangeinp,provided\\nthistimerepresentsalocalmaximumormini-\\nmumvalueforthevariableofinterest,aswenow\\ndemonstrate.\\nThefollowingisastraightforwardextension of\\nresultsin[17].Theproofisomitted duetolack\\nofspace.\\nLetY:Rn£R!Rbesomefunction ofthestate\\nandtheparameter, andidentifyatimet0(p0)2\\n[0;T(p0))atwhichthefunction Y(sper(t;p0);p0)\\nachievesalocalextremum.Then,forintegers m,\\nd\\ndpY(sper(t0(p);p);p)¯¯¯¯\\np=p0= (7)\\nlim\\nm!1@Y\\n@sRs(t0(p0)+mT(p0))+@Y\\n@p;(8)\\nwherethepartialderivativesofYareevaluated\\nat(sper(t0(p0);p0);p0).\\nInaddressing biochemical networks,themost\\ncommon choicesforthefunction Yarespe-\\nci¯cspeciesconcentrations orreaction rates.\\nThissecondchoiceleadstosensitivit yfunctions\\nknownas°uxresponsecoe±cients,denoted by\\nthevectorRv(¢).UsingRs¤andRv¤todenote\\ntheasymptotic responseofthespeciesconcen-\\ntrationsandreaction rates,respectively,(7)pro-\\nvidesthefollowing.Foranyspeciessiorreaction\\nratevjwhichachievesanextreme valueatt0,\\nRsi¤(t0)=lim\\nm!1Rsi(t0(p0)+mT(p0))\\nRvj¤(t0)=lim\\nm!1@vj\\n@sR(t0(p0)+mT(p0))\\n+@vj\\n@p\\nwherethepartialderivativesareevaluated at\\n(sper(t0(p0);p0);p0).\\nSensitivityofthePeriod\\nInaddition tomaximal andminimal values,the\\nasymptotic responseoftheoscillation canalsobe\\ncharacterized bythechangeintheperiod.An\\nauxiliary de¯nition willbeneeded.\\nDe¯nition 3.2GivenasolutionRs(¢)of(4),\\nde¯nethevariationofthesensitivity overthepe-\\nriodofthenominal oscillation by\\n¢Rs(t)=Rs(t+T(p0))¡Rs(t);8t¸0:\\nAsdescribedin[3],assuming T(¢)isdi®eren-\\ntiable,thesensitivit yoftheperiodcanexpressed\\nasfollows:foranyi=1;2;:::;n,let¢Rsi(t)\\np.3', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Sensitivity_analysis_of_autonomous_oscillations_ap.pdf', 'page': 2}),\n",
       " Document(page_content='denotethei-thelementof¢Rs(t),letNibe\\nthei-throwofN,andchooseanyunbounded\\nincreasing sequence oftimesftkg1\\nk=1suchthat\\nNiv(sper(tk;p0);p0)6=0forallk.Then\\ndT(p)\\ndp¯¯¯¯\\np=p0=lim\\nk!1¡¢Rsi(tk)\\nNiv(sper(tk;p0);p0):(9)\\n3.2ARemarkonComputation\\nInprevious treatmen ts,thesensitivit ycoe±-\\ncientsderivedabovehaveonlybeentreatedan-\\nalytically inspecialcases(e.g.[1]).Morecom-\\nmonly,theyhavebeenapproximated through\\nsimulation,e.g.bymakinga1%changeinapa-\\nrameter andobserving theresulting behaviour\\n(e.g.[2]).Suchinvestigations canprovideaccu-\\nrateestimates providedtheperturbation issu±-\\ncientlysmall,butsu®erfromnumerical di±cul-\\nties,sincethesensitivit ycoe±cientisderivedas\\naratioofsmallnumbers(leading toerrorsin-\\ntroducedbylimitsinprecision). Thederivation\\npresentedheredoesnotsu®erfromthisnumeri-\\ncalhandicap.\\n4SensitivityInvariants\\nWhensystemsaresubjectedtosensitivit yanaly-\\nsis,theirformoftenimposescertainrestrictions,\\nknownassensitivityinvariants ,ontheresulting\\nsensitivit yfunctions [17].WithintheMCAcom-\\nmunity,thestandard description ofsensitivit yin-\\nvariantsofbiochemicalsystemsaretheSumma-\\ntionandConnectivit yTheorems. Intheclassical\\ncaseofsteady-state analysis, theseTheorems de-\\nscribealgebraic constrain tsonthesensitivit yco-\\ne±cients[15].Anequivalentstatemen tofthese\\nresultscanbegivenasadescription ofthere-\\nsponseofthesystemtoperturbations inparame-\\ntersofaparticular form.Inthecurrentcontext,\\ngeneralizing theseresultsleadstothefollowing\\nstatemen tsregarding theasymptotic responseco-\\ne±cients.\\nSummation Theorem: Iftheparameter pis\\nchosensothat@v(t)\\n@pliesinthenullspaceofNfor\\neachtimetduringtheoscillationsper(t;p),then\\nRs¤(t)=0\\nRv¤(t)=@v(t)\\n@p;\\nforeacht2[0;T(p)).Moreover,dT\\ndp=0.\\nConnectivit yTheorem: Iftheparameter pis\\nchosensothatthereissomen-vectormforwhich\\nFigure1:Abstracted glycolytic chain\\n@v(t)\\n@p=@v(t)\\n@smforeachtimetduringtheoscil-\\nlationsper(t;p),then\\nRs¤(t)=(Hper(t)¡I)m\\nRv¤(t)=@v(t)\\n@sHper(t)m;\\nforeacht2[0;T(p)).Moreover,dT\\ndp=0.\\nNotethatinthesespecialcasesthesensitivit y\\natarbitrary times tisderived,notjustatthe\\nextremal points.Theproofsareomitted dueto\\nlackofspace.\\n5Application: asimplemodelof\\nglycolytic oscillations\\nThesimpletwo-speciesmodelshowninFigure1\\nhasbeendevelopedasanabstraction ofthegly-\\ncolyticpathway[9].Thekineticsaredescribedby\\nmass-action, withtheactivationbyS2appearing\\nmultiplicativ ely(i.e.therateofproduction of\\nS2isgivenbyk1S1(1+Sq\\n2)).Providedthepa-\\nrameters lieinanappropriate range,thesystem\\nexhibits limitcyclebehaviour,asshownin[9].\\nWithnominalparameter valuesofv0=8,k1=1,\\nk2=5,k3=1,andq=3,weconsider thee®ect\\nofperturbations ink3,whichisameasure ofthe\\nexternal consumption ofS1.Simulations leadto\\nthebehaviourshowninFigure2,wherewesee\\ntheoscillatory behaviorofS1,itssensitivit yfunc-\\ntionRs1,andthecorrespondingvariation¢Rs1.\\nFurtheranalysisprovidesthesensitivit yvaluesin\\nthetablebelow.Theonly°uxsensitivit ywhichis\\nreportedisthatofthereaction producing S2(de-\\nnoted v3){theothernonconstan treaction rates\\narelinearfunctions ofthespeciesconcentrations.\\np.4', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Sensitivity_analysis_of_autonomous_oscillations_ap.pdf', 'page': 3}),\n",
       " Document(page_content='0 2 4 6 8 10 12−20−15−10−50510152025\\nS1\\nR1\\nΔ R1\\nFigure2:S1,thesensitivit yfunctionRs1,andits\\nvariation¢Rs1\\nw.r.t. k3sensitivit yrelativesensitivit y\\nS1max -1.3 -0.44\\nS1min 5.7 4.8\\nS2max -4.4 -2.4\\nS2min 0.61 0.83\\nv3max -33 -3.3\\nv3min 4.7 1.3\\nThesensitivit yoftheperiodwithrespectto\\nk3wasfoundtobe:¡1:2(relativesensitivit y\\n¡0:56).Acomplete analysis woulddetermine\\nthesensitivities withrespecttotheremaining pa-\\nrameters inthemodel.\\nReferences\\n[1]Acerenza, L.,Sauro,H.M.andKacser,H.,\\n\\\\Controlanalysis oftime-dep endentmetabolic\\nsystems,\"J.Theor.Biol.,151(1989),pp.423{\\n444.\\n[2]Bier,M.,Teusink,B.,Kholodenko,B.N.\\nandWesterho®, H.V.,\\\\Controlanalysis ofgly-\\ncolyticoscillations,\"Biophys.Chem.,62(1996),\\npp.15{24.\\n[3]Bur¶e,¶E.G.andRozenvasser,E.N.,\\\\The\\nstudyofthesensitivit yofoscillatory systems,\"\\nAutomat.RemoteContr.,7(1974),pp.1045{\\n1052.\\n[4]Demin, O.V.,Westerho®, H.V.and\\nKholodenko,B.N.,\\\\Controlanalysis ofstation-\\nary¯xedoscillations,\"J.Phys.Chem.B,103\\n(1999),pp.10695{10710.[5]Edelson, D.andThomas, V.M.,\\\\Sensi-\\ntivityanalysis ofOscillating Reactions. 1.The\\nperiodoftheOregonator,\"J.Phys.Chem.,85\\n(1981),pp.1555-1558.\\n[6]Fell,D.A.,Understanding theControlof\\nMetabolism,PortlandPress,London, 1997.\\n[7]Goldbeter,A.,BiochemicalOscillations\\nandCellularRhythms,CambridgeUniversity\\nPress,Cambridge,1996.\\n[8]Hartman, P.,OrdinaryDi®erentialEqua-\\ntions,JohnWiley&Sons,NewYork,1973.\\n[9]Heinrich,R.,Rapoport,S.M.and\\nRapoport,T.A.,\\\\Metabolicregulation and\\nmathematical models,\"Prog.Biophys.&Mol.\\nBiol.,32(1977),pp.1{82.\\n[10]Heinrich,R.andReder,C.,\\\\Metabolic\\ncontrolanalysis ofrelaxation processes,\"J.\\nTheor.Biol.,151(1991),pp.343{350.\\n[11]Heinrich,R.andSchuster,S.,TheRegula-\\ntionofCellularSystems,Chapman &Hall,New\\nYork,1996.\\n[12]Ingalls,B.P.andSauro,H.M.,\\\\Sensitiv-\\nityanalysis ofstoichiometric systems: anexten-\\nsionofmetaboliccontrolanalysis tonon-steady\\nstatetrajectories,\"J.Theor.Biol.,222(2003),\\npp.23{36.\\n[13]Kholodenko,B.N.,Demin, O.V.and\\nWesterho®, H.V.,\\\\Controlanalysis ofperi-\\nodicphenomena inbiological systems,\"J.Phys.\\nChem.B,101(1997),pp.2070{2081.\\n[14]Kramer, M.A.,Rabitz, H.andCalo,\\nJ.M.,\\\\Sensitivit yanalysis ofoscillatory sys-\\ntems,\"Appl.Math.Modelling,8(1984),pp.328{\\n340.\\n[15]Reder,C.,\\\\Metaboliccontroltheory: a\\nstructural approach,\"J.Theor.Biol.,135\\n(1988),pp.175{201.\\n[16]Reijenga, K.A.,Westerho®, H.V.,Kholo-\\ndenko,B.N.andSnoep,J.L.,\\\\Controlanaly-\\nsisforautonomously oscillating biochemicalnet-\\nworks,\"Biophys.J.,82(2002),pp.99{108.\\n[17]Rosenwasser,E.andYusupov,R.,Sensi-\\ntivityofAutomaticControlSystems,CRCPress,\\nBocaRaton,Florida, 2000.\\n[18]Savageau,M.A.,BiochemicalSystems\\nAnalysis.AStudyofFunctionandDesignin\\nMolecularBiology,Addison-W esley,Reading,\\nMA,1976.\\np.5', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Sensitivity_analysis_of_autonomous_oscillations_ap.pdf', 'page': 4}),\n",
       " Document(page_content='Kimchi et al., Sci. Adv. 2020; 6 : eabc1939     16 December 2020\\nSCIENCE ADVANCES | RESEARCH ARTICLE1 of 8BIOPHYSICS\\nSelf-assembly–based posttranslational \\nprotein oscillators\\nOfer Kimchi1*, Carl P. Goodrich1, Alexis Courbet2,3,4, Agnese I. Curatolo1, Nicholas B. Woodall2,3,4, \\nDavid Baker2,3,4, Michael P. Brenner1,5\\nRecent advances in synthetic posttranslational protein circuits are substantially impacting the landscape of cellular \\nengineering and offer several advantages compared to traditional gene circuits. However, engineering dynamic \\nphenomena such as oscillations in protein-level circuits remains an outstanding challenge. Few examples of bio-\\nlogical posttranslational oscillators are known, necessitating theoretical progress to determine realizable oscillators. \\nWe construct mathematical models for two posttranslational oscillators, using few components that interact only \\nthrough reversible binding and phosphorylation/dephosphorylation reactions. Our designed oscillators rely on \\nthe self-assembly of two protein species into multimeric functional enzymes that respectively inhibit and enhance \\nthis self-assembly. We limit our analysis to within experimental constraints, finding (i) significant portions of the \\nrestricted parameter space yielding oscillations and (ii) that oscillation periods can be tuned by several orders of \\nmagnitude using recent advances in computational protein design. Our work paves the way for the rational design \\nand realization of protein-based dynamic systems.\\nINTRODUCTION\\nProtein oscillators play a major regulatory role in organisms ranging \\nfrom prokaryotes to humans. In most known biological cases, the \\noscillation is realized through transcription/translation cycles. Few \\nexamples of purely posttranslational oscillators have been found in \\nbiology (1 ,\\xa02). At the same time, posttranslational protein circuits \\nare increasingly sought after for synthetic applications, since they \\nhave the potential to exhibit faster response to environment changes, \\nallow for more direct control over the circuit behavior, be directly \\ncoupled to a functional output, and can be used in contexts that do \\nnot include the vast genetic apparatus (3 –5). While significant recent \\nwork has enabled the design of synthetic posttranslational protein–\\nbased logic gates (4 ,\\xa05), engineering tunable dynamic phenomena \\nsuch as oscillations in a synthetic posttranslational context remains \\nan outstanding challenge (6,\\xa07).\\nThe best-studied example of biological posttranslational protein \\noscillators is the KaiABC system in cyanobacteria (8 ). By placing \\nonly the proteins KaiA, KaiB, and KaiC in a test tube, along with \\nabundant adenosine triphosphate, the KaiC proteins collectively get \\nsequentially phosphorylated and dephosphorylated, forming an \\noscillatory cycle (9,\\xa0 10). While the KaiC proteins generally exist in a \\nhexameric state, monomers are shuffled among the hexamers during \\nonly a certain phase of the oscillatory cycle (11). The KaiABC system \\ndemonstrates that protein oscillators need not use transcription/\\ntranslation cycles or large numbers of components to achieve oscil-\\nlatory behavior.\\nMotivated by the KaiABC system, we set out to design a protein- \\nbased oscillator that could be reconstituted in\\xa0vitro using only a \\nsmall number of components at relatively high copy numbers, so \\nthat any resulting oscillations are not stochastic. To facilitate the future translation of this theoretical study to an experimental system, \\nwe base the architecture of our system on biochemical constraints \\nand on a design space navigable through computational protein \\ndesign. We constrain the kinetic reaction network to only include \\nthree protein species and to only allow reversible binding and \\nphosphorylation/dephosphorylation enzyme reactions.\\nAs in KaiABC, the oscillating system will cycle through periods \\nof a protein species being phosphorylated or not. When the target', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 0}),\n",
       " Document(page_content='phosphorylation/dephosphorylation enzyme reactions.\\nAs in KaiABC, the oscillating system will cycle through periods \\nof a protein species being phosphorylated or not. When the target \\nprotein is phosphorylated, the phosphorylation must induce a change \\nto propel the global state along the oscillatory cycle. This change can \\nbe achieved by affecting the enzyme kinetics of the kinase and phos-\\nphatase in one of two ways: either by altering the conformation of \\nthe target protein or by directly modifying the kinase and phospha-\\ntase enzymes. We first consider systems that do not modify the kinase \\nor phosphatase.\\nThe simplest such system, a protein with one phosphorylation \\nsite being modified by a kinase and a phosphatase, cannot yield \\noscillations regardless of parameter choices (7 ). When two phos-\\nphorylation sites are included, oscillations are possible only under \\nthe assumption that each of the four possible phosphorylation states \\nhas significantly different rates of subsequent phosphorylations and \\ndephosphorylations (7 ). While biology seems to have designed a \\nsystem in KaiABC capable of undergoing the many conformational \\nchanges necessary to implement this form of oscillations (9 ), the \\ndesign of even two (let\\xa0alone several) protein structures from the \\nsame sequence remains a significant challenge for the field of com-\\nputational protein design (12).\\nThese challenges are not unique to molecules with two phos-\\nphorylation sites. For example, since oscillations for molecules with \\ntwo phosphorylation sites are effected by enzyme sequestration (7), \\nwe consider a molecule containing a single phosphorylation site \\nalongside a kinase- or phosphatase-sequestering domain (or a binding \\ndomain for an external compound that itself contains an enzyme- \\nbinding domain). These systems are capable of producing oscillations—\\nbut only if phosphorylation and binding accompany a significant \\nconformational change in the molecule that modifies the rate constants \\nof subsequent reactions. Even assuming that such a conformational 1Harvard University School of Engineering and Applied Sciences, Cambridge, MA \\n02138, USA. 2Department of Biochemistry, University of Washington, Seattle, WA \\n98105, USA. 3Institute for Protein Design, University of Washington, Seattle, WA \\n98105, USA. 4Howard Hughes Medical Institute, University of Washington, Seattle, \\nWA 98105, USA. 5Kavli Institute for Bionano Science and Technology Harvard \\nUniversity, Cambridge, MA 02138, USA.\\n*Corresponding author. Email: okimchi@g.harvard.eduCopyright © 2020 \\nThe Authors, some \\nrights reserved; \\nexclusive licensee \\nAmerican Association \\nfor the Advancement \\nof Science. No claim to \\noriginal U.S. Government \\nWorks. Distributed \\nunder a Creative \\nCommons Attribution \\nLicense 4.0 (CC BY).\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 0}),\n",
       " Document(page_content='Kimchi et al., Sci. Adv. 2020; 6 : eabc1939     16 December 2020\\nSCIENCE ADVANCES | RESEARCH ARTICLE2 of 8change were designed, we have found no evidence of sustained oscil-\\nlations in these systems within the parameter regimes of typical binding/\\nunbinding rate constants and typical kinase and phosphatase activity \\n(i.e., the catalytic rate and Michaelis constants k cat and K M, discussed \\nfurther below). See section S1 for further discussion.\\nSystems that focus on modifications to the enzymes themselves \\nare therefore more likely candidates for the production of experi-\\nmentally realizable oscillations. Biology has found several ways to tie \\nphosphorylation to enzymatic activity. The most straightforward \\nconceptually, having the activity of an enzyme dependent on its \\nown phosphorylation state (13) remains a challenge to implement \\nin the context of computational protein design (14). However, the \\nfield has achieved remarkable success in the design of protein-protein \\ninteractions (15) which can be modified by phosphorylation (16,\\xa0 17).\\nBootstrapping off of this success, we consider proteins that self- \\nassemble into multimeric functional kinases and phosphatases (18,\\xa0 19). \\nWe are motivated, in part, by the success of using split proteases to \\nimplement posttranslational protein–based logic gates (4,\\xa0 5). In our \\ndesign, when the proteins’ binding interfaces are phosphorylated, \\ntheir self-assembly is impeded, reducing the concentration of func-\\ntional enzymes available in the system.\\nOscillations are thus achieved by the push and pull of two op-\\nposing factors: Self-assembled kinases inhibit the self-assembly of \\nnew proteins by phosphorylating both kinase and phosphatase \\nmonomers; meanwhile, self-assembled phosphatases counteract this \\ninhibition. Incoherent inputs such as these are known to enhance \\nthe robustness of oscillations (20). Our overall oscillator design is \\nmotivated by analogy to known successful oscillators, particularly \\nthe dual-feedback genetic oscillator (1,\\xa0 21–23). Just as that oscillator \\nrelies on the interplay between the inhibiting effects of LacI and the \\nactivating effects of AraC, our oscillations rely on the interplay be-\\ntween kinase and phosphatase multimers, which respectively inhibit \\nand activate their own self-assembly (Fig.\\xa01A).\\nThe rest of our manuscript is organized as follows. First, we describe \\nthe oscillatory circuits and their experimental constraints. Next, we de-\\nvelop simplified mathematical models for two distinct protein-based \\noscillators: In one, multimers are designed to form closed, bounded structures; in the other, they form unbounded fibers. We show that sim-\\nple analytical formulae describing the first oscillator can predict both the \\nregions of parameter space admitting oscillations and the oscillations’ \\nresulting frequencies. We then demonstrate that the second oscillator \\ndesign is complementary to the first in that it can admit robust and ex-\\nperimentally realizable oscillations in a regime of parameter space \\nwhere the first cannot. Finally, we discuss the significance of our findings.\\nRESULTS\\nSelf-assembly–based protein oscillators are designed within \\nexperimental constraints\\nDesigned synthetic oscillators rely on few protein species with \\nspecified interactions\\nThe main components of our oscillators are two proteins, which we \\ncall \\uf06b and \\uf072 . Each individual protein of type \\uf06b  (\\uf072) has two comple-\\nmentary parts of a split kinase (phosphatase) and a phosphorylation \\nsite. When the respective sites are dephosphorylated, copies of protein \\n\\uf06b (\\uf072) can self-assemble into a functional kinase (phosphatase), which \\nwe call K (P). Thus, self-assembled kinases inhibit the self-assembly \\nof new proteins, while self-assembled phosphatases counteract the \\ninhibition. In addition to the proteins \\uf06b and \\uf072, we include a consti-\\ntutive phosphatase    ~ P   ; without it, a fixed point where all proteins are \\nphosphorylated can preclude oscillations (see section S2).', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 1}),\n",
       " Document(page_content='tutive phosphatase    ~ P   ; without it, a fixed point where all proteins are \\nphosphorylated can preclude oscillations (see section S2).\\nThe resulting circuit topology (Fig.\\xa01A) is analogous to that used in \\nthe dual-feedback genetic oscillator (1,\\xa0 23). The multimeric kinase \\nplays an analogous role to the LacI protein in the genetic oscillator, \\nrepressing the production of new multimers; the multimeric phospha-\\ntase plays an analogous role to that of the AraC protein, activating the \\nproduction (or more precisely, counteracting the kinase inhibition).\\nTwo related networks based on these proteins can be designed. In \\nthe first, self-assembly is into closed symmetric homomultimers of \\nspecified size; in the second, the monomers are designed such that \\nthey self-assemble into one-dimensional unbounded fibers.\\nExperimental realizability constrains parameter sets\\nBecause we are motivated by experimental feasibility, we consider \\nonly physically realizable parameters for our models. Binding rates \\nFig. 1. Bounded self-assembly oscillator. ( A) Oscillator topology. By phophorylating monomers, kinase multimers (red; top) inhibit their own and phosphatase multimer \\n(blue; bottom) self-assembly. Similarly, phophatase multimers counteract this inhibition, as do constitutive phosphatases (center). (B) Bounded self-assembly reactions. \\nMonomers contain two halves of a split enzyme: either kinase (red; top) or phosphatase (blue; bottom). Monomers can self-assemble into multimers of specified size \\n(here, tetramers are pictured, corresponding to n = m = 4). Kinase (phosphatase) multimers can (de)phosphorylate the monomers. A constitutive phosphatase is also able \\nto dephosphorylate the monomers (not pictured). Phosphorylated monomers cannot participate in the self-assembly. Reactions are shown in pictorial form above each \\ncorresponding chemical equation. The full set of differential equations corresponding to these reactions is given in eq. S1.\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 1}),\n",
       " Document(page_content='Kimchi et al., Sci. Adv. 2020; 6 : eabc1939     16 December 2020\\nSCIENCE ADVANCES | RESEARCH ARTICLE3 of 8kb are typically in the range 10−2 to 100 \\uf06dM−1s−1 (24) with dissocia-\\ntion constants k d typically in the 10−3 to 103 \\uf06dM range (25). Both of \\nthese quantities can be tuned on the basis of the geometry, energy, \\nand symmetry of the binding interface between the proteins, which \\nwe assume here to be designed de novo. Less straightforward to de-\\nsign are the Michaelis constants and catalytic rates of the kinase and \\nphosphatase, especially since these depend not only on the enzyme \\nbut on the substrate. Mutational screens can be used to adjust these \\nparameters, but predicting the effect of a mutation on k cat or K M is \\nhighly nontrivial (26). We were unable to find studies measuring \\nkinase and phosphatase rates on the same substrate. Instead, as a \\nstandard to demonstrate physical realizability, we consider the pa-\\nrameters for sample Ser/Thr enzymes: wild-type \\uf06c-phosphatase act-\\ning on para-nitrophenylphosphate (k cat = 2.0 × 103 s−1; KM = 1.0 × \\n104 \\uf06dM) and wild-type MST4 (kinase) acting on the short peptide \\nchain NKGYNTLRRKK (k cat = 3.1 s−1; KM = 14 \\uf06dM) (26,\\xa0 27). We \\nassume throughout that the constitutive phosphatase    ~ P    is governed \\nby the same enzymatic rate constants as the self-assembled P. We also \\ntreat the self-assembled enzyme as only one functional protein be-\\ncause the copies of the enzyme are all colocalized.\\nBounded self-assembly can yield oscillations whose \\nbehavior is well-predicted by analytical formulae\\nThe onset of oscillations for the bounded self-assembly system \\nis well-predicted by two dimensionless parameters\\nWe first consider a system where unphosphorylated kinase and \\nphosphatase monomers self-assemble in an all-or-nothing manner \\ninto closed symmetric homomultimers. We denote by n the num-ber of kinase monomers \\uf06b in the functional kinase multimer K and \\nby m the analogous number of phosphatase monomers \\uf072  in the \\nmultimer P. The reaction network is shown in Fig.\\xa01B.\\nSince the full equations describing this bounded self-assembly \\nsystem (eq. S1) are too complex to directly tackle analytically, we \\nnumerically integrate them within the parameter ranges outlined \\nabove (section S5). Our results, shown in Fig.\\xa02, demonstrate a sig-\\nnificant portion of parameter space within experimental constraints \\ncapable of admitting sustained oscillations.\\nTo simplify these equations to an analytically tractable form, \\nwe make the Michaelis-Menten approximation that enzymatic \\nintermediates are in quasi-steady state. We then make the approxi-\\nmation that the enzyme and substrate concentrations are low com-\\npared to the Michaelis constants, such that the concentration of \\nenzymatic intermediates can be entirely neglected within our analytical \\napproximations (section S2). This approximation, like others that \\nwe will consider, is not obeyed by all oscillating solutions found \\nnumerically (Fig.\\xa02) but is nonetheless useful in clarifying the funda-\\nmentals of a large swath of the oscillations. We find that, in contrast \\nto well-known examples from other systems that rely on enzyme \\nsequestration to achieve oscillations (6 ,\\xa07), neglecting enzyme se-\\nquestration does not preclude oscillations for our systems.\\nTo reduce our systems further to only two differential equations, \\nwe assume a separation of timescales between the self-assembly and \\nthe enzymatic activity. In particular, we assume that phosphorylation/\\ndephosphorylation reactions equilibrate much faster than self- assembly. \\nThe opposite separation–of–timescales limit yields oscillations only \\nfor extremely large values of m , which are infeasible to realize \\nFig. 2. Bounded self-assembly can yield oscillations using experimentally realizable parameters. Numerical integration of eq. S1 displays parameter regimes lead-', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 2}),\n",
       " Document(page_content='Fig. 2. Bounded self-assembly can yield oscillations using experimentally realizable parameters. Numerical integration of eq. S1 displays parameter regimes lead-\\ning to oscillations within experimental constraints. Each subplot shows the location of oscillating parameter sets as a function of k d\\uf06b and k d\\uf072 for given k b\\uf06b and k b\\uf072; the \\nlatter two are varied for each subplot. Aside from experimental constraints (see main text for discussion), we set n = m = 2, \\uf06b tot = \\uf072 tot = 10 \\uf06dM, and    \\u202f˜\\u202fP\\u202f  \\u202ftot   =  10  \\u202f−4   \\uf06dM. Blue \\npoints denote parameter sets leading to sustained oscillations; yellow points denote parameter sets leading to steady state. To the right of the plot, we show a few exam-\\nple trajectories in K-P phase space. The closed trajectories correspond to sustained oscillations; the final trajectory, the spiral, corresponds to a decaying oscillation and \\ntherefore to a yellow point in the figure.\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 2}),\n",
       " Document(page_content='Kimchi et al., Sci. Adv. 2020; 6 : eabc1939     16 December 2020\\nSCIENCE ADVANCES | RESEARCH ARTICLE4 of 8experimentally (see further discussion in section S4). We thereby \\narrive at the following two-dimensional system of equations\\n     dK ─ dt   =  k  b\\uf06b     (      \\uf06b  tot   − nK ─  1 +  \\uf068  \\uf06b     K _ P +   ~ P      )     n\\n  −  k  u\\uf06b   K\\n    \\n  dP ─ dt   =  k  b\\uf072     (      \\uf072  tot   − mP ─  1 +  \\uf068  \\uf072     K _ P +   ~ P      )     m\\n  −  k  u\\uf072   P    (1)\\nWe briefly define the parameters: k b\\uf06b is the binding rate for \\uf06b \\ninto its multimeric state, k u\\uf06b is the respective unbinding rate, and \\nkd\\uf06b is the inverse ratio of the two; \\uf068 K\\uf06b is the specificity constant \\nkcat/KM for the kinase K acting on \\uf06b, \\uf068P\\uf06b is the same for the phospha-\\ntase P, and \\uf068 \\uf06b = \\uf068 K\\uf06b/\\uf068P\\uf06b; \\uf06btot is the total concentration of monomeric \\n\\uf06b added to the system, a conserved quantity. Similar quantities are \\ndefined for \\uf072 . We assume the concentration of the constitutive \\nphosphatase    ~ P   > 0  throughout (see section S2).\\nTo describe the oscillatory behavior of the system, we seek the \\neigenvalues of the Jacobian in the vicinity of a fixed point (K⋆, P⋆). \\nOscillations require coupling between the equations, motivating the \\napproximations that in the oscillatory regime,   \\uf068  \\uf06b    K   ⋆  ≫  P   ⋆  +   ~ P    (and \\nsame for \\uf068 \\uf072), \\uf06btot ≫ nK⋆, and \\uf072 tot ≫ mP⋆. Defining the dimension-\\nless concentrations    ˜  K   ⋆   =  K   ⋆  /   ~ P    and    ˜  P   ⋆   =  P   ⋆  /   ~ P   , the fixed point in \\nthese limits is given by\\n      ˜  P   ⋆     (n+1)  = γ  (  ˜  P   ⋆   + 1)   m    \\n  ˜  K   ⋆   = α    ˜  P   ⋆     (n/m)       (2)\\nwhere \\uf067 and \\uf061 are dimensionless parameters defined by\\n   \\uf067\\n  =    k d\\uf06b  m   ─  k d\\uf072  n+1      (      \\uf068  \\uf06b   ─  \\uf06b  tot     )     nm\\n    (      \\uf072  tot   ─  \\uf068  \\uf072     )     (n+1)m\\n     ~ P     (m−n−1) \\n    \\n\\uf061  \\n=    k d\\uf072  n/m  ─  k  d\\uf06b       (      \\uf068  \\uf072    \\uf06b  tot   ─  \\uf068  \\uf06b    \\uf072  tot     )     n\\n     ~ P     (n−m)/m     (3)\\nWe constrain ourselves to m ≤ n + 1 so that, within our approx-\\nimations, there is no more than one physical fixed point in the system \\nas long as    ~ P   >  0 , simplifying our analysis. (When no solutions t o \\nEq. 2 exist, our assumptions leading to it break down.)\\nSustained oscillations in the system typically correspond to com-\\nplex eigenvalues of the Jacobian with positive real parts. However, \\nfollowing the Poincaré-Bendixson theorem, as long as our system \\nhas a single fixed point, instability of the fixed point must imply \\noscillations even if they are beyond the linear regime. Translated into \\nconstraints on P⋆, instability of the fixed point corresponds to\\n    (     (m − 1 )  k  u\\uf072   − (n + 1 )  k  u\\uf06b    ──────────────   (n + 1 )  k  u\\uf06b   +  k  u\\uf072     )      P   ⋆  ─   ~ P     > 1   (4)\\nWe directly verify Eq. 4 in fig. S6.\\nWe now proceed to express Eq. 4 only in terms of the input pa -\\nrameters. Because Eq. 2 cannot be solved for P⋆ for general n, m, \\nwe consider the approximation that   P   ⋆  =  \\uf067   1/(n+1−m)   ~ P   ≫   ~ P   , equiv-\\nalent within the constraint m < n + 1 to \\uf067 ≫ 1. This approximation \\nis most accurate for small values of m, since fewer terms are neglect-ed. The approximation is motivated by the intuition that oscilla-\\ntions require P to be non-negligible compared to    ~ P   ; indeed, an op-\\nposite self-consistent solution, in which   P   ⋆  ≪   ~ P   , is incompatible \\nwith oscillations.\\nBy simplifying Eq. 4 within this limit where   P   ⋆  =  \\uf067   1/(n+1−m)   ~ P   , we \\nfind that the system oscillates when\\n  \\uf06e ≡   (m − 1 )  k  u\\uf072   ─  (n + 1 )  k  u\\uf06b     > 1  (5)\\nThis corresponds to ensuring a positive left-hand side in Eq. 4 (see \\nfig. S6).\\n The implication of Eq. 5, that oscillations require a larger disso-\\nciation rate of the phosphatase monomers compared to the kinase \\nmonomers (at least, for n + 2 ≥ m), agrees with intuition found by \\nvisualizing individual oscillation cycles (Fig.\\xa03A). In K-P space, os-\\ncillations proceed in a counterclockwise fashion: Starting from the', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 3}),\n",
       " Document(page_content='visualizing individual oscillation cycles (Fig.\\xa03A). In K-P space, os-\\ncillations proceed in a counterclockwise fashion: Starting from the \\nunphosphorylated state, the monomers self-assemble into multimers \\n(top right). The larger dissociation rate of phosphatase multimers \\nleads those to dissociate first and get phosphorylated by the abundant \\nkinase multimers (bottom right). Next, the kinase multimers slowly \\ndissociate, enabling the gradual dephosphorylation and self-assembly \\nof the phosphatase monomers by the constitutive and self-assembled \\nphosphatases (bottom left). Once kinase multimer levels have de-\\ncreased and enough phosphatase multimers have formed, the latter \\nquickly dephosphorylate the remaining monomers (top left), and the \\nsystem returns to its initial state (top right).\\nWe verify that the approximate formula given by Eq. 5 is valid in \\ndescribing Eq. 1 by comparing it to oscillations found by random \\nparameter searches in Fig.\\xa03B. We numerically integrate Eq. 1 with \\nrandom parameters chosen to satisfy the experimental constraints \\ndescribed previously (including setting \\uf068 \\uf06b = \\uf068 \\uf072) and with n = m = 2. \\nWe constrain concentrations \\uf06b tot and \\uf072 tot to be within 10−3 to 102\\xa0\\uf06dM, \\nwhile we set the bounds of    ~ P    to 10−8 and 10−2\\xa0\\uf06dM. For each parameter \\nset, we numerically estimate the fixed point using Python’s scipy.\\noptimize.root function. We only show parameter sets estimated to \\nagree with the approximations described before Eq. 2 (with >5× \\nsubstituted for ≫). We found no oscillations in ∼2.5 × 104 parameter \\nsets for which \\uf068 K or \\uf068 P is less than  ( P   ⋆  +   ~ P   ) /  K   ⋆  . Each blue (yellow) \\npoint in the figure corresponds to a single parameter set found to \\nproduce (not produce) oscillations starting from initial conditions \\nof (K , P) = (0,0). Oscillations are almost exclusively found in the \\nquadrant \\uf067 > 1, \\uf06e  > 1. Values of \\uf067  slightly less than unity are also \\nfound to produce oscillations, as shown in the figure.\\nThese results show that oscillations are found when the dissoci-\\nation rate of the phosphatase multimer is in a middle range between \\ntwo extremes. To see oscillations, the phosphatase multimer must \\ndissociate significantly faster than the kinase multimer (\\uf06e > 1); how-\\never, the dissociation rate must concurrently be small enough such \\nthat the fixed point concentration of phosphatase multimer is larger \\nthan the constitutive phosphatase concentration (\\uf067 > 1). In contrast \\nto intuition from other systems, which signifies that higher-order \\nnonlinearities increase the parameter range producing oscillations \\n(6), here, we found that more nonlinear self-assembly (i.e., higher \\nvalues of n  and m ) makes oscillations less frequent. We find that \\noscillations are robust to even order-of-magnitude variations in most \\nother parameters (section S3 and fig. S2).\\nOur results suggest that oscillations in the concentrations of un-\\nphosphorylated monomers or of phosphatase multimers may be \\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 3}),\n",
       " Document(page_content='Kimchi et al., Sci. Adv. 2020; 6 : eabc1939     16 December 2020\\nSCIENCE ADVANCES | RESEARCH ARTICLE5 of 8most straightforward to visualize experimentally, as these concentra-\\ntions typically vary by several orders of magnitude across an oscillatory \\ncycle (fig. S8). Parameter sets yielding largest oscillation amplitudes \\nare also typically farthest from the bifurcation point (fig. S10).\\nThe frequency of resulting oscillations can be well-predicted by \\nassuming that kinase multimer dissociation is rate limiting\\nWe next seek to predict how system parameters tune the frequency \\nof resulting oscillations when they appear. Within the linear regime \\naround the fixed point, in the limit of Eq. 5, the frequency of oscil-\\nlations \\uf077 is predicted to be\\n   \\uf077 pred  2   = −   1 ─ 4    [(n + 1 )  k  u\\uf06b   + (m − 1 )  k  u\\uf072  ]   2  + nm  k  u\\uf06b    k  u\\uf072    (6)\\nWhile Eq. 6 agrees well with the true squared frequency for those \\nparameter sets where it is positive, oscillations are frequently found \\nin the nonlinear regime in which it is not applicable (fig. S7A). \\nHowever, the intuition given by Eq. 6, that oscillation frequency is \\ndetermined by the unbinding rates of the kinase and phosphatase \\nmultimers, may still be valid outside the linear regime of the fixed \\npoint. This intuition is reasonable given the separation–of–timescales \\nlimit in which we are operating, of enzymatic reactions equilibrating \\nmuch faster than self-assembly. Since oscillations for the n = m = 2 system that we considered numerically require that ku\\uf072 > ku\\uf06b (i.e., \\uf06e  > 1), \\nEq. 6 implies that the limiting reaction in the oscillations is the un-\\nbinding of the kinase multimers. To test that implication, we com-\\npare ku\\uf06b to the frequency of oscillations found through numerical \\nintegration in Fig.\\xa03C. (We make no constraints on the fixed points \\nof the parameter sets considered here.) We find a strong correlation \\n(R2 = 0.66; R2 = 0.93\\xa0in log space) and a root mean square relative \\nerror of ∼3.9, demonstrating that k u\\uf06b can accurately predict the fre-\\nquency of oscillations in this system.\\nOscillations found within experimental constraints for Eq. 1 have \\nperiods ranging from fractions of a second to >1 day (fig. S7B). For \\noscillations found for the full system of equations plotted in Fig.\\xa02, \\nwe find periods within a slightly more constrained range than for \\nthe simplified system but still spanning orders of magnitude, between \\n~1\\xa0min and >1 day.\\nUnbounded self-assembly can yield oscillations within \\nexperimental constraints in the limit of fast self-assembly \\ncompared to enzymatic activity\\nWe now consider a second system in which individual species \\uf06b and \\n\\uf072 can self-assemble incrementally into one-dimensional unbounded \\nfibers (Fig.\\xa04A). Unlike the bounded case in which no phosphorylation \\nFig. 3. Analytical results for bounded self-assembly oscillator. In this figure, we show results from the analytically simplified bounded self-assembly oscillator (Eq. 1), \\nusing n = m = 2. (A) Oscillation schematic. We visualize a sample oscillation using randomly and arbitrarily chosen parameters satisfying experimental constraints. Oscil-\\nlations require phosphatase multimers (blue) to dissociate faster than kinase multimers (red). The system starts with self-assembled kinases and phosphatases (top right). \\nAfter rapid phosphatase disassembly and phosphorylation by the kinase multimers (bottom right), the kinases slowly disassemble, which enables the gradual dephos-\\nphorylation and self-assembly of the phosphatase monomers (bottom left). The assembled phosphatases are then able to rapidly promote their own and kinase self- \\nassembly through dephosphorylation, returning the system to its initial state (top right). (B) Onset of oscillations. Numerical integration demonstrates consistency with \\nEq. 5 for the appearance of oscillations in the appropriate limits. Each point represents a random set of parameters, sampled within the experimentally realizable limits as', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 4}),\n",
       " Document(page_content='Eq. 5 for the appearance of oscillations in the appropriate limits. Each point represents a random set of parameters, sampled within the experimentally realizable limits as \\ndescribed in the main text. Oscillating (blue) and non-oscillating (yellow) parameter sets can be well-separated by dimensionless combinations of parameters \\uf067 and \\uf06e. \\nDashed lines show where the dimensionless parameters on the axes equal unity. (C) Oscillation frequency. Intuition from linear stability analysis of the fixed point sug-\\ngests that for the n = m = 2 system considered numerically, oscillation frequency may be determined by the dissociation rate of kinase multimers,   k \\u202fu\\uf06b   . Numerical integra-\\ntion demonstrates that   k \\u202fu\\uf06b    is indeed highly predictive of oscillation frequency (R2 ≈ 0.66; R2 ≈ 0.93 in log space) and underestimates the true frequency by a typical factor \\nof ~4. Black dashed line shows \\uf077 = k u\\uf06b.\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 4}),\n",
       " Document(page_content='Kimchi et al., Sci. Adv. 2020; 6 : eabc1939     16 December 2020\\nSCIENCE ADVANCES | RESEARCH ARTICLE6 of 8sites are accessible in the multimeric state, in this system, one is \\n(corresponding to the final protein in the fiber). An n -mer of spe -\\ncies X (where X  is either \\uf06b  or \\uf072 ), Xn, can be created either from binding \\ntwo smaller molecules X m and X n−m or from the spontaneous break-\\ning of a bond of a larger molecule. The concentration of X n decreases \\nwhen an X n molecule either binds to any other molecule or breaks \\nany of its n − 1 bonds. The equations for self-assembly of species X \\nare therefore given by\\n      dX  n   ─ dt   =\\n   \\xa0 k  bX   (    \\uf053  m=1  n−1\\n    X  m    X  n−m   − 2  X  n     \\uf053  \\nm=1  ∞\\n    X  m   )   \\n    \\n   \\n +  k  uX   (  2  \\uf053  \\nm=n+1  ∞\\n    X  m   − (n − 1 )  X  n   )       (7)\\nAs in the first system, each protein of type \\uf06b (\\uf072) includes a split \\nkinase (phosphatase). While in the bounded system, enzymes re-\\nquired exactly n or m monomers to self-assemble, here, an enzyme \\nis created by any group of more than one monomer (i.e., X n is a \\nfunctional enzyme as long as n ≥ 2). We assume that when a multi-\\nmer is phosphorylated, its final monomer dissociates from the fiber \\nand cannot reassociate in its phosphorylated state. A less stringent \\nassumption, that the phosphorylated monomer does not dissociate \\nautomatically but merely prevents new monomers from binding to \\nthat end of the molecule, appears to be incompatible with oscilla-\\ntions, at least in both separation–of–timescales limits.The full equations describing the system are given in the Supple-\\nmentary Materials (eq. S8). As previously, we search for a two- \\ndimensional set of simplified equations by considering a separation \\nof timescales between enzymatic reactions and self-assembly, along \\nwith the same simplifying assumptions as considered for the bound-\\ned self-assembly system (eq. S9). The limit considered for the first \\nsystem, of fast phosphorylation/dephosphorylation compared to \\nself-assembly, would disallow multimers from forming in this system, \\nsince, here, phosphorylation is accompanied by dissociation of the final \\nmultimer in the chain. Therefore, the separation–of–timescales limit \\nconsidered in the bounded self-assembly system is no longer applicable \\nfor this system. Instead, we consider the opposite limit, of fast self- \\nassembly compared to enzymatic activity. At steady state, X n is given by\\n   X  n   =    k  uX   ─  k  bX       (     x ─   1 + x +  √ _ 1 + 2x     )     n\\n   (8)\\nwhere x = 2k bXXtot/kuX = 2X tot/kdX. The same steady state is reached \\neven if self-assembly involves binding and unbinding only a single \\nmonomer at a time.\\nThe concentrations of phosphorylated monomers as a function \\nof time are given by \\uf06bp and \\uf072p. The total amount of kinase present is \\ngiven by  K =  \\uf053 i=2  ∞    \\uf06b  i    and similarly for phosphatase. Since only the \\nphosphorylation site of the final monomer in a multimer is exposed, the \\ntotal number of available phosphorylation sites in the \\uf06b species is given \\nby   \\uf053 i=1  ∞    \\uf06b  i    (and similarly for \\uf072). The system can be described by two \\ndifferential equations for k = 2(\\uf06b tot − \\uf06bp)/kd\\uf06b and p = 2(\\uf072 tot − \\uf072p)/kd\\uf072\\n     dk ─ dt   =\\n   \\xa0  − 2  \\uf068  K\\uf06b   k ─  1 +  √ _ 1 + 2k     K +  \\uf068  P\\uf06b   (     2  \\uf06b  tot   ─  k  d\\uf06b     − k )  (P +   ~ P  ) \\n       dp ─ dt   =\\n   \\xa0  − 2  \\uf068  K\\uf072   p ─  1 +  √ _ 1 + 2p     K +  \\uf068  P\\uf072   (     2  \\uf072  tot   ─  k  d\\uf072     − p )  (P +   ~ P  ) \\n     \\nK =  \\n\\xa0 k  d\\uf06b      k   2   ──────────────────    (1 +  √ _ 1 + 2k   ) (1 + k +  √ _ 1 + 2k  )       \\nP =  \\n\\xa0 k  d\\uf072      p   2   ──────────────────    (1 +  √ _ 1 + 2p   ) (1 + p +  √ _ 1 + 2p  )      (9)\\nUnlike in the previous system for which the frequency and onset \\nof oscillations can be determined by simple formulae by taking a \\nlimit of the two-dimensional system, no such limits give similarly \\nstraightforward results for Eq. 9. Instead, we analyze Eq. 9 through', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 5}),\n",
       " Document(page_content='limit of the two-dimensional system, no such limits give similarly \\nstraightforward results for Eq. 9. Instead, we analyze Eq. 9 through \\nrandom parameter searches (Fig.\\xa04B). As shown in the figure, we \\nfind that although these equations assume an opposite separation–\\nof–timescales limit to that yielding experimentally realizable oscilla -\\ntions for the system of bounded self-assembly, they can nevertheless \\nyield oscillations within a significant region of parameter space con-\\nsistent with experimental constraints.\\nWe find that increased values of \\uf06b tot/\\uf072tot and decreased values \\nof    ~ P   /  \\uf072  tot    lead to more robust oscillations in this system; see fig. S3 \\nfor further discussion of oscillation robustness. The periods of os-\\ncillations found ranged from <1\\xa0min to >1 day (fig. S7B). Finally, we \\nfind that the concentrations of phosphatase multimers and unphos-\\nphorylated phosphatase monomers typically vary by several orders \\nof magnitude across an oscillatory cycle (fig. S9), suggesting that \\noscillations may be most straightforward to visualize experimentally \\nby measuring these concentrations.\\nFig. 4. Unbounded self-assembly oscillator. ( A) Unbounded self-assembly reactions. \\nWe consider a related system to that shown in Fig. 1 but relying on kinase and \\nphosphatase monomers that self-assemble into unbounded fibers of arbitrary length. \\nIn addition, we assume that the final monomer of each fiber can get phosphorylated \\nby a kinase multimer, at which point it can no longer rejoin the fiber until it is \\ndephosphorylated. (B) Unbounded self-assembly oscillations using experimentally \\nrealizable parameters. Numerical integration of Eq. 9 displays parameter regimes \\nleading to oscillations within experimental constraints. Eq. 9 was used in place of \\nthe full system of equations (eq. S8) because of the infinite dimensionality of the \\nlatter. \\uf072 tot sets the concentration scale.\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 5}),\n",
       " Document(page_content='Kimchi et al., Sci. Adv. 2020; 6 : eabc1939     16 December 2020\\nSCIENCE ADVANCES | RESEARCH ARTICLE7 of 8DISCUSSION\\nIn summary, we have presented two posttranslational protein–based \\noscillators motivated by the biological KaiABC system and by the \\nsynthetic dual-feedback genetic oscillator. Both systems that we \\npresent rely on split kinase and phosphatase self-assembling to form \\nfunctional enzymes and on that self-assembly being inhibited by \\nphosphorylation of the split monomers. The two systems differ mainly \\nin the nature of the self-assembly as all-or-nothing into bounded \\nstructures of specified size or incremental into unbounded one-  \\ndimensional fibers.\\nBoth systems are capable of producing oscillations within exper-\\nimental constraints, using experimentally determined wild-type \\nvalues for kinase and phosphatase activity and for a range of de-\\nsigned self- assembly rates. We have shown that neither complex \\nreactions nor large number of species are necessary to achieve oscil-\\nlations: Both networks that we present use only three protein \\nspecies interacting only through reversible binding and phospho-\\nrylation/dephosphorylation reactions, and the resulting oscillations \\ncan be understood as arising from a minimal system of two differ-\\nential equations in both cases.\\nAlthough the systems that we described shared much in com-\\nmon, they produced robust oscillations in opposite separation–of–\\ntimescales limits from one another: The first primarily oscillates when \\nself-assembly is much slower than enzymatic reactions; the second \\nwhen it is much faster. These two networks are thus complementary: \\nDepending on the parameter regime most easily accessible to an \\nexperimentalist, one or the other network might be preferable to \\nimplement. Nevertheless, the conditions giving rise to oscillations \\nshared similarities in the two systems. Smaller values of constitutive \\nphosphatase (equivalent to larger values of \\uf067 in the case of bounded \\nself-assembly) lead to more robust oscillations in both systems, as \\ndo smaller values of k d\\uf06b (i.e., larger values of \\uf06e).\\nOur work paves the way toward the rational design and experi-\\nmental realization of protein-based far-from-equilibrium dynamic \\nsystems. The models described here were designed to be feasible to \\nsynthesize experimentally and are guiding an implementation in \\nthe test tube that is currently under way.\\nSUPPLEMENTARY MATERIALS\\nSupplementary material for this article is available at http://advances.sciencemag.org/cgi/\\ncontent/full/6/51/eabc1939/DC1\\nView/request a protocol for this paper from Bio-protocol.\\nREFERENCES AND NOTES\\n 1. O. Purcell, N. J. Savery, C. S. Grierson, M. di Bernardo, A comparative analysis of synthetic \\ngenetic oscillators. J. R. Soc. Int. 7, 1503–1524 (2010).\\n 2. M. W. Young, S. A. Kay, Time zones: A comparative genetics of circadian clocks. Nat. Rev. \\nGenet. 2, 702–715 (2001).\\n 3. D. Bray, Protein molecules as computational elements in living cells. Nature 376, 307–312 \\n(1995).\\n 4. X. J. Gao, L. S. Chong, M. S. Kim, M. B. Elowitz, Programmable protein circuits in living \\ncells. Science 1258, 1252–1258 (2018).\\n 5. T. Fink, J. Lonzarić, A. Praznik, T. Plaper, E. Merljak, K. Leben, N. Jerala, T. Lebar, Ž. Strmšek, \\nF. Lapenta, M. Benčina, R. Jerala, Design of fast proteolysis-based signaling and logic \\ncircuits in mammalian cells. Nat. Chem. Biol. 15, 115–122 (2018).\\n 6. B. Novák, J. J. Tyson, Design principles of biochemical oscillators. Nat. Rev. Mol. Cell Biol. 9, \\n981–991 (2008).\\n 7. C. C. Jolley, K. L. Ode, H. R. Ueda, A design principle for a posttranslational biochemical \\noscillator. Cell Rep. 2, 938–950 (2012). 8. M. Nakajima, K. Imai, H. Ito, T. Nishiwaki, Y. Murayama, H. Iwasaki, T. Oyama, T. Kondo, \\nReconstitution of circadian oscillation of cyanobacterial KaiC phosphorylation in vitro. \\nScience 308, 414–415 (2005).\\n 9. J. Michael, J. S. Markson, W. S. Lane, D. S. Fisher, E. K. O’Shea, Ordered', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 6}),\n",
       " Document(page_content='Reconstitution of circadian oscillation of cyanobacterial KaiC phosphorylation in vitro. \\nScience 308, 414–415 (2005).\\n 9. J. Michael, J. S. Markson, W. S. Lane, D. S. Fisher, E. K. O’Shea, Ordered \\nphosphorylation governs oscillation of a three-protein circadian clock. Science  \\n318, 809–812 (2007).\\n 10. M. J. Rust, Orderly wheels of the cyanobacterial clock. Proc. Natl. Acad. Sci. U.S.A. 109, \\n16760–16761 (2012).\\n 11. H. Kageyama, T. Nishiwaki, M. Nakajima, H. Iwasaki, T. Oyama, T. Kondo, Cyanobacterial \\ncircadian pacemaker: Kai protein complex dynamics in the KaiC phosphorylation cycle \\nin vitro. Mol. Cell 23, 161–171 (2006).\\n 12. S. E. Boyken, M. A. Benhaim, F. Busch, M. Jia, M. J. Bick, H. Choi, J. C. Klima, \\nZ. Chen, C. Walkey, A. Mileant, A. Sahasrabuddhe, K. Y. Wei, E. A. Hodge, S. Byron, \\nA. Quijano-Rubio, B. Sankaran, N. P. King, J. Lippincott-Schwartz, V. H. Wysocki, K. K. Lee, \\nD. Baker, De novo design of tunable, pH-driven conformational changes. Science 364, \\n658–664 (2019).\\n 13. B. Kholodenko, Negative feedback and ultrasensitivity can bring about oscillations \\nin the mitogen-activated protein kinase cascades. Eur. J. Biochem. 267, 1583–1588 \\n(2000).\\n 14. M. Strumillo, P. Beltrao, Towards the computational design of protein posttranslational \\nregulation. Bioorg. Med. Chem. 23, 2877–2882 (2015).\\n 15. L. Doyle, J. Hallinan, J. Bolduc, F. Parmeggiani, D. Baker, B. L. Stoddard, P. Bradley, Rational \\ndesign of \\uf061-helical tandem repeat proteins with closed architectures. Nature 528, \\n585–588 (2015).\\n 16. H. Nishi, K. Hashimoto, A. R. Panchenko, Phosphorylation in protein-protein binding: \\nEffect on stability and function. Structure 19, 1807–1815 (2011).\\n 17. H. Nishi, A. Shaytan, A. R. Panchenko, Physicochemical mechanisms of protein regulation \\nby phosphorylation. Front. Genet. 5, 270 (2014).\\n 18. K. Camacho-Soto, J. Castillo-Montoya, B. Tye, I. Ghosh, Ligand-gated split-kinases. J. Am. \\nChem. Soc. 136, 3995–4002 (2014).\\n 19. K. Camacho-Soto, J. Castillo-Montoya, B. Tye, L. O. Ogunleye, I. Ghosh, Small molecule \\ngated split-tyrosine phosphatases and orthogonal split-tyrosine kinases. J. Am. Chem. \\nSoc. 136, 17078–17086 (2014).\\n 20. Z. Li, S. Liu, Q. Yang, Incoherent inputs enhance the robustness of biological oscillators. \\nCell Syst. 5, 72–81.e4 (2017).\\n 21. P. Smolen, Frequency selectivity, multistability, and oscillations emerge from models \\nof genetic regulatory systems. Am. J. Physiol. 274, C531–C542 (1998).\\n 22. J. Hasty, M. Dolnik, V. Rottschäfer, J. J. Collins, Synthetic gene network for entraining \\nand amplifying cellular oscillations. Phys. Rev. Lett. 88, 148101 (2002).\\n 23. J. Stricker, S. Cookson, M. R. Bennett, W. H. Mather, L. S. Tsimring, J. Hasty, A fast, robust \\nand tunable synthetic gene oscillator. Nature 456, 516–519 (2008).\\n 24. M. Schlosshauer, D. Baker, Realistic protein-protein association rates from a simple \\ndiffusional model neglecting long-range interactions, free energy barriers, and landscape \\nruggedness. Protein Sci. 13, 1660–1669 (2004).\\n 25. S. E. A. Ozbabacan, H. B. Engin, A. Gursoy, O. Keskin, Transient protein–protein \\ninteractions. Protein Eng. Des. Sel. 24, 635–648 (2011).\\n 26. S. Zhuo, J. C. Clemens, R. L. Stone, J. E. Dixon, Mutational analysis of a Ser/Thr \\nphosphatase. J. Biol. Chem. 269, 26234–26238 (1994).\\n 27. C. Chen, B. H. Ha, A. F. Thévenin, H. J. Lou, R. Zhang, K. Y. Yip, J. R. Peterson, M. Gerstein, \\nP. M. Kim, P. Filippakopoulos, S. Knapp, T. J. Boggon, B. E. Turk, Identification of a major \\ndeterminant for serine-threonine kinase phosphoacceptor specificity. Mol. Cell 53, \\n140–147 (2014).\\n 28. R. A. Langan, S. E. Boyken, A. H. Ng, J. A. Samson, G. Dods, A. M. Westbrook, \\nT. H. Nguyen, M. J. Lajoie, Z. Chen, S. Berger, V. K. Mulligan, J. E. Dueber, W. R. P. Novak, \\nH. El-Samad, D. Baker, De novo design of bioactive protein switches. Nature 572, \\n205–210 (2019).\\n 29. A. H. Ng, T. H. Nguyen, M. Gómez-Schiavon, G. Dods, R. A. Langan, S. E. Boyken,', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 6}),\n",
       " Document(page_content='H. El-Samad, D. Baker, De novo design of bioactive protein switches. Nature 572, \\n205–210 (2019).\\n 29. A. H. Ng, T. H. Nguyen, M. Gómez-Schiavon, G. Dods, R. A. Langan, S. E. Boyken, \\nJ. A. Samson, L. M. Waldburger, J. E. Dueber, D. Baker, H. El-Samad, Modular and tunable \\nbiological feedback control using a de novo protein switch. Nature 572, 265–269 (2019).\\nAcknowledgments: We thank A. Murugan for helpful discussions. Funding: This work was \\nsupported by the Harvard Materials Research Science and Engineering Center grant no. \\nDMR-1420570 and ONR grant no. N00014-17-1-3029 (to M.P.B.). O.K. acknowledges funding \\nfrom the DoD through NDSEG Fellowship 32 CFR 168a, the NSF-Simons Center for \\nMathematical and Statistical Analysis of Biology at Harvard University award no. 1764269, and the \\nHarvard Quantitative Biology Initiative. A.C. is a recipient of the Human Frontier Science \\nProgram Long-Term Fellowship and a Washington Research Foundation Senior Fellow. \\nM.P.B. is an investigator of the Simons Foundation. M.P.B. is a Research Scientist at Google \\nResearch. Author contributions: O.K., C.P.G., A.I.C., and M.P.B. designed and implemented the \\nnumerical analyses. O.K. and A.I.C. designed and performed the analytical analyses. A.C., \\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 6}),\n",
       " Document(page_content='Kimchi et al., Sci. Adv. 2020; 6 : eabc1939     16 December 2020\\nSCIENCE ADVANCES | RESEARCH ARTICLE8 of 8N.B.W., and D.B. developed the experimental feasibility. O.K. and A.C. implemented the figures. \\nAll authors developed the oscillator designs, designed the figures, and wrote the paper. \\nCompeting interests: The authors declare that they have no competing interests. Data and \\nmaterials availability: All data needed to evaluate the conclusions in the paper are present in \\nthe paper and/or the Supplementary Materials. The code used in this study is available at \\nhttps://github.com/ofer-kimchi/protein-oscillator. Additional data related to this paper may \\nbe requested from the authors.Submitted 10 April 2020\\nAccepted 3 November 2020\\nPublished 16 December 2020\\n10.1126/sciadv.abc1939\\nCitation: O. Kimchi, C. P. Goodrich, A. Courbet, A. I. Curatolo, N. B. Woodall, D. Baker, M. P. Brenner, \\nSelf-assembly–based posttranslational protein oscillators. Sci. Adv. 6, eabc1939 (2020).\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 7}),\n",
       " Document(page_content='Use of this article is subject to the Terms of service\\nScience Advances  (ISSN 2375-2548) is published by the American Association for the Advancement of Science. 1200 New York Avenue\\nNW, Washington, DC 20005. The title Science Advances  is a registered trademark of AAAS.\\nCopyright © 2020 The Authors, some rights reserved; exclusive licensee American Association for the Advancement of Science. No claim\\nto original U.S. Government Works. Distributed under a Creative Commons Attribution License 4.0 (CC BY).Self-assembly–based posttranslational protein oscillators\\nOfer KimchiCarl P. GoodrichAlexis CourbetAgnese I. CuratoloNicholas B. WoodallDavid BakerMichael P. Brenner\\nSci. Adv., 6 (51), eabc1939. • DOI: 10.1126/sciadv.abc1939\\nView the article online\\nhttps://www.science.org/doi/10.1126/sciadv.abc1939\\nPermissions\\nhttps://www.science.org/help/reprints-and-permissions\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/SelfAssemblyOscillations.pdf', 'page': 8}),\n",
       " Document(page_content=\"RESEA RCH ARTICL E\\nCytosolic proteins canexploit membrane\\nlocalization totrigger functional assembly\\nOsman N.Yogurtcu, Margaret E.Johnson*\\nDepartment ofBiophysics, Johns Hopkins Univers ity,Baltimor e,MD, United States ofAmerica\\n*Margare t.johnson@j hu.edu\\nAbstract\\nCell division, endocytosis, andviral budding would notfunction without thelocalization and\\nassembly ofprotein complexes onmembranes. What ispoorly appreciated, however, isthat\\nbylocalizing tomembranes, proteins search inareduced space thateffectively drives up\\nconcentration. Here wederive anaccurate andpractical analytical theory toquantify thesig-\\nnificance ofthisdimensionality reduction inregulating protein assembly onmembranes. We\\ndefine asimple metric, aneffective equilibrium constant, thatallows forquantitative compar-\\nison ofprotein-protein interactions with andwithout membrane present. Totesttheimpor-\\ntance ofmembrane localization fordriving protein assembly, wecollected theprotein-\\nprotein andprotein-lipid affinities, protein andlipid concentratio ns,andvolume-to-surface-\\narea ratios for46interactions between 37membrane-targeting proteins inhuman andyeast\\ncells. Wefindthatmany oftheprotein-prote ininteractions between pairs ofproteins in-\\nvolved inclathrin-mediated endocytosis inhuman andyeast cells canexperience enormous\\nincreases ineffective protein-protein affinity (10±1000 fold) duetomembrane localization.\\nLocalization ofbinding partners thus triggers robust protein complexation, suggesting thatit\\ncanplay animportant roleincontrolling thetiming ofendocytic protein coat formation. Our\\nanalysis shows thatseveral other proteins involved inmembrane remodeling atvarious\\norganelles have similar potential toexploit localization. The theory highlights themaster role\\nofphosphoinosit idelipid concentratio n,thevolume-to-sur face-area ratio, andtheratio of3D\\nto2Dequilibrium constants intriggering (orpreventing) constitutive assembly onmem-\\nbranes. Oursimple model provides anovel quantitative framework forinterpreting ordesign-\\ninginvitro experiments ofprotein complexation influenced bymembrane binding.\\nAuthor summary\\nInamultitude ofcellular processes, including celldivision and endocytosis, proteins must\\nbind tooneanother toform large multi-protein complexes. Toinitiate theformation of\\nthese critical multi-protein assemblies attheright time and theright place, theconstituent\\nproteins must bepresent atsufficient concentrations. Weshow here that membrane local-\\nization offers apowerful way ofcontrolling protein concentrations byreducing the\\ndimensionality oftheprotein's search space. Wepresent asimple and practical analytical\\ntheory that determines thesignificance ofmembrane localization fortriggering protein-\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 1/28a1111111111\\na1111111111\\na1111111111\\na1111111111\\na1111111111\\nOPEN ACCESS\\nCitation: Yogurtcu ON,Johnson ME(2018)\\nCytosolic proteins canexploit membran e\\nlocalization totrigger functiona lassembly. PLoS\\nComput Biol14(3): e1006031. https://do i.org/\\n10.1371/ journal.pcbi.10 06031\\nEditor: Peter M.Kasson, University ofVirginia,\\nUNITED STATES\\nReceived: October 12,2017\\nAccepted: February 9,2018\\nPublished: March 5,2018\\nCopyright: ©2018 Yogurtcu, Johnson. Thisisan\\nopen access article distributed under theterms of\\ntheCreative Commons Attributio nLicense, which\\npermits unrestricte duse,distribu tion,and\\nreproduction inanymedium, provided theoriginal\\nauthor andsource arecredited.\\nData Availabilit yStatement: Allrelevant dataare\\nwithin thepaper anditsSupport ingInformation\\nfiles.\\nFunding: Thisresearch wassupported byNational\\nInstitutes ofGeneral Medical Sciences ofthe\\nNational Institutes ofHealth Award R00GM0983 71\\ntoMEJ. Thefunders hadnoroleinthestudy\\ndesign, datacollection andanalysis, decision to\\npublish, orpreparation ofthemanuscript.\\nCompeting interests :Theauthors have declared\\nthatnocompeting interests exist.\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 0}),\n",
       " Document(page_content=\"protein interactions. Weshow that protein binding partners willoften form substantially\\nmore complexes when both partners canlocalize tosurfaces, and thus localization can\\nregulate thetiming ofmulti-protein assembly. Wecollectinvitro binding data and cellular\\nconcentrations ofproteins and lipids involved inpathways including clathrin-mediated\\nendocytosis todemonstrate how cellular proteins could exploit membrane localization to\\nregulate assembly.\\nIntroduction\\nWhen clathrin, theessential cytosolic protein ofclathrin-mediated endocytosis (CME), self-\\nassembles into multi-protein cages, thesame protein-protein contacts areused regardless of\\nwhether clathrin isinsolution oronthemembrane [1±3]. However, more binding [2]isob-\\nserved onthemembrane. Afundamental phenomenon forexplaining thischange isdimen-\\nsionality reduction: ifproteins onthemembrane search asmaller space, then thisincreases\\ntheir relative concentration; higher concentration ofproteins (reactants) shifts theequilibrium\\ntoproduce more protein-protein complexes (products) asdefined byLeChatelier's principle.\\nThe question weaddress is,how significant arole canthisdimensionality reduction play for\\ndriving protein-protein interactions between cytosolic proteinsinvitro? Understanding this\\nrole canhelp interpret mechanisms ofassembly invivo. Despite thewide-ranging cellular pro-\\ncesses such ascelldivision and viral budding that could exploit thisphenomenon, ithassofar\\nlacked apredictive theoretical framework. Hence while theconcept that membrane localiza-\\ntion canenhance binding may befamiliar orintuitive, wehere make that concept quantitative\\nforsoluble binding partners. Incontrast, theory forunderstanding reduction ofdimensionality\\ninchemoreception and receptor mediated signaling (where itcanalso befunctionally signifi-\\ncant [4]) hasbeen studied fordecades [5,6].Membrane localization canaccelerate aligand's\\nsearch formembrane bound targets [5±9] and increase activation ofintracellular receptors,\\ninfluencing downstream response [8±10]. However, inthese cases, asoluble protein always\\ntargets amembrane bound receptor. Here wecapture thedynamic cases where both binding\\npartners aresoluble and target lipids present inlimited concentrations, asoccurs, forexample,\\ninCME. Our theory determines how binding enhancement depends onprotein and lipid con-\\ncentrations, protein-protein and protein-lipid affinities, thevolume-to-surface area ratio, and\\nthechange inbinding affinities from 3Dto2D.Quantifying thisbehavior iscritical tounder-\\nstanding assembly onsurfaces because 2Dlocalization canstrengthen binding reactions\\nregardless ofwhether additional factors, such ascurvature generation [11], membrane micro-\\ndomains [12, 13], orconformational switches [1],also influence binding.\\nWeshow here that membrane localization offers apowerful way ofcontrolling protein con-\\ncentrations and therefore ofregulating thetiming ofmulti-protein assembly. Inmany cases,\\nwefind that thepower ofmembrane localization todrive binding ishighly robust; strong and\\nweak protein-protein interactors, athigh orlowconcentrations, willallbenefit significantly\\nfrom membrane localization. The analytical theory wepresent describes arelatively simple\\nmodel atequilibrium where apair ofsoluble binding partners canform complexes insolution\\nand also canboth bind and continue toform complexes onthesurface ofamembrane (Fig 1).\\nThus itisuseful asatool toquantify protein-protein interactions that, while physiologically\\nrelevant, arebeing studiedinvitro. Without accounting forthecomplex array offactors pres-\\nentinvivo, such asvariability inmembrane composition, competition forprotein and lipid\\nbinding from diverse proteins, spatial distributions ofproteins orlipids, and non-equilibrium\\ndynamics, wecanonly speculate about thebehavior inthecell. However, thetheory provides a\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 2/28\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 1}),\n",
       " Document(page_content=\"novel and valuable metric forinterpreting how important membrane localization canbegiven\\ntheconcentrations and binding properties ofcomponent proteins, and itisolates therole of\\nmembrane localization from other factors. Since even mostinvitro experiments contain more\\ncomponents and complexity than iscaptured inoursimple model, wediscuss how itcanstill\\nbeused asaquantitative guide forestimating how membrane heterogeneity, competition for\\nbinding, and mutations would influence theparameters ofthemodel (volume, surface area,\\nbinding affinities and concentrations) and thus theproteins' subsequent response tolocaliza-\\ntion. Wespecifically address inourresults how lipids such asPI(4,5)P 2canbetargeted by\\nmany proteins atanytime [14, 15], how some membrane binding domains such asBAR\\ndomains bind membranes with widely varying lipid composition and inacurvature dependent\\nmanner [16, 17], and how mutations and multiple protein binding partners would alter pro-\\ntein complex formation. Despite thelimitations ofapplying anequilibrium theory tounder-\\nstand complexation that ultimately occurs inthenonequilibrium cell, webelieve thetheory\\nrepresents awell-defined starting point from which toprobe more complex systems, justas\\nusinginvitro studies provide auseful guide forinterpreting behavior inthecell. Itisalso aref-\\nerence point forstudying thetime-dependence ofassembly through computer simulation, as\\nwedohere, and astarting point from which tobuild further complexity into themodel.\\nFig1.Quantifyin ghow protein binding partners insolution can increase complex formation bybinding tolipids onmembrane surfaces. This model contains two\\ntypes ofproteins (P1and P2)and onetype oflipid (M). Weshow alltenpossible bindin ginteractions that canoccur between thispair ofproteins that bind each other\\n(P1+P2\\x84P 1P2)and also bind specific lipids (M), producing asystem ofnine total distinct species: P1,P2,M,P1P2,MP 1,P2M,MP 1P2,P1P2M,and MP 1P2M.a)Solution\\n(3D) binding. b)Interactio nsinsolution (3D) that pull proteins tothemembran esurface through protein orlipid binding. In(c-e) thebinding interacti onsarein2D\\n(species concentrat ioninArea-1)and canexploit thelower search space. Conversio nfrom 3Dto2Dequilibriu mconstant isdefined bythevariableσ,where onlyσ=σPP\\nappears inEq3.Tosolve forallspecies inconsistent units (i.e.Volume-1),Ka2Dvalues must bemultiplie dbyV/A. The sizeofsolution volume Vvs.membrane surface\\narea Aisthus acritical parameter controlling binding enhanceme nt.The membran esurface canbetheplasma membran e,forexample, butalso liposome ssuspended in\\nsolution. d,e) Proteins localized atthesurface willalso bind lipids ina2Dsearch. There areover 100functionally diverse peripheral membrane proteins inyeast alone\\n[14] whose binding interactions with oneanother could strengthen substa ntially viabinding tomembranes. Wesimulated thismodel foracompreh ensive range of\\nconditions using mainly systems ofordinary different ialequations (ODE), butalso single-part iclereaction-di ffusion (RD) simulations [19, 20](Methods) .\\nhttps://do i.org/10.1371/j ournal.pc bi.1006031. g001\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 3/28\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 2}),\n",
       " Document(page_content='Weapply thetheory here tocharacterizing, within aquantitative framework, therole of\\nmembrane localization forenhancing 55binding interactions involving 33distinct protein\\npair interaction sets(S1Table). Through simulation, wealso move beyond themodel illus-\\ntrated inFig1ofonly pairs ofsoluble binding partners toshow how complexation involving\\nnon-membrane binding scaffold proteins such asclathrin, orhow formation ofhigher-order\\noligomers, which isfunctionally important fordriving membrane remodeling [11, 17,18], can\\nalso beregulated bymembrane localization (13additional interaction setsinS2Table). Our\\ntheory only applies tothepair interactions illustrated inFig1.Weinclude 22proteins involved\\ninCME inboth human and yeast cells, aswell as15proteins involved inlipid regulation, vesi-\\ncleformation onendosomes, budding, and morphogenesis inyeast cells (Table 1).Wecol-\\nlected concentration and cellular geometry data based oninvivo values tobetter connect to\\nphysiologic regimes (S3Table and S4Table). Although ourtheory represents anapproximate\\nsolution tothefullmodel shown inFig1,weshow through extensive simulations using both\\nsystems ofordinary differential equations and single-particle reaction-diffusion, [19, 20]that it\\nishighly accurate. Through simulation, weadditionally find that membrane localization alters\\nthetimescales ofprotein-protein assembly, butthat theresult isnotdominated bychanges in\\nprotein diffusion between solution and themembrane. Rather, forphysiologic binding\\nstrengths, therate-limiting step isthespeed ofbinding tothemembrane surface from solution.\\nFinally, apractical application ofoursimple formula isthat itcanbeused toexperimentally fit\\nprotein-protein binding affinities onsurfaces (Ka2D),which arerarely measured [21, 22]. The\\nadvantage oftheformula isthat itapplies toinvitro experiments where apair ofproteins can\\nreversibly bind tothemembrane, thus avoiding theneed torestrict proteins tothesurface.\\nResults\\nModel and theory\\nInourprimary model, weconsider twoproteins P1and P2,that bind insolution with equilib-\\nrium constantKPP;3D\\na\\x88\\x89P1P2\\x8aeq\\n\\x89P1\\x8aeq\\x89P2\\x8aeq\\x88\\x89P1P2\\x8aeq\\n\\x85\\x89P1\\x8a0\\x00\\x89P1P2\\x8aeq\\x86\\x85\\x89P2\\x8a0\\x00\\x89P1P2\\x8aeq\\x86,where total concentrations ofthe\\nproteins are[P1]0=[P1P2]+[P1]and same for[P2]0.Ifthese proteins canalso reversibly bind\\ntomembranes viatargeting aspecific lipidM,and continue tobind oneanother, their binding\\nequilibrium willshift asatotal ofnine distinct species canform (Fig 1,Methods). The bound\\nprotein-protein complexes caneither beinsolution oronthemembrane, [P1P2]sol+mem=\\n[P1P2]sol+[P1P2]mem=[P1P2]+[P 1P2M]+[MP 1P2][MP 1P2M],and unbound species aresimi-\\nlarly defined [P1]sol+[P1]mem=[P1]+[MP 1],and [P2]sol+[P2]mem=[P2]+[P 2M],whereMindi-\\ncates acopy ofatarget membrane lipid bound toP1orP2.The model thus assumes each\\nprotein binds membrane viatargeting asingle copy ofaspecific lipid type. Proteins onthe\\nmembrane must beable todiffuse tobind oneanother, which isconsistent with experimental\\nobservations [23] even ofRNA-protein complexes (>6000kDa) that areanchored viamultiple\\nlipid binding sites along with myristoyl groups [24]. Each ofthenine distinct species willbe\\nconstrained topreserve detailed balance atequilibrium, asdefined bythe10pairwise binding\\ninteractions ofFig1(see Methods), and thetotal concentrations ofproteins isfixed atthe\\nsame values asabove, butnow [P1]0=[P1P2]sol+mem+[P 1]sol+mem,and thesame for[P2]0.Simi-\\nlarly, [M] 0=[M]+[MP 1]+[P 2M]+[P 1P2M]+[MP 1P2]+2[MP 1P2M].Wenote that species onthe\\nmembrane have concentrations normally ofμm-2,matching theunits ofequilibrium constants\\nin2D(Ka2D)-1.Allspecies copy numbers, whether onoroffthemembrane, however, canbe\\nsolved forinvolume units when theappropriate Solution volume/Membrane surface Area (V/\\nA)conversion factor scales the2Dbinding constants, sowealways report volume units for\\nconcentrations. Toquantify thechange inbound protein-protein complexes asafunction of\\nMembrane localizatio ncantrigger protein assembly', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 3}),\n",
       " Document(page_content='concentrations. Toquantify thechange inbound protein-protein complexes asafunction of\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 4/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 3}),\n",
       " Document(page_content='membrane localization wewilldefine aneffective equilibrium constant\\nKeff\\na\\x88\\x85\\x89P1P2\\x8asol\\neq\\x87\\x89P1P2\\x8amem\\neq\\x86\\n\\x85\\x89P1\\x8asol\\neq\\x87\\x89P1\\x8amem\\neq\\x86\\x85\\x89P2\\x8asol\\neq\\x87\\x89P2\\x8amem\\neq\\x86\\x88\\x85\\x89P1P2\\x8asol\\x87mem\\neq\\x86\\n\\x85\\x89P1\\x8a0\\x00\\x89P1P2\\x8asol\\x87mem\\neq\\x86\\x85\\x89P2\\x8a0\\x00\\x89P1P2\\x8asol\\x87mem\\neq\\x86:\\x851\\x86\\nTable 1.Proteins studied, copy numbers, protein-lipi dinteraction s(PLI) and affinities (KdPM).\\nProtein Species Copy\\nNumberdKdPM\\n(μM)Literature Refs\\n1 OSH2 Yeast 850 1±1.5 PLI: PMID:11238 399. Affinity, measured: same ref.\\n2 SWH1 Yeast 505 3.5±6. 2 PLI: PMID:21119 626. Affinity, measured: same ref.\\n3 KES1 Yeast 21166 0.055±2. 2 PLI: PMID:22162 133,1191698 3.Affinity, measured: same.\\n4 VPS17 Yeast 1077 >100 PLI: PMID:11557 775. Affinity, measured: same ref.\\n5 SNX4 Yeast 1483 >100 PLI: PMID:11557 775. Affinity, measured: same ref.\\n6 SNX41 Yeast 367 >100 PLI: PMID:11557 775. Affinity, measured: same ref.\\n7 VPS5 Yeast 1326 >100 PLI: PMID:11557 775. Affinity, measured: same ref.\\n8 ATG20 Yeast 519 >100 PLI: PMID:11557 775. Affinity, measured: same ref.\\n9 BOI2 Yeast 567 6.6±19.5 PLI: PMID:15023 338. Affinity, measured: same ref.\\n10 CLA4 Yeast 397 20.2±100 PLI: PMID:15023 338. Affinity, measured: same ref.\\n11 SKM1 Yeast 16 3.9±6. 4 PLI: PMID:15023 338. Affinity, measured: same ref.\\n12 BEM1 Yeast 1037 >100 PLI: PMID:11557 775. Affinity, measured: same ref.\\n13 BOI1 Yeast 399 20 PLI: PMID:15023 338. Affinity, measured: same ref.\\n14 VAM7 Yeast 210 2±3 PLI: PMID:11557 775. Affinity, measured: same ref.\\n15 SNX3 Yeast 5092 2±3 PLI: PMID:11557 775. Affinity, measured: same ref.\\n16 SLA2 Yeast 3904 0.27±3.4 PLI: PMID:15574 875. Affinity, homology (AP180): PMID: 12740367.\\n17 SYP1 Yeast 2467 Used 0.1,10,100 PLI: PMID:19713 939,1321812 .Affinity, notknown, estimat ed.\\n18 ENT1 Yeast 1750 0.08 PLI: PMID:22193 158,1044940 4.Affinity, homology (EPN1) :PMID:17825 837.\\n19 ENT2 Yeast 1325 0.08 PLI: PMID:22193 158,1044940 4.Affinity, homology (EPN1) :PMID:17825 837.\\n20 YAP1802 Yeast 264 0.27±3.4 PLI: PMID:22193 158,2111962 6.Affinity, homology (AP180) :PMID:1274 0367.\\nCHC1 (Not Studied) Yeast 19278aNobinding -\\n21 SLA1 Yeast 2964 Nobinding -\\n22 EDE1 Yeast 5964 Nobinding -\\n23 FCHO1 Human 3706 Used 0.1,10,100 PLI: PMID:22484 487. Affinity, notknown, estimat ed.\\n24 AP-2 Human 244537b2.86 (0.072) PLI: PMID:15916 959. Affinity, measured: same ref.\\n25 EPN1 Human 570949 0.08 PLI: PMID:17825 837. Affinity, measured: same ref.\\n26 PICALM Human 358673 2.7±3. 4 PLI: PMID:25090 048. Affinity, homology (AP180): PMID:12 740367.\\n27 DAB2 Human 1078162 0.08 PLI: PMID:12234 931. Affinity, homology (EPN1): PMID:17 825837.\\n28 FCHO2 Human 36302 Used 0.1,10,100 PLI: PMID:20448 150. Affinity, notknown, estimat ed.\\n29 SNAP91/AP 180 Human 21716c0.27±3.4 PLI: PMID:12740 367. Affinity, measured: same ref.\\n30 LDLRAP1/A RH Human 1048 0.08 PLI: PMID:12234 931. Affinity, homology (EPN1): PMID:17 825837.\\n31 HIP1 Human 13771 0.27±3.4 PLI: PMID:14732 715. Affinity, homology (AP180): PMID:12 740367.\\n32 HIP1R Human 24161 0.27±3.4 PLI: PMID:14732 715. Affinity, homology (AP180): PMID:12 740367.\\n33 AMPH Human 89536 0.1 PLI: PMID:22888 025. Affinity, measured: same ref.\\n34 SH3GL2/En dophilin Human 55621 0.03 PLI: PMID:22888 025. Affinity, measured: same ref.\\n35 EPS15 Human 91354 Nobinding -\\n36 ITSN1 Human 20184 Nobinding -\\n37 CLTC Human 1495814aNobinding -\\na)Tosimulate clathrin trimers, theheavy chain copy number sreported here aredivided by3.\\nb)Consideri ngAP2A1 gene.\\nc)Based onppm value from PMID: 24920484. Wescaled thisvalue bythenumber ofAP-2s from PMID: 26496610 toobtain thepredicted number ofAP180s inthecell.\\nd)Copy number foryeast from Ref. [49] and humans Ref: [50]\\nhttps://do i.org/10.1371/j ournal.pc bi.1006031. t001\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 5/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 4}),\n",
       " Document(page_content='This isnotatrue equilibrium constant, asboth bound and unbound states asdefined above\\ncontain several species that donotallstepwise interconvert with oneanother. However, from\\nKaeffand initial protein concentrations [P1]0and [P2]0,onecanimmediately solve forbound\\ncomplex concentration using Eq1.Ifproteins cannot bind tothemembrane, thevalue ofKaeff\\nwillrevert tothesolution bound value, KaPP,and thus theratio ofKaeff/KaPPdetermines the\\nextent towhich membrane localization either enhances ordiminishes protein-protein com-\\nplex formation. Aswediscuss further below, intheextreme limits where allproteins areeither\\ninsolution oronthemembrane, Kaeffreduces toatrue equilibrium constant. The strength of\\nKaeffisthat italso quantitatively describes alltheconditions inbetween these limits. Thus, our\\nKaeffdefinition offers avaluable metric forquantifying theequilibrium ofthemodel inFig1,\\nwhich must otherwise bedefined bymultiple quantities.\\nWederive below anexact expression forKaeffbased onthe10individual equilibrium rela-\\ntions foreach reversible binding reaction (Fig 1,Methods). The value ofKaeffforanyprotein\\npair willdepend onvolume V,surface area A,total protein [P1]0,[P2]0,and lipid concentra-\\ntions [M] 0,and allsixtrue equilibrium constants between protein and lipid interactions in3D\\n\\x85KPP\\na;KP1M\\na;KP2M\\na\\x86and in2D\\x85K2D;PP\\na;K2D;P 1M\\na;K2D;P 2M\\na\\x86.Importantly, theKa2Dvalues (with\\nunitsμm2/mol, e.g.) aredifferent from thecorresponding 3Dvalues, butthey arerelated\\nthrough Ka2D=Ka3D/(2σ). The variableσ,with units oflength, isathermodynamic property\\nofeach binding pair that captures changes tobinding free energy asaresult ofsurface restric-\\ntion and changes instandard state units. Changes tofree energy arelargely entropic, dueto\\naltered rotational freedom and protein flexibility [22], although limitations ontheorientation\\nofthebinding interfaces could alter theenthalpy. The bending rigidity ofthemembrane can\\nalso affectσ,bycontrolling therelative orientations that thebinding pairs canadopt [25]. The\\nvariableσthus represents anindependent variable that isspecific toeach protein pair studied.\\nItispossible that even ifconcentrations increase onthemembrane surface, adecrease inKa2D\\nwillcause lesscomplex formation, and wequantify thisregime intheResults section. Tokeep\\ntrack ofthese distinct 3Dand 2Dequilibrium constants, weexplicitly retain the2Dsuperscript\\nfor2Dbinding, otherwise Ka(including Kaeff)describes a3Dconstant. Toderive asimple ana-\\nlytical expression forKaeff,weinput thepairwise equilibria (Methods) into Eq1and after can-\\nceling terms, weusetheequilibrium expression\\nKPnM\\na\\x88\\x89PnM\\x8aeq\\n\\x89M\\x8aeq\\x89Pn\\x8aeq; \\x852\\x86\\nwithn=1or2,tocomplete thederivation (Methods).\\nOur main result isthen asurprisingly simple and exact analytical relationship that quanti-\\nfiestheequilibrium solution ofourmodel (Fig 1)viaKaeffand theenhancement relative to\\nKaPP.\\nKeff\\na\\nKPP\\na\\x88gKP1M\\naKP2M\\na\\x85\\x89M\\x8aeq\\x862\\x87\\x85KP1M\\na\\x87KP2M\\na\\x86\\x89M\\x8aeq\\x871\\n\\x851\\x87KP1M\\na\\x89M\\x8aeq\\x86\\x851\\x87KP2M\\na\\x89M\\x8aeq\\x86; \\x853\\x86\\nwhere γisadimensionless constant V/(2Aσ), σ=KaPP/2K a2D,PP,[M] eqistheunbound lipid\\nconcentration atequilibrium, and allequilibrium constants (including Kaeff)and concentra-\\ntions areinvolume units (Fig 2).[M] eqisafunction ofallthemodel parameters, and canonly\\nbesolved exactly using numerical methods (Methods); wetherefore derive anadditional\\napproximate analytical equation for[M] eqdescribed below (Fig 2B). However, intheregime\\nwhere lipids areinexcess, theresult ofEq3isparticularly simple because [M] eq~[M] 0,the\\ntotal concentration oflipids (Fig 2A). Critically, thismeans that theinitial experimental condi-\\ntions then directly determine theenhancement. Inthisregime only twofactors control\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 6/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 5}),\n",
       " Document(page_content='enhancement, theratio V/(2Aσ), and thedimensionless strengths ofmembrane localization,\\nKPnM\\na\\x89M\\x8a0,which report theratio ofmembrane bound versus solution proteins (Eq2)and\\nwhich weterm themembrane stickiness. Hence thevolume-to-surface-area ratio, KaPP/Ka2D,PP,\\nandmembrane stickiness play aprimary role intriggering (orpreventing) constitutive assembly\\nonmembranes. The right-hand side ofEq3isalso constant forallKaPPvalues (Fig 2A). Inthis\\nregime, ourEq(3)canalso beapplied toextract binding affinities onmembranes (Ka2D)from\\nexperiments where binding occurs both onmembranes andinsolution. This practical applica-\\ntion ofourresult should help simplify therelatively rare experimental characterization ofpro-\\ntein-protein affinities onsurfaces, astheproteins need notberestricted tothesurface foritto\\nwork. The condition ofexcess lipids canbesatisfied even with alipid recruiter such asPI(4,5)\\nP2,present at2.5x104μm-2intheplasma membrane [15], or~1% oflipids [13] (Fig 2A), aswe\\nexplore further below forproteins involved inCME.\\nWederive anadditional approximate equation for[M] eqtoprovide acomplete equilibrium\\ntheory ofcomplex formation applicable toallexperimental regimes, and wevalidate thisequi-\\nlibrium theory through extensive simulations ofordinary differential equations (ODEs) (Fig 2\\nand Fig3,Methods). Tobriefly outline thederivation, weconsider twolimiting conditions for\\nlocalization tothemembrane: either there arenoprotein-protein interactions (KaPP=0),giv-\\ning\\x89M\\x8a0\\neq,orcomplete protein-protein complex formation (KaPP=1), giving\\x89M\\x8aCoop\\neq(Fig 1D\\nFig2.The equilibrium theory developed here accurately predicts how and when membra nelocaliza tion will\\nenhance protein interaction strength. (a)Our theory (Eq3)shown insolid black lines inallpanels, iscompared with\\nODE simulatio nresults shown inredcircles. Forreference, theblue lineshows thetrend forpure solution binding, i.e.\\nKaeff=KaPP\\n.The green lineshows themaximum achievable enhancement ,occurring ifallproteins were onthe\\nmembran e,Kaeff=γKaPP.The gray dashed lines areincluded tocontrast theKaeffcalculate dusing asimple\\napproxi mation that lacks cooperativ ity(S1Text section 2D). From a1toa2,theKaPMisdecreased, producin glower but\\nstillconstant enhanceme nts,asthelipids areinexcess relative totheproteins .(b)The number ofunbound lipids is\\nplotted asafunction ofKaPPwith allother parame tersfixed (S1Text section 3A), showing how lipid binding isa\\nfunction oftheprotein interaction strength duetothecooperativ eeffect (Fig 1Dand 1E). The theoretical prediction\\nfor[M] eqisshown insolid black. (c)With fewer lipid recruiter srelative tototal cytosolic proteins ([P] tot/[M] tot>1),\\ntheenhanceme ntislesspronounced, although forweak binders (low KaPP)even limited membrane localization causes\\nsignifican tincreases incomplex formation. From c1toc2theprotein concentra tions areincrease d.Allresults useσ=\\n1nm, seeS1Text section 3Aforallother parameters .(d)Ifonly onepartner binds themembrane, theprotein\\ninteraction remains fully 3Dand noenhanceme ntoccurs.\\nhttps://d oi.org/10.1371/j ournal.pc bi.1006031. g002\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 7/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 6}),\n",
       " Document(page_content='and 1E). These twobounds areindicated bydashed lines inFig2Band both limits areinde-\\npendent ofKaPP.Wecancontinuously interpolate between them (Fig 2B)using thedefinition\\n\\x89M\\x8aeq\\x88\\x89M\\x8a0\\neq\\x851\\x00l\\x86\\x87\\x89M\\x8aCoop\\neql; \\x854\\x86\\nwhere\\x89M\\x8a0\\neqistheroot ofaquadratic equation,\\x89M\\x8aCoop\\neqistheroot ofacubic equation, and λis\\nthefunction ofKaPP(also theroot ofaquadratic) that smoothly interpolates between them\\n(see Methods fordetailed derivations). This theory then provides acomplete description ofthe\\nequilibrium concentrations ofallspecies, asfrom Kaeff,onecandirectly calculate thetotal com-\\nplexes formed. The partitioning ofcomplexes between solution and themembrane cansubse-\\nquently bederived utilizing theequilibrium relations ofFig1(S1Text section 2A). The larger\\ntheenhancement, themore complexes must beonthemembrane (S1E and S1FFig).\\nBounds onbinding enhancement aredetermined byV/A and KaPP/Ka2D,PP\\nUsing ourmain result, Eq3,onecanpredict when and byhow much membrane binding\\nwillenhance complex formation ofbinding pairs without performing anysimulation orex-\\nperiment. Further, wecanassess whether thisenhanced complex formation isrobust topertur-\\nbations inbinding affinities orconcentrations. Toestablish possible values forKaeff,wefirst\\nask: arethere anycases where membrane binding willreduce protein complexation, i.e.\\nKaeff<K aPP?Toanswer this, weconsider thecase where allproteins areonthemembrane\\n(KPnM\\na\\x89M\\x8a0!1), such that wehave pure 2Dbinding and Kaeff=γKaPP.Reduced protein\\nFig3.Protein interaction saided bystrong protein-lipi dinteraction s,abundant lipid recruiters, and low protein\\nconcent rations benefit most widely from membrane localizatio n.a,b)Enhancemen tratios from ODE simulation\\n(colored lines) and theory (black solid lines inallpanels). The dashed black lineistheupper limit fortheenhancement\\nratio given byKaeff/KaPP=γ.σ=1nm. Between thelimiting cases where themembran esurface area iseither toolarge\\ntoenhance bindin g(V/A\\x00!0) oristoosmall toeffect binding (V/A\\x00!1), abroad region ofenhanced binding\\noccurs. The vertical gray lineistheV/A ratio fortheyeast plasma membran e(0.5μm), forreference, and most\\nphysiologi cV/A values fallintherange ~0.05±20μm (S3Table). Within thisphysiologi crange ofV/A values iswhere\\nwegenerally observe thelargest enhanceme nts.The cellgeometry pictures provide examples ofhow onecanproduce\\ndifferent V/A ratios along thex-axis relative tothecentral sphere. Amaximu menhancement foreach parame tersetis\\nreached atavalue ofV/A where lipids stilloutnumbe rproteins (shown onupper xaxis). Increasing (a)protein-l ipid\\naffinities and (b)lipid concentratio nsproduces greater possibili tiesforenhanceme nt.(c)Increases inthemembran e\\nstickiness (KaPM[M] eq)produces monotonic increases inenhanceme ntforallvalues of γ>1. (d)Forlower expression\\nlevels relative tobinding strength (KaPP=106M-1),membran elocalizati oncanactasaswitch toturn onassembly from\\n<50% to>50% (shaded areas). (e)Timescales toequilibrat ewere calculate dfrom simulatio nsofboth ODEs (solid\\nlines) and reaction-di ffusion (RD) (green points) (Methods) .Weak lipid binding canreduce speeds (blue) relative to\\npure solution binding (dashed). The approximate theoretical bounds shown here fortime-scales ofbinding either\\npurely insolution (dashed) oronthemembran e(gray) derive from thekinetics ofirreversible association (S1Text\\nsection 4C).\\nhttps://d oi.org/10.1371/j ournal.pc bi.1006031. g003\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 8/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 7}),\n",
       " Document(page_content='complexation willoccur only ifγ<1, orV/A<2σ. The sizeofσcontrols therelative strength of\\nKa2D,PPvs.KaPPand varies from onebinding pair toanother, butexperiment and theory indi-\\ncate itisofthenanometer length-scale [21, 22,26]. Wetherefore collected theV/A ratios fora\\nwide range ofcelltypes and organelles toillustrate that innearly allcells, V/A>2σ (~20nm)\\nand thus membrane localization willenhance binding (S2Fig). Indeed, targeting theplasma\\nmembrane inmost cells results in γvalues much greater than 1,intherange 10±1000 (S2Fig).\\nUltimately, theV/(Aσ) ratio isabsolutely central incontrolling observed enhancement, asit\\nsetsthemaximum achievable Kaeff.Inmost cases proteins willend upmixed between solution\\nand membrane, and from Eq3thisgives usKaPP\\x14K aeff<γK aPP(Fig 2).Inthecases where only\\noneprotein binds tolipids, all2Dlocalization benefits arelostand noenhancement occurs:\\nKaeff=KaPP(Fig 2E).\\nCooperativity emerges inprotein-lipid binding\\nAnimportant feature that ouranalysis captures isthecoupling that emerges between protein-\\nprotein affinity and protein-lipid binding duetomembrane localization. Ifabound protein-\\nprotein complex localizes tothemembrane bybinding onelipid, binding toasecond lipid\\nthen becomes a2Drather than a3Dsearch [12] (Fig 1Dand 1E). Thus, stabilization ofpro-\\nteins onthemembrane isachieved notonly through strong protein-lipid interactions, butby\\nfeedback from strong protein-protein interactions. This cooperative effect forlipid binding\\n(binding ofonelipid changes theaffinity forthesecond lipid) produces theunexpected result\\nthat thenumber ofproteins bound tothemembrane isdependent ontheprotein-protein\\ninteraction strength (Fig 2B). Tocontrast, onecould instead consider lipid binding assimply\\npartitioning proteins between solution and membrane, after which they form complexes asin\\nFig1Aand 1C.This simplified interpretation, shown ingray dashed lines inFig2,does not\\ncapture thecooperative effect (Fig 1Dand 1E)and isclearly wrong forstrong binding proteins\\nthat rely oncooperativity tostabilize complex formation (see S1Text section 2Dfordetails).\\nHow binding enhancement responds toperturbations and factors external\\ntothemodel\\nWith ourtheory, itispossible todirectly probe how changes incellgeometry, binding affini-\\nties, orconcentrations willregulate enhancement. Although themodel istoosimple tofully\\ndescribe multi-component assembly, byvarying theinput parameters tomimic competing\\ncytosolic factors, onecanevaluate therelative importance ofconcentration, affinities, and\\ngeometry. This isparticularly true ofequilibrium invitro experiments. Forchanges togeome-\\ntry,wefirst note that theequilibrium results depend only ontheratio and nottheabsolute size\\nofVorA.The membrane does nothave tosurround thesolution volume liketheplasma\\nmembrane butcanreflect binding totheoutsides ofliposomes, forexample, which allows for\\nstudying anyV/A ratio. Although itmay seem that increasing V/A (through γinEq3)always\\nincreases Kaeff,thisisnotthecase when [M] 0iskept constant initsnatural units ofμm-2(itis\\nthen converted into Volume units bymultiplication byA/V). Vand Athus also control theini-\\ntialcopy numbers ofproteins and lipids separately, such that large values ofV/A have agreat\\nexcess ofproteins over lipids. This drives Kaeff\\x00!K aPP(Fig 3Aand 3B). Formost physiologic\\nV/A values (~0.05±20μm S3Table) however, and physiologic concentrations ofproteins\\n(1nM-10μM) orlipids (103−105μm-2S4Table), proteins arenotingreat excess, meaning sig-\\nnificant enhancement isachievable depending onthemembrane localization strength (Fig 3A\\nand 3B).\\nMembrane composition. Because themembrane isdescribed inourmodel only byits\\nsurface area Aand theconcentration oflipids targeted bytheprotein lipid-binding domains,\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 9/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 8}),\n",
       " Document(page_content=\"thespatial and chemical heterogeneity ofcellmembranes isnotcaptured. The stickiness ofa\\nmembrane forspecific proteins iscaptured inourmodel, however, bythedimensionless prod-\\nuctKaPM[M] eq,which issimply equivalent tooneprotein's ratio ofmembrane bound to\\nunbound copies (e.g. [MP 1]/[P 1])atequilibrium. Asisclear from Fig3C,theenhancement isa\\nmonotonically increasing function ofKaPM[M] eqforall γ>1 geometries. Hence with either\\nstronger affinity forthemembrane (KaPM)orahigher concentration oftarget lipids ([M] 0),\\nlocalization tothemembrane and thus enhancement willincrease. Fig3Calso illustrates how\\nonce KaPM[M] eqexceeds values of~10, maximum enhancement isreached and increasing the\\nstickiness ofthemembrane does notchange theresulting binding. IntheS1Text section 2B,\\nwederive anexplicit formula forthiscritical value ofKaPM[M] eq,aswell asthecorresponding\\ncritical value forlipid concentration, [M] c(S3A and S3B Fig). Ifthelipid concentration and\\naffinity contribution tostickiness canbedeconvoluted, asisthecase forproteins that target\\nindividual lipids (such asmany PI(4,5)P 2binders, Table 1)atessentially a1:1ratio, onecan\\nindividually evaluate how decreases ineither lipid concentration (Fig 3B)oraffinity (Fig 3A)\\nwilllower enhancement. Importantly, wefind that surprisingly lowlipid concentrations are\\nsufficient todrive maximal enhancement. ForV/A =1μm, (about thevalue foryeast cells), the\\nPI(4,5)P 2concentration of2.5x104μm-2produces close tothemaximum inenhancement,\\nmeaning adding more lipids makes minimal difference (Fig 3B,S3A and S3B Fig). Forproteins\\nsuch asBAR domains, incontrast, assigning values foraffinity and lipid concentration would\\nrequire acomposition dependent interpretation ofthese values, asBAR domains arelessselec-\\ntiveforindividual lipid types and may only bind stably toclusters oflipids rather than 1:1(see\\nS1Text section 5forextended discussion). Nonetheless, anychange inmembrane composi-\\ntion that increases itsstickiness towards anyspecific membrane binding domains willclearly\\ndrive upbinding interactions between associated protein pairs (Fig 3C).\\nCompetition forprotein and lipid binding. Thus farwehave said little about protein\\nconcentrations orKaPP,asenhancement isindependent oftheir magnitude when lipids arein\\nexcess. However, these protein variables always determine when enhancement acts asaswitch\\ntoturn oncomplex formation. Some proteins areperfectly capable offorming strong com-\\nplexes insolution, whereas protein pairs with lessthan 50% complexes formed insolution can\\nexperience dramatic increases inbound complexes (Fig 3D,S3E Fig). The dependence of\\nenhancement onprotein concentration isalso monotonic forfixed geometries: asprotein con-\\ncentration drops, enhancement increases (S3D Fig). Hence, competition forbinding anyof\\nourprotein binding pairs insolution would beexpected tolower initial concentrations ofeach\\ncomponent. This willincrease theultimate enhancement (S3D Fig) and inmany cases, make\\ntheproteins more sensitive tolocalization asatrigger forassembly (Fig 3D). Ifcompetition for\\nprotein binding partners involved other proteins that also bound tothemembrane, then\\nenhancement could beincreased, butthisextension beyond theFig1model would have tobe\\nquantified viasimulation. Competition forlipid pools, ontheother hand, willalways decrease\\nenhancement, asshown inFig3Band explored further below forCME proteins.\\nSensitivity toprotein-protein affinity. Mutations toproteins would largely affect their\\naffinities, either fortheir protein ortheir lipid partners. Asnoted above, however, even many-\\nfold changes inaffinity may have minimal consequences onmeasured enhancement (S3G\\nFig), and onresulting complex formation, duetothenonlinear dependence ofcomplexes on\\naffinity. Formutations toENTH/ANTH domain containing proteins, ahigh concentration of\\ntarget PI(4,5)P 2lipids means that most proteins bind tothemembrane, where they willform\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 9}),\n",
       " Document(page_content='affinity. Formutations toENTH/ANTH domain containing proteins, ahigh concentration of\\ntarget PI(4,5)P 2lipids means that most proteins bind tothemembrane, where they willform\\nmaximal complexes, atmoderate KaPMvalues. Under these conditions, enhancement and total\\ncomplexes willnotchange significantly even with 10-fold changes inKaPM[27] (S3H Fig).\\nDecreases inKaPPcanhave similar effects, either notaffecting enhancement when membrane\\nstickiness isalready high, orotherwise increasing enhancement (S3C Fig,Fig2).Forcomplex\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 10/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 9}),\n",
       " Document(page_content='formation, decreases inKaPPcanbemuch more significant, acting toincrease thesensitivity of\\ncomplex formation tolocalization, such that itismore likely toactasatrigger forassembly\\n(S3E Fig). Mutations that asymmetrically affect theKa2Dvalues would result inchanges inσ\\nvalues, with smaller values always driving larger enhancement and stronger complexation on\\nthemembrane (S3F Fig).\\nTimescales ofassembly vary strongly with binding affinity\\nOur theory (Eq3)only describes theequilibrium state ofthemodel. However, wecandeter-\\nmine speeds ofassembly viasimulation. Now, thebinding rates and theabsolute values ofV\\nand A(not justtheratio) willinfluence thekinetics (allsimulation inputs inS2,S3,and S4\\nDatasets). Forthese time-scales, wefind that protein-lipid affinities KaPMaremost often\\nshown tobecritical incontrolling theoverall time-scales ofcomplexation, even driving slow-\\ndowns inspeeds relative tosolution binding (Fig 3E,S4Fig). Changes indiffusion from solu-\\ntion tothemembrane (about 100times slower) affect themagnitude ofassociation and disso-\\nciation rates and arecaptured implicitly inourODE simulations (Methods), and explicitly in\\nourspatially resolved reaction-diffusion simulations [19, 20]. However, theinfluence ofdiffu-\\nsion onthereaction rates israrely adominant factor inphysiological rate regimes (S4Fig),\\nindicating itisthebinding strengths rather than slow 2Ddiffusion that determine assembly\\nspeeds. However, wenote that ourcomparison ofODE and RDkinetics wasperformed inrel-\\natively small RDsystems duetosimulation costs, and itistrue that asspatial dimensions\\nincrease, times todiffuse toreach themembrane willinfluence theoverall equilibration times.\\nThe timescales wecalculated forprotein pairs and scaffold mediated systems (S4B and S4C\\nFig) were performed using ODEs attheir corresponding cellular dimensions (S3Table):\\nV=1200μm3(human) and V=37.2μm3(yeast). Performing RDsimulations atthese dimen-\\nsions would produce slower relaxation times, particularly forhuman cells, duetothetime\\nrequired toreach thesurface. Crowding would also lower effective diffusion constants ofpro-\\nteins, although thedecrease intime-scales toequilibrate would benegligible unless binding\\nrates were strongly diffusion-influenc ed(Methods).\\nBiological relevance forproteins inCME\\nTotestthebiological relevance ofmembrane localization fordriving complex formation and\\nassembly, wecollected biochemical (Table 1,S1Table, S2Table), concentration (Table 1,S4\\nTable), and cellular geometry data (S3Table) forinteractions among 37membrane targeting\\nproteins inyeast and human cells, including 22proteins involved inclathrin-mediated endo-\\ncytosis (CME). Wefirst study only individual protein pairs that canbind according toour\\nmodel ofFig1,(S1Table) shown inFig4A:themembrane binding proteins AP-2, DAB2,\\nARH, FCHo1, FCHo2, HIP1, HIP1R, PICALM, SH3GL2, EPN, AP180, SLA2, and SYP1. In\\nFig4Band 4Cweshow results ofbinding between specific pairs. Weused cytosplasmic con-\\ncentrations oftheproteins (Table 1)and thetargeted lipids (S4Table), and therelative Volume\\nand Area from their respective celltypes (S3Table). Binding constants arecollected from pre-\\nvious experimental studies (Table 1,S1Table, S3Dataset), and for2Dbinding constants we\\ntestvalues ofσ=1nm (Fig 4)and 10nm (S5Fig). Our results thus provide quantitative insight\\ninto how these pairs inisolation would usemembrane localization atphysiologic conditions to\\ndrive their protein-protein interactions. Forsome proteins, such asAP-2, solutionKPP\\navalues\\nhave been measured with partners (Fig 4A,S1Table), butfurther experiments indicate that the\\nproteins undergo minimal binding insolution duetoconformational regulation [1].Despite\\nthisadditional regulation, membrane localization willstillincrease complex formation relative\\ntowhat isobserved insolution (γ>1), sotheeffect isquantified here using themeasured KPP\\na\\nMembrane localizatio ncantrigger protein assembly', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 10}),\n",
       " Document(page_content='towhat isobserved insolution (γ>1), sotheeffect isquantified here using themeasured KPP\\na\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 11/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 10}),\n",
       " Document(page_content='value (S1Table). Using ourtheory along with simulations forverification and time-scales, we\\nfind that affinities ofthese CME binding pairs canbeenhanced 10±1000 fold bybinding to\\nmembranes (Fig 4B,S5Figforresults withσ=10nm). With limited binding insolution for\\nmost pairs, membrane localization then triggers adramatic increase incomplex formation\\n(Fig 4C,S5Fig). The central adaptor protein AP-2 isresponsible formany ofthese interac-\\ntions, showing thecapacity totrigger assembly with nearly allofitsbinding partners (Fig 4C,\\ngreen bars). Even though weassume binding ispossible forAP-2 insolution, itisstillquite\\nlimited prior tolocalization. Not surprisingly, knockdown ofAP-2 inmammalian cells causes\\nsevere disruption ofendocytosis [28], underlining itssecondary importance only totheirre-\\nplaceable clathrin [28] and PI(4,5)P 2[29]. Wenote that AP-2 canpotentially bind uptothree\\nPI(4,5)P 2copies [30, 31], meaning that there willbelessfree lipids available foreach AP-2.\\nWith fewer lipids, enhancement and complexation willbereduced, butisstillquite large (S5\\nFig). Forsome proteins such asFCHo1 (SYP1 inyeast), binding affinities (KaPPorKaPM)are\\nnotavailable, and thisF-BAR protein does nottarget asingle lipid specifically. However, by\\nconsidering ranges ofmembrane stickiness values, wecanuseourmethod toidentify which\\ncombinations (S6Fig) best describe theexperimental observation that these proteins only\\nlocalize effectively tomembranes when they canbind other proteins [18, 32]. Wefind forthis\\nprotein, membrane stickiness values of~0.5 produce membrane targeting that issensitive to\\nprotein-protein interactions, whereas once values exceed ~1,nopartners areneeded totarget\\nthemembrane effectively (S6Fig).\\nWefurther interrogate twoadditional mechanisms forstabilization atthemembrane by\\nlipid binding proteins such asAP-2, epsin, and Dab2 [3].First, they each bind transmembrane\\ncargo after membrane localization, which acts toeffectively increase theKaPMbyincreasing\\ntheir residence time onthemembrane. KaPMisafactor of~40 higher forAP-2 binding toPI\\n(4,5)P 2when cargo isavailable [30]. Interestingly, these cargo stabilized interactions (Fig 4B,\\nlight green) donotmake asignificant impact oncomplexation when weassume thefull1%PI\\n(4,5)P 2concentration isfree tobind, asthenumerous lipids outweigh aneed forstronger\\nbinding (Fig 4C,light green). However, when weevaluate complexation with PI(4,5)P 2pools\\ndiminished byafactor of10duetoassumed competition from other PI(4,5)P 2binders, now\\nFig4.Membrane localizatio ntriggers strong complex formation forpairs ofprotein binding partners involved in\\nclathrin-me diated endocytosis .(a)Interactio nsbetween lipid binding human (green) and yeast (blue) CME proteins\\nareshown along with KdPPvalues (μM) measured (red text), inferred through structur aland functiona lhomology\\n(green), orestimated (blue) (Table 1,S1Table, S3Dataset). Sizes indicate concentra tions (Table 1).(b)Enhancem ents\\nforeach ofthese CME binding pairs, following themodel ofFig1,were calculated using Eq3and verified through\\nnumerical simulation ofODEs. Pairs involving human AP-2 areindark green, light green points involve AP-2 with\\ncargo binding-ad justed KaPM,with redand blue points showing other human and yeast proteins, respective ly.The x-\\naxis istheaverage membran estickiness foreach pair, andhere weassumeσ=1nm (see S5Figforσ=10nm). For\\npoorly characte rized lipid binding affinities, weconside redranges ofvalues (error bars inx),producing ranges of\\nenhanceme nts(error bars iny).Gray lines areguides forfixed V/A ratios. Protein names inparenthes esarehomo-\\ndimers. (c)The percent ofproteins incomplexe sincreases from solution bindin glevels (gray bars) asaresult of\\nmembran elocalizati on(colored bars). Allresults and parameters used foralldata points inS3Dataset.\\nhttps://d oi.org/10.1371/j ournal.pc bi.1006031. g004\\nMembrane localizatio ncantrigger protein assembly', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 11}),\n",
       " Document(page_content='https://d oi.org/10.1371/j ournal.pc bi.1006031. g004\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 12/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 11}),\n",
       " Document(page_content='cargo stabilization viahigher KaPMdoes help recover strong complexation onthemembrane\\n(S5Fig). This suggests that cargo binding, which isamain functional goal ofCME, becomes a\\nsignificant regulator ofadaptor stabilization when competition from multiple adaptors limits\\nPI(4,5)P 2binding. Second, when these adaptors canbind multiple partners with distinct\\nappendage domains [33], weseemore proteins onthemembrane duetotheincreased diffi-\\nculty ofun-tethering from themembrane domain (S6B Fig).\\nInthecellular environment, CME proteins areofcourse notinisolation and canboth com-\\npete and cooperate with oneanother toform higher order assemblies, induce conformational\\nchanges, and occupy lipid binding sites onthemembrane. Thus wecanonly speculate about\\ntherole oflocalization innucleating clathrin-coated pitsinvivo. However, based ontheabove\\nanalysis showing that, physiologically, γisgreater than 1,membrane localization willdrive cla-\\nthrin towards complex formation. The initial nucleation ofclathrin-coated pitsites isdifficult\\ntoresolve experimentally because ofthechallenges intracking themany participatory proteins\\nsimultaneously, and because prior tocage formation, thedensity ofmolecules is,bydefinition,\\nlow. Experiments have tracked therole ofAP-2 and clathrin innucleating sites [34], which we\\ndiscuss below.\\nScaffold mediated interactions ofCME proteins also exploit membrane\\nlocalization\\nTogobeyond ourFig1model ofpairwise protein binding and thus characterize how scaffold\\nproteins (Table 1:ITSN1, EPS15, EDE1 and SLA2) stabilize complex formation atthemem-\\nbrane despite notdirectly interacting with thelipids (model inS7Fig,listofinteractions inS2\\nTable), wesimulate systems ofODEs, asEq3nolonger applies (Methods). Wethus simulate\\ninteractions involving three proteins, twoofwhich canbind lipids butnoteach other, and the\\nthird that binds both peripheral membrane proteins butnotthemembrane (Fig 5A). Our\\nresults inFig5Band 5Cshow that while scaffold mediated complexes canstillcapitalize on2D\\nlocalization forbinding (Fig 5B), because localization isnow mediated byperipheral mem-\\nbrane proteins that areatmuch lower concentrations than thelipid recruiters, wefind that the\\nincrease incomplex formation islessrobust (Fig 5C), and islimited byconcentration ofthe\\nscaffold protein (S8A Fig).\\nClathrin cage nucleation and BAR domain oligomerization can exploit\\nmembrane localization\\nThus farwehave notdiscussed clathrin, thecentral component oftheCME vesicles that does\\nnotactually bind tolipids itself.Invitro experiments find that clathrin polymerization onthe\\nmembrane (via adaptor binding) ismore robust than occurs insolution (with adaptors still\\npresent), supporting arole formembrane localization initsnucleation and assembly [2].Cla-\\nthrin isatrimeric protein with three binding sites totarget peripheral membrane proteins. It\\npolymerizes with itself into hexagonal lattices without competition from theperipheral mem-\\nbrane proteins. Thus, itsinteractions with peripheral membrane proteins notonly increase the\\nquantity ofprotein bound tothemembrane, itcanhelp drive 2Dpolymerization between cla-\\nthrin trimers. Through (non-spatial) stochastic simulations (Methods) set-up tomimic recent\\ninvitro experiments [1],weexplored arange ofclathrin-clathrin interaction strengths toshow\\nhow membrane recruitment bytheAP-2 adaptor [1]canenhance clathrin polymerization\\nyield (S8Fig). Although these simulations lack molecular structure orspatial resolution, they\\ncantrack formation ofmulti-protein complexes and theimportant role ofaffinity and concen-\\ntration incontrolling these complexes. Wefind that clathrin localizes tothemembrane first\\nviaAP-2 binding before assembling into cages in2Dforthemost reasonable Kdsof10±100μM\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 13/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 12}),\n",
       " Document(page_content='[35]. This result issupported byevidence frominvivo experiments that probe theearly stages\\ninthenucleation ofclathrin coated pitsites through tracking ofAP-2 and clathrin [34]. They\\nfound that clathrin arrives atthemembrane most frequently (75%) asasingle trimer, and\\nbound toatleast onebutmost often twoAP-2 molecules [34]. Nucleation canthen initiate in\\ntwoways: (A)another clathrin trimer localizes tothemembrane viaAP-2 and these trimers\\ndimerize in2Dor(B)another clathrin trimer isdirectly recruited bytheclathrin onthesur-\\nface. Although thesubsequent clathrin dimerization events were notresolved experimentally,\\npreventing definitive evidence ofmembrane localized clathrin-clathrin assembly, thefactthat\\neach clathrin isbound toAP-2 suggests that AP-2 binding ofclathrin isaprerequisite forini-\\ntialclathrin dimerization. From oursimulations, the(A)nucleation process ismarkedly domi-\\nnant. There isastrong driving force both from affinity and from concentration forAP-2 to\\nbind anyofthe25000 PI(4,5)P 2/um2,and correspondingly minimal drive forasolution cla-\\nthrin tobind asmall number ofclathrin trimers localized tothesurface. Wenote that because\\nclathrin also arrives atthemembrane asdimers orhigher order complexes 25% ofthetime,\\nsolution binding ofclathrin also contributes tonucleation ofpitsites, buttoamuch lower\\nextent [34]. Interestingly, once pitsites have formed, assembled clathrin cages exchange with\\nsolution clathrin with theaidofATP-consuming proteins that facilitate remodeling ofthecla-\\nthrin cage [36]. Thus clathrin-clathrin interactions from solution certainly play animportant\\nrole inthecellinmaturing thepitsites [36].\\nCME proteins with BAR domains that dimerize, appear tooligomerize only onthemem-\\nbrane, and arefunctionally important fordriving membrane deformation [17, 18]canalso\\nexploit localization todrive their binding interactions. Westudy isolated FCHO1/2 oligomeri-\\nzation and endophilin (SH3GL2) oligomerization, again using non-spatial stochastic simula-\\ntions (S9Fig). Here again weconsider arange ofKaPMvalues tocapture uncertainty inthe\\nmembrane stickiness ofthese domains. Wefind thestoichiometry ofthedimerization pair\\nFig5.Scaffold-medi ated interaction scan also exploit membra nelocaliza tion todrive complex formation inclathrin-med iated endocyto sis.a)When setsofthree\\nproteins canform acomplex andtwoofthem also bind tolipids, localizati onisagain capable ofdriving stronger complex formatio n.Yeast proteins inblue and human\\nproteins ingreen, where theupperm ostfour (SLA1, EDE1, EPS15, ITSN1) arescaffold proteins that donotbind lipids. Affinities follow thesame color scheme asFig4,\\nand wehave included only binding interactions between these proteins that areshown inparts b-c.(b)Because ourprimary model (Fig 1)nolonger applies, wecannot\\nuseEq3topredict enhancement s,and instead usesimulations ofODEs. The scaffold model isdefined inS1Text section 1Cand S7Fig,along with thedefinitions of\\nKaeff,SPand Kasol,SP,where neithe risatrue equilibriu mconstant andtheequilib rium results ofthesimulatio nsareused tocalculate their defined values. Without\\nmembran elocalizati on,Kaeff,SP!K asol,SP.Scaffold proteins (sinthelabels) areeps15/ed e1(orange), oritsn1/sla1 (pink). (c)Complexatio ninvolves allthree proteins (see\\nS1Text section 1D) and here again, thepercent ofproteins incomplexe sincreases from solution binding levels (gray bars) asaresult ofmembrane localization (colored\\nbars). Allresults andparameters inS4Dataset.\\nhttps://do i.org/10.1371/j ournal.pc bi.1006031. g005\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 14/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 13}),\n",
       " Document(page_content='(homo orhetero) iscentral indetermining whether large oligomers form. With matched pairs,\\nhomodimers such asendophilin form larger oligomers that feedback into higher stabilization\\natthemembrane, whereas thedisparity inFCHo1 and FCHo2 concentration (S1Table) pro-\\nduces more isolated dimers. Experiments have shown that BAR domains exhibit stronger\\nbinding tocurved membranes [17]. Because welack thiscooperative feedback inourmodel\\nbetween oligomers tubulating membranes and thus potentially increasing affinity forsubse-\\nquent proteins, ourresult canbeinterpreted asalower bound onobserved oligomerization.\\nInallcases, animportant outcome ofthese strong binding interactions onthemembrane is\\nthat they aredifficult todisassemble, consistent with findings that unproductive assembly\\nevents observed invivo [37] require theATP-driven uncoating machinery fordisassembly [2,\\n38]. Our results demonstrate that establishing thephysiologic significance ofthese polymeriza-\\ntion observations depends notonly onprotein concentrations and solution conditions, but\\nalso theV/A ratio. Thus, thisratio should beregarded asacritical factor indesigning invitro\\nexperiments tobetter reflectinvivo behavior.\\nBiological relevance formembrane remodeling pathways inyeast\\nLastly, ouranalysis motivates why diverse proteins that target membranes inyeast canfollow\\npathways toassembly both similar and distinct from theCME proteins. Inparticular, theCME\\nprotein pairs produce limited protein-protein complexes when isolated insolution, butexperi-\\nence large enhancements duetomembrane localization, triggering widespread protein-protein\\ninteractions only after binding tothemembrane (Fig 4).For15yeast proteins involved in\\ndaughter cellbudding, lipid regulation, and morphogenesis, westudied their pairwise binding\\ninusing cytoplasmic (yeast) concentrations (Table 1),lipid concentrations (S4Table), cyto-\\nplasmic V/A ratios (S3Table) and experimentally measured protein-lipid affinities (Table 1).\\nWefind binding enhancements arehigh (100±1000), similar totheCME proteins, indicating\\nthat binding willbepromoted once proteins areonthemembrane (S10 Fig,S2Dataset).\\nAlthough enhancements were readily measured forthese yeast binding pairs because they\\nwere independent ofKaPPvalues (S2Dataset), wecould notdirectly compare complexation for\\nthese interactions aswedidfortheCME interactions because they lacked anyKaPPdata. For\\nbinding enhancements, wefound anexception inthecoat forming proteins targeting endo-\\nsomes (VPS5, VPS17, SNX4, SNX41), which only exhibit enhancements <20. These proteins\\ntarget thePI(3)P lipid butmost bind only weakly (KdPM>100μM) [39], limiting their enhance-\\nments despite afavorable V/A ratio attheendosome (S10 Fig,S3Table). Unlike inCME, how-\\never, these coat proteins form stable interactions insolution [40]. Thus, rather than membrane\\nbinding triggering protein interactions, wewould first expect thereverse: strong protein inter-\\nactions insolution function totarget and stabilize protein atthemembrane through thecoop-\\nerative effect (Fig 1Dand 1E,Fig2B). Wetesthow thebinding oftheretromer components\\nVPS5 and VPS17 totheendosome willbesignificantly enhanced byforming ahigher order\\nassembly insolution with thestrong lipid binding cargo adaptor, SNX3 (S9Fig). SNX3 targets\\nPI(3)P with stronger affinity (~2μM) than either VPS5 orVPS17 [39], and isknown to\\nimprove recruitment oftheretromer toendosomes [40]. Once these small pre-assembled coat\\nsubunits areonthemembrane, they canthen continue toexploit localization toform larger\\nprotein coats.\\nConclusions\\nWeconclude bynoting that assembly onmembranes isregulated tooccur atspecific times or\\nsub-cellular locales, and ourtheory provides auseful aidinpredicting thechanges inlocal pro-\\ntein, lipid concentrations, and affinities that arenecessary totrigger (orprevent) such\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 15/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 14}),\n",
       " Document(page_content='assembly. Ultimately, ourtheory ismost powerfully applied tointerpreting invitro results, due\\ntothesimplifying assumptions ofthemodel, andcanimprove thedesign andquantitative inter-\\npretation ofassays probing multi-protein complexation atmembrane surfaces. Also, given\\nknown protein-protein andprotein-lipid binding affinities, ourtheory canquantitatively pre-\\ndicttheresults ofinvitro experiments that mimic Fig1,thus avoiding theneed forsuch mea-\\nsurements. Our results indicate that even relatively lowlipid concentrations (i.e.PI(4,5)P 2at\\n~1% ofplasma membrane lipids) canbesufficient inmany cases tostabilize proteins tomem-\\nbranes anddrive protein-protein interactions. Wefound that additional factors, such ascargo\\nbinding byadaptor proteins inCME, areonly strong regulators ofmembrane localization or\\nprotein interactions under specific conditions. Since cargo-binding isknown toinfluence the\\nsuccess ofvesicle formation invivo[41, 42],thissuggests that thecondition where total PI(4,5)\\nP2concentration isreduced tomimic competition from other proteins ismore physiologically\\nrelevant. Afruitful means ofexploring inmore detail therole ofcytoplasmic factors, aswell as\\nspatial heterogeneity, crowding, andnon-equilibrium dynamics, isthrough reaction-diffusion\\nsimulations, although wenote theresults willthen bedependent onmany additional parame-\\nters. Overall, thetheory weprovide here offers ageneral anduseful quantitative guide forpre-\\ndicting when orifmembrane localization plays arole inthecellular control ofself-assembly.\\nMethods\\nTheoretical derivations\\nDerivation details ofKaeff(Eq 3). The exact solution (both equilibrium and\\ntime-dependence) toourmodel ofproteins interacting and recruiting tomembranes\\n(Fig 1)canonly beobtained numerically. Starting from ourdefinition inEq(1),(Keff\\na\\x88\\n\\x85\\x89P1P2\\x8aeq\\x87\\x89MP 1P2\\x8aeq\\x87\\x89P1P2M\\x8aeq\\x87\\x89MP 1P2M\\x8aeq\\x86\\n\\x85\\x89P1\\x8aeq\\x87\\x89MP1\\x8aeq\\x86\\x85\\x89P2\\x8aeq\\x87\\x89P2M\\x8aeq\\x86),weinput pairwise equilibrium expressions foreach species in\\nthenumerator, where thecomplete listofpairwise equilibria illustrated inFig1aregiven by\\nequations:\\nP1\\x87P2⇋P1P2\\x85KPP\\na\\x86 5:1\\nM\\x87P1⇋MP1\\x85KP1M\\na\\x86 5:2\\nM\\x87P2⇋P2M\\x85KP2M\\na\\x86 5:3\\nM\\x87P1P2⇋MP1P2\\x85KP1M\\na\\x86 5:4\\nP1P2\\x87M⇋P1P2M\\x85KP2M\\na\\x86 5:5\\nMP1\\x87P2⇋MP1P2\\x85KPP\\na\\x86 5:6\\nP1\\x87P2M⇋P1P2M\\x85KPP\\na\\x86 5:7\\nMP1\\x87P2M⇋MP1P2M\\x85KPP\\na=\\x852sPP\\x86\\x86 5:8\\nM\\x87P1P2M⇋MP1P2M\\x85KP1M\\na=\\x852sP1M\\x86\\x86 5:9\\nMP1P2\\x87M⇋MP1P2M\\x85KP2M\\na=\\x852sP2M\\x86\\x86 5:10\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 16/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 15}),\n",
       " Document(page_content=\"where 5.8,5.9and 5.10 areallin2D.Reactions in2Dlistthe2DKavalues and thus require spe-\\nciesbeinunits ofArea-1.Tosolve allspecies inconsistent units, where wewillusesolution\\nconcentrations with units V-1,thelisted 2DKavalues must bemultiplied byV/A. This isthe\\norigin ofthe γfactor, g\\x88V\\n2As.Wenote that theprotein-protein interactions have thesame\\nequilibrium constant when oneorboth proteins areinsolution (Eqs 5.1,5.6and 5.7). This def-\\ninition preserves detailed balance, and thus anequilibrium steady-state. This isbased onthe\\nassumption that because thebinding equilibrium isstilldriven bythesolution exchange of\\n(one) protein with thecomplex using thesame interfaces, therelative fraction ofbound and\\nunbound states isthesame.\\nInputting Eqs5.1,5.6,5.7and 5.8into thenumerator ofEq1and dividing numerator and\\ndenominator bythefactor [P1]eq[P2]eqgives:\\nKeff\\na\\x88KPP\\na\\x00\\ng\\x89MP2\\x8aeq\\x89MP1\\x8aeq\\n\\x89P2\\x8aeq\\x89P1\\x8aeq\\x871\\x87\\x89MP1\\x8aeq\\n\\x89P1\\x8aeq\\x87\\x89MP2\\x8aeq\\n\\x89P2\\x8aeq !\\n\\x01= 1\\x87\\x89MP1\\x8aeq\\n\\x89P1\\x8aeq !\\n1\\x87\\x89MP2\\x8aeq\\n\\x89P2\\x8aeq !  !\\n\\x856\\x86\\nAnd finally using Eqs5.2and 5.3above (Eq2ofthemain text), werecover ourmain result, Eq\\n3.Wealso note ifP1and P2target distinct lipids, the[M] concentrations willbesubscripted\\naccordingly. Thus Eq3ofthemain textisexact. However, aseparate equation for[M] eqis\\nneeded that willbeapproximate.\\nDerivation details of[M] eq(Eq 4). Our equation fortheunbound lipids atequilibrium,\\n[M] eq,isafunction oftwolimiting cases forprotein localization tothemembrane with a\\nsmooth interpolation inbetween defined viaEq4ofthemain text(\\x89M\\x8aeq\\x88\\x89M\\x8a0\\neq\\x851\\x00l\\x86\\x87\\n\\x89M\\x8aCoop\\neql).Inthefirst extreme (KaPP=0),wesolve forunbound lipids atequilibrium based\\nsolely onprotein-lipid interactions, M+P⇋MP, giving thefamiliar quadratic root\\n\\x89M\\x8a0\\neq\\x881\\n2\\x89M\\x8a0\\x00\\x89Ptot\\x8a0\\x001\\nKPM;av\\na\\x87\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81 \\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81 \\x81\\n\\x89M\\x8a0\\x00\\x89Ptot\\x8a0\\x001\\nKPM;av\\na\\x12 \\x132\\n\\x874\\x89M\\x8a0=KPM;av\\nas 0\\n@1\\nA\\x857\\x86\\nThis equation recovers [M] eq~[M] 0,theinitial concentration oflipids, when [M] 0>>[P tot]0=\\n[P1]0+[P 2]0.The KaPM,avistheaverage from both proteins. Forsignificant differences between\\naffinities and protein populations, theweighted average ismost accurate, KaPM;av\\x88\\x85KP1M\\na\\x89P1\\x8a\\x87\\nKP2M\\na\\x89P2\\x8a\\x86=\\x85\\x89P1\\x8a\\x87\\x89P2\\x8a\\x86,where using [P1]eq'srather than [P1]0ismore accurate. Forthesecond\\nextreme, (KaPP=1)wenow treat allproteins asbound incomplex, creating anequilibrium\\nbetween proteins with twolipid binding sites and themembrane. This gives us\\x89M\\x8aCoop\\neq,which\\nhasacubic root analytical solution (see S1Text section 2C). Lastly, weinterpolate between\\nthese twoextremes (both independent ofKaPP),using thefunction λ.The λfunction isthe\\nfraction ofproteins that arebound tooneanother (based onP1+P2⇋P 1P2with Kaeff)outofthe\\nmaximum possible,\\nl\\x881\\n2\\x85\\x89P1\\x8a0\\x87\\x89P2\\x8a0\\x87Keff\\x001\\na\\x00\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81 \\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\n\\x85\\x89P1\\x8a0\\x87\\x89P2\\x8a0\\x87Keff\\x001\\na\\x862\\x004\\x89P1\\x8a0\\x89P2\\x8a0q\\n\\x86=min\\x85\\x89P1\\x8a0;\\x89P2\\x8a0\\x86\\x858\\x86\\nThese three results thus complete theequilibrium theory forEqs3and 4.The function λ\\ndepends onKaeff,which canbecalculated based onplugging\\x89M\\x8a0\\neqinto Eq3.Allourtheoretical\\nresults shown arebased onthisdefinition. However, wenote that thefinal value ofKaeffdepen-\\ndent on λcanalso bethen fedback into thisEq8toself-consistently converge toanimproved\\nresult forKaeff.\\nAnimportant feature ofEq4isthat itproduces thecorrect limiting behavior as λgoes\\nfrom 0to1.Weemphasize that although theequation for\\x89M\\x8aCoop\\neqiscumbersome, simply\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 17/28\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 16}),\n",
       " Document(page_content='setting λ=0willalready give very good accuracy inreproducing theexact result, with no-\\nticeable errors only expected when both thelipid concentration issmall relative tothetotal\\nproteins and theKaPPislarge. Wenote that therelative error inthefullEq4isquite small,\\nalthough forlarge V/A ratios (larger than those observed physiologically), theerror grows and\\nproduces overestimates oftheoretical enhancement ratios relative tothenumerical solution\\n(Fig 3Aand 3B). Weprovide Matlab code (solveKaeff.m) that performs thiscomplete calcula-\\ntion foranymodel.\\nDefinition ofmicroscopic and macroscopic rates forsimulation\\nTosimulate thesystems ofordinary differential equations (ODEs) forFig1(S1E Fig,S1A Text\\nsection), weneed macroscopic rates, and tosimulate thesingle-particle reaction-diffusion sys-\\ntem (RD), weneed microscopic rates (also known asintrinsic rates intheSmoluchowski the-\\nory[43]) inboth 3Dand 2D.The macroscopic rates emerge based onthedynamics ofthe\\nmore detailed microscopic system, and cantherefore beconstructed tooptimally match the\\nkinetics oftheODE simulations totheRDsimulations. Wenote that these definitions arespe-\\ncific tothekinetics, astheequilibrium ofboth simulation approaches willbeidentical dueto\\ntheir matching equilibrium constants.\\nThe ODE simulations donotaccount forspace orexplicit diffusion. Here, wedefine their rates\\ntoimplicitly account forchanges todiffusion andthus best match theRDsimulation kinetics.\\nThat way, discrepancies between kinetics ofODE andRDresults canbeattributed toexplicit spa-\\ntialheterogeneity influencing thebinding interactions. Macroscopic association (on-) rates canbe\\ndefined in3Dfrom theintrinsic rateoftheSmoluchowski model viatherelation [44]:\\nk3D\\non\\x881\\nk3D\\na\\x871\\n4psD3D\\ntot\\x12 \\x13\\x001\\n; \\x859\\x86\\nwhere kaistheintrinsic association ratethatcaptures thebarrier tocomplex formation forspecies\\nincontact atbinding radiusσ,andDtotisthesum ofboth species\\' diffusion constants. The macro-\\nscopic off-rate canbedefined inalldimensions via\\nkoff\\x88kon=Ka: \\x8510\\x86\\nThe intrinsic dissociation ratekbisdefined viathecorresponding equation, kb=ka/Ka,with all\\noff-rates having thesame units inalldimensions ofs-1.In2D,there isnosingle macroscopic rate\\nconstant independent ofthesystem sizeorconcentrations [20]. However, onecandefine amacro-\\nscopic 2Drate, built ontheory from Szabo etal[45], thatprovides optimal agreement with the\\ncorresponding spatial reaction-diffusion simulations via[20]:\\nk2D\\non\\x881\\nk2D\\na\\x871\\n8pD2D\\ntot4log\\x85b\\x85r\\x86=s\\x86\\n\\x851\\x00s2=b\\x85r\\x862\\x862\\x002\\n\\x851\\x00s2=b\\x85r\\x862\\x86\\x001\" #  !\\x001\\n\\x8511\\x86\\nwhere\\nb\\x85r\\x86\\x882\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81\\x81 \\x81\\nA=\\x85pmax\\x85NP1;NP2\\x86\\x87s2\\x86q\\n\\x8512\\x86\\nisalength scale thatisdefined based onthemore concentrated ofthereacting species P1orP2in\\nthesurface areaA.\\nThe important interpretation ofEqs9and 11isthat, unless kaislarge, even substantial (fac-\\ntorof10ormore) changes tothediffusion constant willhave arelatively small impact onthe\\nmacroscopic rate. Itisnotuntil macroscopic rates reach values of~106-107M-1s-1that they\\nbecome strongly diffusion influenced and thus sensitive tochanges indiffusion.\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 18/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 17}),\n",
       " Document(page_content='Our 2Dintrinsic rates aredefined relative toour3Drates via\\nka2D\\x88ka3D=\\x852s\\x86; \\x8513\\x86\\nand unbinding rates\\nkb2D\\x88kb3D; \\x8514\\x86\\nwhich produces theequilibrium relation defined inthemain text, Ka2D=Ka3D/(2σ). We\\nassume here that thedissociation rates arethesame from 3Dto2D.Itistheassociation rates\\nthat capture twospecies finding oneanother inaspecific spatial dimension. This definition of\\nEq13also canbeshown topreserve thereactivity ofthebinding interaction intheSmolu-\\nchowski model from 3Dto2D,independent ofchanges todiffusion (S1Text section 4A). For\\nthemacroscopic 2Drates, kon2D,weused Eq13inEq11,which allows ustocapture effects of\\ndiffusion towards timescales ofbinding inkon2D,asD2Dis~100 times lower than D3D.Transi-\\ntioning from solution tothemembrane viabinding lipid orprotein involves a3Dsearch, and\\nthus uses thecorresponding 3Drates. SeeS1Text section 3Bforfurther discussion.\\nUltimately, theresults ofKaeffareonly sensitive toequilibrium constants such asKa2Dand\\ntherefore thesizeofσ,rather than sizes ofrelative rates. This length scaleσencodes thermody-\\nnamic properties ofthemolecules involved inthebinding reaction andisofthenanometer\\nrange [22]. Ingeneral, thevalue ofσtherefore depends ontheproteins involved, butσ(or\\nKa2D),isalmost never measured. Weextractσ~7nm (from V/A =6.7μm andKaeff/KaPP\\x19500)\\nintheexperimental measurement of2Dbinding between calmodulin andatarget peptide [21].\\nSmallerσvalues have been observed [26]. Forsimulations, wethus used either 1or10nm. We\\nused thesame value fortheprotein-protein (σPP)orprotein-lipid (sP1M,sP2M)2Dbinding inter-\\nactions, although onlyσPPappears inEq3.The sizeofthese values isconstrained toensure an\\nequilibrium steady-state isreached, andthesimplest solution hasthatsPP\\x88sP1M\\x88sP2M.\\nComputer simulation methods\\nNumerical solutions ofODEs. The majority ofoursimulation results (exceptions noted\\nbelow) come from numerically solving thesystem ofODEs describing thechange intime ofthe\\nconcentrations ofallprotein, lipid, andbound species (S1Fig) viaMathematica (Equations listed\\ninS1Text section 1A). The initial conditions hadallproteins andlipids unbound andallproteins\\ninsolution. Forallsimulations, ourdefault waskoffrates of1s-1.Then kon3Dwasdefined viaEq\\n10.Exceptions were forproteins with known rates, andforthefewsimulations where toprevent\\nkon3Dfrom exceeding thediffusion-limited value of4πσD tot,weused koff=4πσD tot/KaPP.Al-\\nthough theODEs donotusediffusion constants, wedidneed them todefine ka3D(Eq9),then\\nka2D(Eq13),then kon2D(Eq11).Weused D3D=50μm2/sandD2D=0.5μm2/sforeach species,\\nboth reasonable estimates fordiffusion insolution andlipid diffusion[12] [23]. Forequilibrium\\nmeasurements (Fig 2andFig3Aand3B)wealsosimply defined kon2D=kon3D/(2σ). Tocalculate\\nthepercentage ofproteins incomplex, weused%Complexation\\x88100\\x03\\x89Complex\\x8aeq\\nmin\\x85\\x89P 1\\x8a0;\\x89P2\\x8a0\\x86.\\nSimulations with scaffold proteins. Forthescaffold-mediated system (S7Fig), theaddition\\nofthescaffold protein (SP) with twobinding sites, oneforeach peripheral membrane protein,\\nmeant atotal of14species could beformed, producing alarger system ofODEs tosolve. The\\nODEs were solved with Mathematica. Both Kaeff,SPandKasol,SPwere extracted from simulations\\nforallsystems (S1Text sections 1Cand1D,S2Table), with andwithout membrane present,\\nrespectively. This allowed ustomeasure theenhancement inbinding duetolocalization, justas\\nforthepairs, even though here Kaisnotatrue equilibrium constant forcomplex formation.\\nRule-based stochastic simulations ofhigher order oligomers and clathrin lattice forma-\\ntion. Tostudy notonly dimerization orbinding mediated byasingle scaffold protein, but\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 19/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 18}),\n",
       " Document(page_content='binding ofcomponents into chain-forming oligomers orclathrin lattices, weperformed Gilles-\\npiesimulations[46] written inourlabusing arule-based implementation. Rule-based imple-\\nmentations[47] allow onetotrack formation oflarge multi-protein complexes including\\ndimers, trimers, n-order oligomers, etc., without having toenumerate allpossible complexes\\ninadvance, which isahuge challenge toencode inasystem ofODEs. These simulations lack\\nspatial orstructural detail, soalthough wecantrack complexes formed, wecannot visualize\\nassemblies orstructural features. Nonetheless, theresults correctly capture how binding rates,\\nconcentrations, and membrane localization control complex formation, and thus areauseful\\ninitial model approach toquantifying therole ofdimensionality reduction. Tostudy BAR\\ndomain proteins forming oligomers, theBAR proteins each contained their dimer forming\\ninteraction sites aswell asanadditional non-competing site, allowing oligomeric filaments to\\nform. Because oligomerization wasnotobserved insolution even at~100μM concentrations\\n[11], weassume weak oligomer contacts of500μM, which willproduce<5% ofproteins in\\nhigher order complexes insolution. Wecalculate %oligomerization asthenumber ofbound\\noligomer sites onthepartner atlower concentration, relative toitstotal concentration. Full\\nsimulation conditions areinS4Dataset.\\nTostudy clathrin polymerization, each trimer leg(one clathrin molecule hasthree trimer\\nlegs) wasable tobind toanytrimer legofanother clathrin molecule, and these interactions did\\nnotcompete with adaptor binding. The ability ofclathrin tointeract with other trimers was\\nassumed tobeindependent ofitsinteractions with theadaptor AP-2 and nocooperative bind-\\ningofclathrin wasincluded, tominimize thenumber ofadjustable parameters and consider\\nthesimplest model ofcage formation (S2Table, S8Fig). Clathrin polymerization wassimu-\\nlated fortheinvitro experimental conditions reported inKelly etal[1].Weextracted aV/A\\nratio of9.46μm and alipid concentration of54,668μm-2from thestudy.\\nSpatially resolved reaction-diffusion simulation details. Single particle reaction-diffu-\\nsion (RD) simulations were used tomeasure time-scales ofassembly formation (Fig 3E,S4\\nFig) inaway that explicitly captured thespatial distribution ofproteins and lipids and thedif-\\nfusion ofspecies tocontact. Weused theFree Propagator Reweighting (FPR) algorithm, an\\nefficient and highly accurate method forstudying reactions between diffusing species atspatial\\nand single molecule resolution both insolution [19] and onthemembrane [20]. Alllipids are\\ninitialized inthemembrane plane, which isthebottom plane ofthesimulation box, distributed\\nrandomly. Each protein isasphere, and binding toalipid (also asphere) does notprevent\\nbinding totheprotein partner, and vice versa. The simulation boxhasperiodic boundaries in\\nthexand ydimensions, and thezdimensions areboth reflective, with thelower zplane con-\\ntaining thereactive lipids. The equilibrium properties oftheRDsimulations agreed with the\\nODE simulations, because oftheconserved equilibrium constants (S4Fig). The time-depen-\\ndent properties oftheRDsimulations didnotdiffer significantly from theODEs (Fig 3E,S4\\nFig) duefirstly because wetook care inassigning corresponding macroscopic and microscopic\\nrate constants above. Secondly, thespatial dimensions oftheRDsystems wesimulated were\\nsmall enough (box of0.47x0.47x0.76μm) that diffusion toreach themembrane didnotslow\\ndown equilibration. Forboxsizes with larger distances toreach themembrane, however, the\\nRDequilibration time slows relative totheODEs duetothisspatial effect.\\nThe FPR code forperforming these RDsimulations isavailable fordownload from github.\\ncom/mjohn218/FPR_simulat or.\\nCollecting biochemical data,invivo geometry, and concentrations\\nInTable 1welistallthehuman and yeast proteins forwhich wewere able tocollect sufficient\\nbiochemical data onlipid and protein interactions. The 20lipid-binding yeast proteins were', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 19}),\n",
       " Document(page_content='InTable 1welistallthehuman and yeast proteins forwhich wewere able tocollect sufficient\\nbiochemical data onlipid and protein interactions. The 20lipid-binding yeast proteins were\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 20/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 19}),\n",
       " Document(page_content=\"retained from alarger listof139peripheral membrane proteins (PMP) identified from the\\nUniprot database ashaving lipid binding activity inyeast (S1Dataset). Between thissetof139\\nPMPs, wefound 396interactions viaBioGRID, however, only 17pairs (S1Table) involved\\npartners with known KaPM's.The 15human proteins studied areallinvolved inCME and their\\nbiochemical data (Table 1,S1Table, S3and S4Datasets) wascollected viaextensive literature\\ncuration. Tostudy scaffold-mediated interactions (S2Table), weidentified allpossible inter-\\nactions that involved anon-membrane binding protein that could simultaneously and non-\\ncompetitively bind totwoofourPMPs. Fortheyeast proteins, these interactions could be\\nidentified from themanually curated interface interaction network forCME proteins [48].\\nThere wasarelatively small number ofexamples where asingle scaffold protein wascapable of\\nbridging twoPMPs (S2Table). These interactions inhumans/yeast involved clathrin/clathrin,\\neps15/ede1, oritsn1/sla1.\\nInS3Table wecollected volume and surface areas forcells and organelles with justifications\\nprovided. Because thecytoplasmic volume typically constitutes 50±60% ofthetotal cellvolume\\ninmammalian cells, ourV/A ratios setthesolution volume as60% ofthetotal cellvolume for\\nallcelltypes. Lipid concentrations arecollected inS4Table. The concentrations ofspecific lip-\\nidsonspecific membranes have only been quantified inafewcases, such asPI(4,5)P 2having\\nanaverage concentration of2.5x104μm-2ontheplasma membrane inmouse fibroblasts [15].\\nWeused thisconcentration asagold standard, duetoitsrelative consistency across measure-\\nments [13, 15], and other phosphoinositide concentrations were quantified relative tothisone.\\nWecurated literature tocollect thenecessary copy numbers ofeach lipid inthecell, and their\\ndistributions across organelles. Lastly, protein concentrations were defined from copy num-\\nbers measured inyeast [49] and human cells [50] (Table 1).\\nSupporting information\\nS1Text. Sections 1±5: Extended methods and model descriptions.\\n(PDF)\\nS1Matlab program. Takes user inputs onsystem parameters and calculates Kaeffand equi-\\nlibrium concentrations ofallspecies.\\n(M)\\nS1Table. Pairwise protein-protein interactions (PPIs) and affinities.\\n(PDF)\\nS2Table. Scaffold-mediated PPIs and higher-order assemblies, with affinities.\\n(PDF)\\nS3Table. Volume and membrane surface area estimates fordifferent cells and organelles.\\n(PDF)\\nS4Table. Phosphoinositide (PtdInsP n)and phosphatidylserine concentrations across vari-\\nous organelles inamammalian and ayeast cell.\\n(PDF)\\nS1Dataset. Lipid binding affinities and literature foryeast peripheral membrane proteins.\\n(XLSX)\\nS2Dataset. Protein-protein interaction pairs studied here foryeast. Allsimulation inputs\\nand literature references.\\n(XLSX)\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 21/28\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 20}),\n",
       " Document(page_content='S3Dataset. Protein-protein interaction pairs studied here forclathrin-mediated endocyto-\\nsis.Allsimulation inputs and literature references.\\n(XLSX)\\nS4Dataset. Scaffold-mediated and oligomer forming interactions studied here. Allsimula-\\ntion inputs and literature references.\\n(XLSX)\\nS1Fig. The simple theory developed here quantifies bound protein complexes forallsys-\\ntems with great accuracy. a,b) The relative error ofthetheoretically predicted Kaeffvalues\\ncompared with theexact numerical result from simulation, with data corresponding tothe\\nresults ofFig2A1and 2C1,respectively. Asexpected, theerror (~10−8)isnegligible in(a)\\nunder theconditions ofexcess lipids simulated inFig2A1,as[M] eqisnearly exactly predicted\\nbyourapproximate theory. In(b),theerror increases now that lipids areoutnumbered. Error\\nishighest formoderate KaPPvalues, because here thepredicted value of[M] eqisfarthest from\\neither ofthelimiting (and exact) predictions of\\x89M\\x8a0\\neqor\\x89M\\x8aCoop\\neq.The error reduces tovalues of\\n10−3and 10−2near these limits. c,d) From Eq3Kaeff,wecandirectly calculate theconcentra-\\ntion ofbound protein-protein complexes, asKaeff=[Complex] eq/(([P 1]0-[Complex] eq)([P 2]0-\\n[Complex] eq)).Simulation (red) vstheory (black). c)V/A =0.76, KaPM=104M-1,[P1]0=\\n[P2]0=0.1μM, [M] 0=2.5104μm−2.(d)Same as(c)except [P1]0=[P2]0=2μMand [M] 0=103\\nμm−2e,f)The fraction ofthese complexes that arespecifically onthemembrane (Eq. S7). g)\\nNetwork ofreactions between allstates. States with black outline allcontain protein-protein\\ncomplexes. Reactions in2Dareingreen text. Protein-protein binding isotherwise innavy\\ntext, and protein-lipid binding inorange text.\\n(TIF)\\nS2Fig. Membrane localization ofproteins inavariety ofcelltypes will produce increased\\nprotein binding interactions. a)Proteins inthecytosol canlocalize tomembranes bybinding\\nspecific lipids (yellow). Peripheral membrane proteins that donotbind directly canbebridged\\nbyascaffold protein (green/gray) b)Wecollected thesolution volumes (V)and membrane\\nsurface areas (A)forboth plasma and organellar membranes inavariety ofcelltypes (S3\\nTable). Only when theV/A ratio drops below 2σ,where hereσissetto10nm, does themem-\\nbrane reduce binding relative tosolution (bottom black line). The only case found here isfor\\nproteins inside theyeast Mitochondria, which hasasmall volume butalarge surface area due\\ntothehighly invaginated structure ofthemembrane. The V/A ratio forasphere (V/A =R/3) is\\nshown forreference inthediagonal black line.\\n(TIF)\\nS3Fig. Role ofprotein concentration, KaPP,mutations, and lipid concentration in\\nenhancement and complex formation. a)Once theenhancement duetomembrane localiza-\\ntion isnear tothemaximum value, theaddition ofmore lipids changes thebinding equilib-\\nrium imperceptibly. Even arelatively lowconcentration oflipids isneeded totrigger the\\nmaximum binding interactions, particularly with strong KaPM.b)This critical lipid concentra-\\ntion, [M] c,beyond which nofurther changes areobserved inbinding isderived inS1Text\\n(simulation results arepoints, theory islines). Wedefine maximum binding aswithin εof\\nKaeff=γKaPP,with results here shown for ε=0.01. c)Protein interactions between proteins\\nwith weak solution binding (low KaPP)ord)lowprotein concentrations benefit more widely\\nfrom recruitment. This isbecause these systems willform minimal complexes insolution (with\\n(KaPP)-1>[P] 0/2,fewer than halfofproteins areincomplex). Increased concentrations onthe\\nmembrane canthen substantially increase complex formation. e)Similar toFig3D,membrane\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 22/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 21}),\n",
       " Document(page_content='localization canactasaswitch toturn onassembly from<50% to>50% (shaded areas) de-\\npending onKaPP.Here weused KaPM=104M-1and[P] 0=1μM. f)Enhancement increases with\\nsmallerσasisclear from Eq3.Ing)Weshow how mutations that would alter KaPM(initially set\\nhere to106M-1)toanew value, KaPM\\x03,would result inachange from KaefftoKaeff\\x03.Here weset\\nKaPP=106M-1and[P] 0=1μM. Forsystems with higher [M] 0,only significant (>factor of50)\\ndecreases inaffinity duetomutation affect theenhancement. h)ForEpsin andAP180, theeffect\\nofpHandmutations onlipid binding affinity have been measured experimentally[27]. Weillus-\\ntrate here that because these proteins target PI(4,5)P 2at[M] 0=2.5x104μm-2,these upto\\n10-fold changes inaffinity have relatively minor impact onenhancement.\\n(TIF)\\nS4Fig. Average time toreach equilibrium isshifted bymembrane localization. a)ODE\\nsimulations show how localization canproduce relative speed-ups and slow-downs toreach\\nequilibrium relative topure solution binding. The dashed lineisatheoretical maximum esti-\\nmated from comparing time-scales ofpure 2Dbinding topure 3Dbinding (Methods and\\nSupplementary Text). Values ofkoff=1s-1were used, and diffusion wasonly captured implic-\\nitlyinbinding rates, asODEs have nospatial resolution. Westarted with V=50μm3and A=\\n65.63μm2,and then kept thevolume constant and varied thearea. KaPP=106M-1,KaPM=\\n106M-1b)FortheCME binding pairs, many binding reactions areultimately slowed bymem-\\nbrane localization. Forhuman proteins, V=1200μm3and A=767μm2and foryeast proteins,\\nV=37.2μm3and A=75.8μm2(S3Table). c)The trend iseven more evident forscaffold-medi-\\nated interactions. Same interactions asFig4(S3and S4Datasets). d)Time-dependence ofthe\\nsimulations ofthemodel inFig1comparing ODES (black lines) with RDsimulations using\\nFPR [19] [20], averaged over 24trajectories (colors). Time-scales from ODEs aresimilar toRD\\nmethods despite lacking explicit diffusion because ourdefinitions ofmacroscopic rates implic-\\nitlyaccount fordiffusion [20]. KaPP=107M-1koff=1s-1,KaPM=2x106M-1,[P1]0=[P2]0=\\n1μM, [M] 0=17000μm-2.FortheODE, V=50μm3and A=65.63μm2,and fortheRD, weused\\naboxsizeof0.467x0.467x0.762μm, producing thesame V/A ratio butinasmaller Volume.\\nForlarge systems, theRDsimulations willbeslower toreach equilibrium duetothetime\\nneeded todiffuse tothemembrane. Equilibrium values ofallspecies forthissystem arecol-\\nlected inthelower Table. e)Inpurely 2Dsimulations, wealso verify that theODE (black lines)\\nand RDsimulations (colors) give thesame equilibrium, asexpected. The surface area wasset\\ntoof0.467x0.467μm fortheRDsimulations, and theequivalent area (0.218μm2)fortheODEs.\\nTime-dependence isalso similar toreach that equilibrium. These results areused todefine the\\naverage time-scales toequilibrate inpart (f)[P1]0=[P2]0=458.92μm-2koff=1s-1f)Pure 2D\\nbinding (green-RD orblue-ODE) isgenerally faster than 3D(red) despite slow-downs (factor\\nof100here) indiffusion. Only forvery strong (diffusion-limited) binding reactions does the\\nimpact ofthediffusional search make adominant impact onbinding. V/A =0.762, and same\\ncopy numbers in3Dand 2D,with solution concentrations of[P1]0=[P2]0=1μM. Data on2D\\ntimescales isdefined from results ofpart (e).ODE system sizes and RDboxsizes arethesame\\nasforpart (d-e). Theory inblack dashed (3D) and gray (2D). (g)Equilibrium values ofallspe-\\nciesforthesystem studied inpart (d).Forthestochastic FPR simulations, standard deviations\\nfrom 24trajectories areshown. Theory values arederived from Eq3,then S1Text Section 2A.\\n(TIF)\\nS5Fig. CME interactions assuming weaker 2Dbinding orfewer lipids have lower butstill\\nsignificant enhancements. a)Ifweincreaseσfrom 1nm to10nm, themaximal enhancement\\ndecreases byafactor of10.b)Ifwereduce thelipid concentration byafactor of10(withσ=\\n1nm), enhancement again decreases relative toFig4.Now weareintheregime where lipids\\nonly slightly outnumber proteins. c)Complexation forpairs of(a)isstillquite large, because of\\nMembrane localizatio ncantrigger protein assembly', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 22}),\n",
       " Document(page_content='only slightly outnumber proteins. c)Complexation forpairs of(a)isstillquite large, because of\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 23/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 22}),\n",
       " Document(page_content=\"theoverall high enhancement. Gray bars indicate complex formation without membrane pres-\\nent,colors arewith membrane present, matching thelegend in(a).d)Complexation forpairs\\nin(b)isnow much more sensitive toKaPM.Notably, when AP-2 binds cargo (light green rela-\\ntivetodark green bars), KaPMis40times higher. The consequence ofthisstronger KaPMin(c)\\nismarginal, butwith limited lipids in(d),itdrives significantly larger increases incomplex for-\\nmation. Allresults inS3Dataset.\\n(TIF)\\nS6Fig. Peripheral membrane proteins can bestabilized onthemembrane viatheir pro-\\ntein-protein interactions. a)Wecompare FCHo1's localization tothemembrane byitself\\n(gray bars) orwith help from protein-protein interactions (colors). Weconsider arange of\\npossible values forKaPMand twodifferent KaPPforbinding toAP-2 toshow how theability to\\nbind other proteins willhelp stabilize thelipid binding FCHo1 onmembranes, where weesti-\\nmated [M] 0as25000μm-2.b)The same effect ispossible iftheprotein pairs canbind through\\nmultiple domains. Because AP-2 canuseboth its αand βappendages tobind epsin, itcan\\nform more complexes and stay tethered more strongly tothemembrane. Results here used 10\\ntimes lessPI(4,5)P 2.Other model inputs arebased oninvivo measurements and arecollected\\ninS3Dataset.\\n(TIF)\\nS7Fig. Model ofscaffold-mediated interactions with membrane localization. Weshow all\\nthepossible interactions forasystem with three cytosolic proteins (P3,P4,S)and amembrane\\nlipid (M). The twoperipheral membrane proteins P3and P4donotdirectly bind oneanother,\\nbutboth canbind toascaffold protein S.The scaffold protein thus hastwobinding sites, one\\nforP3and oneforP4.Only P3and P4canbind thelipid, notS.a)Binding interactions occur-\\nring purely insolution (3D) areshown inthisbox. b)Alltheorange boxed interactions involve\\nthelocalization ofaprotein orprotein complex from solution tothemembrane viabinding a\\nlipid ormembrane localized protein. Hence these areall3Dinteractions. Inpanels c-fwe\\nshow allthe2Dinteractions that canthus exploit membrane localization toenhance complex\\nformation.\\n(TIF)\\nS8Fig. Scaffold-mediated interactions and clathrin polymerization also benefit from local-\\nization onsurfaces. a)Forscaffold-mediated interactions (S7Fig), thetwoperipheral mem-\\nbrane proteins donotdirectly bind oneanother. Thus, noshift inlocalization willoccur unless\\nascaffold protein bridges them. Increasing concentration ofthescaffold protein increases\\nenhancements (see S1Text fordefinition ofKavalues from simulation). Inset shows how the\\nenhancement atafixed V/A increases with increasing scaffold toperipheral protein. Black\\ndashed lineismaximal enhancement ofKaeff,SP/KaSol,SP=γ.b)Wesimulated asystem ofcla-\\nthrin and theadaptor AP-2 tomimicinvitro experiment [1]using rule-based Gillespie simula-\\ntions (Methods). Weextracted aV/A ratio of9.46μm and alipid concentration of54,668μm-2\\nfrom thestudy, andused clathrin andAP-2 concentrations of0.4μM each. Stronger Ka(=Kd-1)\\nvalues fortheclathrin-clathrin (CC) interaction produce more polymerization, particularly\\nwith membrane localization included (green lines). Yellow pieispercent clathrin onthemem-\\nbrane forsimulations with AP-2. The enhancement isnotlimited bytheAP-2:PI(4,5)P 2interac-\\ntion, butrather because therecruitment ofclathrin tothemembrane requires AP-2, which is\\nonly at0.4μM (~25 times lower than lipid atthisV/A =9.46μm). c)Clathrin binds moderately\\ntoAP-2 (22μM). However, because clathrin hasthree legdomains that caneach bind AP-2,\\nonce onthemembrane, itwillquickly bind multiple AP-2s. d)Time-scales toreach equili-\\nbrium areslowed (relative topure solution inred) duetothetime needed tobind AP-2 tothe\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 24/28\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 23}),\n",
       " Document(page_content='membrane, andthen clathrin. Simulation inputs inS4Dataset.\\n(TIF)\\nS9Fig. Higher order assemblies studied viarule-based stochastic simulation. a)Retromer\\ncomponents VPS17 andVPS5 bind weakly toPI(3)P onendosomes. KaPPvalues arenotknown\\nforthese interactions, soweestimate arange (0.1±100μM: error bars). Since theprotein-lipid\\naffinity isonly known tobe>100μM (26), wecompare values of100μM andweaker binding of\\n300μM. Wecompare dimerization without membrane (gray bars) andwith (dark red). When\\nassisted byan(putative) interaction between SNX3 andVPS17, more complexation occurs of\\nthenow 3-protein complexes (light red). Because SNX3 binds strongly toPI(3)P, itwilldrive\\nmore complexation even without VPS5 (right bars). b)VPS17 ismore effectively recruited to\\ntheendosome when italso interacts with SNX3. Pink isVPS17 byitself, dark rediswith dimer\\nformation allowed, andlight rediswith thethird protein added (SNX3 onleft,VPS5 onright).\\nAlthough thisdirect interaction between SNX3 andVPS17 isnotphysiological, SNX3 does\\nbind thefull5-protein retromer complex [40]. These results illustrate how theretromer com-\\nplex could bemore strongly recruited toendosomes with thehelp ofSNX3. c)Wesimulated\\nBAR (SH3GL2:SH3GL2) andF-BAR (FCHo1:FCHo2) domain proteins forming both dimers\\nandhigher-order oligomers (Methods). Both pairs aregiven aweak oligomer binding strength\\nof500μM. Binding strength ofproteins tothemembrane iseither notknown orisreported at\\nwidely varying values, soweconsider arange ofvalues (0.1±100μM). Oligomerization insolu-\\ntion is<0.01%, butwith membrane (light blue) itisespecially prominent forthehomodimer\\nforming endophilin (SH3GL2). This isbecause SH3GL2 forms large oligomers (>20 proteins\\npercomplex) feeding back into stabilization atthemembrane. Incontrast, FCHo1 hasmuch\\nlower concentration than FCHo2, sooligomer contacts aremuch lesslikely toform large fila-\\nments. d)Dimers (nooligomer allowed) indark blue. Simulation inputs inS4Dataset.\\n(TIF)\\nS10 Fig. Membrane localization can enhance binding forpairs ofprotein binding partners\\nindiverse pathways targeting distinct organellar membranes. Yeast proteins that canalso\\nbind lipids including PI(4,5)P 2,PI(3)P, PI(4)P, and PI(3,5)P 2atdistinct organelles arereported\\ninTable 1,with interactions collected inS1Table and S2Dataset. These proteins areinvolved\\ninoxysterol binding (yellow), membrane remodeling (mauve, purple and turquoise) and vesi-\\ncleassembly onendosomes (green). Most oftheproteins exhibit significant enhancements,\\nexcept fortheendosome assembly proteins (VPS5, VPS17, SNX4, SNX41, ATG20) dueto\\ntheir lowaffinity forPI(3)P ontheendosomal membrane.\\n(TIF)\\nAcknowledgmen ts\\nThe research used theNSF XSEDE computational resources ofStampede and Comet, the\\nHomewood high-performance compute cluster atJHU and theMaryland Advanced Research\\nComputing Center clusters. Wethank Doug Barrick, Attila Szabo, and BillEaton forhelpful\\ncomments onthemanuscript.\\nAuthor Contributions\\nConceptualization: Margaret E.Johnson.\\nData curation: Osman N.Yogurtcu, Margaret E.Johnson.\\nFormal analysis: Osman N.Yogurtcu, Margaret E.Johnson.\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 25/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 24}),\n",
       " Document(page_content='Funding acquisition: Margaret E.Johnson.\\nInvestigation: Margaret E.Johnson.\\nMethodology: Margaret E.Johnson.\\nProject administration: Margaret E.Johnson.\\nResources: Margaret E.Johnson.\\nSoftware: Osman N.Yogurtcu, Margaret E.Johnson.\\nSupervision: Margaret E.Johnson.\\nValidation: Osman N.Yogurtcu, Margaret E.Johnson.\\nVisualization: Osman N.Yogurtcu, Margaret E.Johnson.\\nWriting ±original draft: Margaret E.Johnson.\\nWriting ±review &editing: Osman N.Yogurtcu, Margaret E.Johnson.\\nReferences\\n1. Kelly BT,Graham SC,Liska N,Dannhauser PN,Honing S,Ungewic kellEJ,etal.Clathrin adaptors .\\nAP2 controls clathrin polymeriza tionwith amembrane -activated switch. Science. 2014; 345\\n(6195):459 ±63. https:// doi.org/10.11 26/science. 1254836 PMID: 25061211 ;PubMed Central PMCID:\\nPMCPMC4 333214.\\n2. Dannhauser PN,Ungewicke llEJ.Reconsti tution ofclathrin-c oated budandvesicle formation with mini-\\nmalcomponen ts.NatCell Biol. 2012; 14(6):634± 9.https://doi.or g/10.103 8/ncb2478 PMID: 22522172.\\n3. Mishra SK,Keyel PA,Hawryluk MJ,Agostinelli NR, Watkins SC,Traub LM.Disabled- 2exhibits the\\nproperties ofacargo- selective endocytic clathrin adaptor .The EMBO journal. 2002; 21(18):491 5±26.\\nhttps://doi.or g/10.109 3/emboj/cdf4 87PMID: 12234931; PubMed Central PMCID: PMCPMC1 26284.\\n4. Christensen SM, TuHL,JunJE,Alvarez S,Triplet MG, Iwig JS,etal.One-way membrane trafficking of\\nSOS inreceptor-tri ggered Ras activation. NatStruct MolBiol. 2016; 23(9):838± 46.https://doi.or g/10.\\n1038/nsmb.3 275PMID: 27501536; PubMed Central PMCID: PMCPM C5016256 .\\n5. Adam G,Delbruck M.Reduction ofDimen sionality inBiological Diffusi onProcesses .Structur alChem is-\\ntryandMolecular Biology. San Francisc o:Freeman; 1968. p.198±215.\\n6. Berg HC, Purcell EM. Physics ofChemorec eption. Biophys J.1977; 20(2):193± 219. WOS:\\nA1977EA2 5300005. https:// doi.org/10.10 16/S0006-3 495(77)85 544-6 PMID: 911982\\n7. McCloskey MA, Poo MM. Rates ofmembrane -associat edreactions: reduction ofdimension ality revis-\\nited. The Journal ofcellbiology. 1986; 102(1):88± 96.PMID: 3001105; PubMed Central PMCID:\\nPMCPMC2 114064.\\n8. Haugh JM,Lauffenb urger DA.Physical modulation ofintracel lular signaling processe sbylocationa lreg-\\nulation. Biophys J.1997; 72(5):2014 ±31. https://doi.or g/10.101 6/S0006-34 95(97)788 46-5 PMID:\\n9129805; PubMed Central PMCID: PMCPM C1184397 .\\n9. Abel SM, Roose JP,Groves JT,Weiss A,Chakrabor tyAK.The membrane environmen tcanpromote or\\nsuppress bistabilit yincellsignaling network s.JPhys Chem B.2012; 116(11 ):3630±40. https://doi.or g/\\n10.1021/ jp2102385 PMID: 22332778; PubMed Central PMCID: PMCPMC3 321252.\\n10. Kholodenko BN,Hoek JB,Westerhoff HV.Why cytoplasm icsignalling protein sshould berecruite dto\\ncellmembrane s.Trends Cell Biol. 2000; 10(5):173± 8.PMID: 107545 59.\\n11. Youn JY,Friesen H,Kishimoto T,Henne WM, Kurat CF,YeW,etal.Dissecting BAR domain function in\\ntheyeast Amphip hysins Rvs161 andRvs167 during endocytosis. MolBiolCell. 2010; 21(17):305 4±69.\\nhttps://doi.or g/10.109 1/mbc.E10- 03-0181 PMID: 20610658 ;PubMed Central PMCID:\\nPMCPMC2 929998.\\n12. McLaughli nS,Wang J,Gambhir A,Murray D.PIP(2) andproteins :interactions ,organization, andinfor-\\nmation flow. Annu Rev Biophys Biomol Struct. 2002; 31:151± 75.https://doi.or g/10.114 6/annurev.\\nbiophys.31 .082901. 134259 PMID: 11988466.\\n13. Cambi A,Lidke DS.Cell Membrane Nanodomain s.Boca Raton: CRC Press; 2015.\\n14. Gallego O,Betts MJ,Gvozden ovic-Jeremi cJ,Maeda K,Matetzki C,Aguilar-Gur rieriC,etal.Asystem-\\naticscreen forprotein-lip idinteractions inSaccharom yces cerevisiae .MolSyst Biol. 2010;6. ARTN 430\\ndoi:10.1038 /msb.2010.87 .WOS:0 00285067900 009. PMID: 21119626\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 26/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 25}),\n",
       " Document(page_content='15. Yoon Y,LeePJ,Kurilova S,Cho W.Insituquantita tiveimaging ofcellular lipids using molecular sen-\\nsors. NatChem .2011; 3(11):868± 74.https://d oi.org/10.103 8/nchem.116 3PMID: 22024883; PubMed\\nCentral PMCID: PMCPM C3205457 .\\n16. ItohT,Erdmann KS,Roux A,Habermann B,Werner H,DeCamilli P.Dynamin andtheactin cytoskele-\\ntoncoopera tively regulate plasma membrane invagina tionbyBAR andF-BAR proteins .Dev Cell. 2005;\\n9(6):791±8 04.https://d oi.org/10.101 6/j.devcel.20 05.11.00 5PMID: 1632639 1.\\n17. Peter BJ,Kent HM, Mills IG,Vallis Y,Butler PJ,Evans PR,etal.BAR domains assensors ofmembrane\\ncurvature :theamphiph ysin BAR structure. Science. 2004; 303(5657) :495±9. https://doi. org/10.1126/\\nscience.1092 586PMID: 14645856.\\n18. Reider A,Barker SL,Mishra SK,ImYJ,Maldona do-Baez L,Hurley JH,etal.Syp1 isaconserved endo-\\ncytic adaptor thatcontains domains involved incargo selection andmemb rane tubulati on.The EMBO\\njournal. 2009; 28(20):310 3±16. https://doi.or g/10.1038/ emboj.2009. 248PMID: 19713939; PubMed\\nCentral PMCID: PMCPM C2771086 .\\n19. Johnson ME, Hummer G.Free-Prop agator Reweighting Integrat orforSingle-Pa rticle Dynami csin\\nReaction-D iffusion Models ofHeteroge neous Protein-Protei nInteractio nSystems. Phys Rev X.2014; 4\\n(3):031037 .Artn 031037 doi:10.1103 /PhysRev X.4.031037. WOS:0 00341266900 001. PMID:\\n26005592\\n20. Yogurtcu ON, Johnson ME. Theory ofbi-molecu larassociation dynamics in2Dforaccurate model and\\nexperime ntalparamete rization ofbinding rates. JChem Phys. 2015; 143:084 117. https://doi.or g/10.\\n1063/1.4 929390 PMID: 26328828\\n21. Ziemba BP,Knight JD,Falke JJ.Assem blyofmembrane -bound protein complexes :detection andanal-\\nysis bysingle molecule diffusion. Biochemi stry. 2012; 51(8):1638 ±47. https://doi.or g/10.1021/\\nbi201743a PMID: 22263647; PubMed Central PMCID: PMCPMC3 318961.\\n22. WuY,Vendo meJ,Shapiro L,Ben-Shau lA,Honig B.Transformi ngbinding affinities from three dimen-\\nsions totwowith applicati ontocadherin clustering .Nature. 2011; 475(7357) :510±3. https://doi.or g/10.\\n1038/nat ure10183 PMID: 21796210; PubMed Central PMCID: PMCPM C3167384 .\\n23. Knight JD,Lerner MG, Marcano- Velazquez JG,Pastor RW, Falke JJ.Single molecule diffusion ofmem-\\nbrane-bou ndproteins: window intolipid contacts andbilayer dynamic s.Biophys J.2010; 99(9):2879 ±\\n87.https://doi. org/10.1016/j .bpj.201 0.08.046 PMID: 210445 85;PubMed Central PMCID:\\nPMCPMC2 966005.\\n24. Jouvenet N,Simon SM, Bienias zPD.Imaging theinteracti onofHIV-1 genomes andGag during assem-\\nblyofindividu alviral particles. Proc Natl Acad SciUSA.2009; 106(45):19 114±9. https://doi. org/10.\\n1073/pnas .0907364106 PMID: 19861549; PubMed Central PMCID: PMCPM C2776408.\\n25. Matthews R,Likos CN. Influence ofFluctuating Membrane sonSelf-Asse mbly ofPatchy Colloids. Phys-\\nicalReview Letters. 2012; 109(17). ARTN 178302 doi:10.1103 /PhysRev Lett.109.1 78302.\\nWOS:000310 200100036. PMID: 232152 27\\n26. Weikl TR,HuJ,XuGK, Lipowsk yR.Binding equilibrium andkinetics ofmemb rane-anchor edreceptors\\nandligands incelladhesion: Insights from computation almodel systems andtheory. Cell Adh Migr.\\n2016; 10(5):576± 89.https://doi.or g/10.1080/ 19336918.201 6.11804 87PMID: 27294442; PubMed Cen-\\ntralPMCID: PMCPMC5 079412.\\n27. Hom RA,Vora M,Regner M,Subach OM, Cho W,Verkhush aVV,etal.pH-depe ndent binding ofthe\\nEpsin ENTH domain andtheAP180 ANTH domain toPI(4,5 )P2-contain ingbilayers. JMolBiol. 2007;\\n373(2):412 ±23. https:// doi.org/10.10 16/j.jmb.20 07.08.01 6PMID: 17825837; PubMed Central PMCID:\\nPMCPMC2 144917.\\n28. Motley A,Bright NA,Seaman MN, Robinson MS. Clathrin-m ediated endocytosis inAP-2-deplete dcells.\\nThe Journal ofcellbiology. 2003; 162(5):909 ±18. https://doi.or g/10.108 3/jcb.2003 05145 PMID:\\n12952941; PubMed Central PMCID: PMCPM C2172830 .\\n29. Jost M,Simpson F,Kavran JM,Lemmon MA, Schmid SL.Phosph atidylinosi tol-4,5-bisph osphate is\\nrequired forendocytic coated vesicle formation. Curr Biol. 1998; 8(25):1399 ±402. PMID: 9889104.', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 26}),\n",
       " Document(page_content='29. Jost M,Simpson F,Kavran JM,Lemmon MA, Schmid SL.Phosph atidylinosi tol-4,5-bisph osphate is\\nrequired forendocytic coated vesicle formation. Curr Biol. 1998; 8(25):1399 ±402. PMID: 9889104.\\n30. Honing S,Ricotta D,Krauss M,Spate K,Spolaore B,Motley A,etal.Phosph atidylinosit ol-(4,5)-bispho -\\nsphate regulates sorting signal recognition bytheclathrin-ass ociated adaptor complex AP2. Molecular\\ncell. 2005; 18(5):519± 31.https://doi.or g/10.101 6/j.molcel.20 05.04.01 9PMID: 15916959.\\n31. Kadlecov aZ,Spielma nSJ,Loerke D,Mohanakr ishnan A,Reed DK,Schmid SL.Regulation ofclathrin-\\nmediated endocytosis byhierarchical alloster icactivation ofAP2. The Journal ofcellbiology. 2017; 216\\n(1):167±79 .https://doi. org/10.1083/j cb.20160 8071 PMID: 280033 33;PubMed Central PMCID:\\nPMCPMC5 223608.\\n32. MaL,Umasanka rPK,Wrobel AG, Lymar A,McCoy AJ,Holkar SS,etal.Transien tFcho1/2E ps15/\\nRAP-2 Nanoclus tersPrime theAP-2 Clathrin Adaptor forCargo Binding. Dev Cell. 2016; 37(5):\\n428±43. https:// doi.org/10.10 16/j.devcel.2 016.05.0 03PMID: 272377 91;PubMed Central PMCID:\\nPMCPMC4 921775.\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 27/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 26}),\n",
       " Document(page_content='33. Schmid EM, Ford MGJ, Burtey A,Praefcke GJK, Peak-Che wSY,Mills IG,etal.Role oftheAP2 beta-\\nappendage hubinrecruit ingpartners forclathrin-coat edvesicle assembly. Plos Biol. 2006; 4(9):1532±\\n48.ARTN e262 doi:10.1371/jour nal.pbio. 0040262. WOS:000240 740900006. PMID: 169037 83\\n34. Cocucci E,Aguet F,Boulant S,Kirchh ausen T.The firstfiveseconds inthelifeofaclathrin-c oated pit.\\nCell. 2012; 150(3):495 ±507. https://do i.org/10.1016 /j.cell.201 2.05.047 PMID: 22863004; PubMed Cen-\\ntralPMCID: PMCPMC3 413093.\\n35. Wakeham DE,Chen CY,Greene B,Hwang PK,Brodsky FM. Clathrin self-assembly involves coordi-\\nnated weak interacti onsfavorabl eforcellular regulation. The EMBO journal. 2003; 22(19):498 0±90.\\nhttps://doi.or g/10.109 3/emboj/cdg 511PMID: 145172 37;PubMed Central PMCID: PMCPMC2 04494.\\n36. WuX,Zhao X,Baylor L,Kausha lS,Eisenbe rgE,Greene LE.Clathrin exchange during clathrin-m edi-\\nated endocytos is.The Journal ofcellbiology. 2001; 155(2):291 ±300. https://do i.org/10.1083 /jcb.\\n200104085 PMID: 11604424; PubMed Central PMCID: PMCPM C2198830.\\n37. Mettlen M,Loerke D,Yarar D,Danuser G,Schmid SL.Cargo- andadaptor-spec ificmechanism sregu-\\nlateclathrin-m ediated endocytosis .The Journal ofcellbiology. 2010; 188(6):919 ±33. https://doi.or g/10.\\n1083/jcb.2 00908078 PMID: 202313 86;PubMed Central PMCID: PMCPMC2 845073.\\n38. Holkar SS,Kamerkar SC,Pucadyil TJ.Spatial Control ofEpsin-induc edClathrin Assembly byMem-\\nbrane Curvature .The Journal ofbiological chemistry .2015; 290(23):14 267±76. https:// doi.org/10.10 74/\\njbc.M115.65 3394 PMID: 25837255; PubMed Central PMCID: PMCPM C4505496 .\\n39. YuJW, Lemmon MA. Allphox homology (PX) domains from Saccharo myces cerevisiae specifically rec-\\nognize phospha tidylinosito l3-phosphat e.The Journal ofbiologic alchemistry. 2001; 276(47):44 179±84.\\nhttps://doi.or g/10.107 4/jbc.M1088 11200 PMID: 11557775.\\n40. Harrison MS, Hung CS,LiuTT,Christiano R,Walther TC,Burd CG. Amechanis mforretromer endoso-\\nmalcoat complex assemb lywith cargo. Proc Natl Acad SciUSA.2014; 111(1):267 ±72. https:// doi.org/\\n10.1073/ pnas.1316482 111PMID: 24344282; PubMed Central PMCID: PMCPM C3890810 .\\n41. Loerke D,Mettlen M,Yarar D,Jaqaman K,Jaqaman H,Danuser G,etal.Cargo anddynamin regulate\\nclathrin-c oated pitmaturatio n.Plos Biol. 2009; 7(3):e57. https://d oi.org/10.137 1/journal.pb io.1000 057\\nPMID: 192967 20;PubMed Central PMCID: PMCPMC2 656549.\\n42. Ehrlich M,BollW,Van Oijen A,Hariharan R,Chandran K,Nibert ML,etal.Endocytosis byrandom initi-\\nation andstabilizatio nofclathrin -coated pits. Cell. 2004; 118(5):591 ±605. https://doi.or g/10.101 6/j.cell.\\n2004.08. 017PMID: 15339664.\\n43. Collins FC,Kimball GE. Diffusion-C ontrolled Reaction Rates. JColl SciImpUTok. 1949; 4(4):425±3 7.\\ndoi:10.1016 /0095-8522( 49)90023- 9.WOS:A1949 UJ15900007 .\\n44. Calef DF,Deutch JM.Diffusion-C ontrolled Reactions. Annu Rev Phys Chem. 1983; 34:493± 524. doi:\\n10.1146/ Annurev.Pc.34 .100183.00242 5.WOS:A 1983RN9470 0019.\\n45. Szabo A,Schulten K,Schulten Z.1stPassage Time Approach toDiffusio nControlled Reactions .J\\nChem Phys. 1980; 72(8):4350 ±7.doi:10.1063/1.439 715. WOS:A1980 JQ96800 014.\\n46. Gillespie DT.Exact stochastic simulation ofcoupled chemical reaction s.JPhys Chem. 1977; 81:2340±\\n61.\\n47. Sneddon MW, Faeder JR,Emonet T.Efficient modelin g,simulation andcoarse-grain ingofbiologic al\\ncomplexity with NFsim. Nature methods. 2011; 8(2):177±U 12.doi:10.1038 /nmeth.154 6.\\nWOS:000286 654600017. PMID: 211863 62\\n48. Johnson ME, Hummer G.Interface-res olved network ofprotein-pr otein interacti ons. Plos Comp utBiol.\\n2013; 9:e1003 065. https://doi.or g/10.137 1/journal.pcbi .100306 5PMID: 23696724.\\n49. Kulak NA,Pichler G,Paron I,Nagaraj N,Mann M.Minimal, encapsula tedproteo mic-sample processing\\napplied tocopy-numbe restimation ineukaryotic cells. NatMethods. 2014; 11(3):319± 24.https://doi.\\norg/10.1038/ nmeth.283 4PMID: 24487582.\\n50. Hein MY, Hubner NC, Poser I,Cox J,Nagaraj N,Toyoda Y,etal.Ahuman interactome inthree quanti-', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 27}),\n",
       " Document(page_content='org/10.1038/ nmeth.283 4PMID: 24487582.\\n50. Hein MY, Hubner NC, Poser I,Cox J,Nagaraj N,Toyoda Y,etal.Ahuman interactome inthree quanti-\\ntative dimension sorganized bystoichiometr iesandabunda nces. Cell. 2015; 163(3):712 ±23. https://doi.\\norg/10.1016/ j.cell.2015.09 .053 PMID: 26496610.\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 28/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 27}),\n",
       " Document(page_content='10. Materials and methods are available as supporting\\nmaterial on Science Online.\\n11. U.N. Population Division (UNPD), World Urbanization\\nProspects: The 2007 Revision Population Database\\n(UNPD, New York, 2007); available at http://esa.un.org/unup (accessed September 2007).\\n12. R. P. Cincotta, J. Wisnewski, R. Engelman, Nature 404,\\n990 (2000).\\n13. D. M. Olson et al .,Bioscience 51, 933 (2001).\\n14. A. de Sherbinin, Oryx 42, 26 (2008).\\n15. UNEP-WCMC, Prototype Nationally Designated Protected\\nAreas Database (WCMC/UNEP, Cambridge, 2003);\\navailable at www.unep-wcmc.org/protected_areas/data/nat.htm (accessed September 2007).\\n16. A. N. James, M. J. B. Green, J. R. Paine, Global Review of\\nProtected Area Budgets and Staff (WCMC, Cambridge,\\n1999).\\n17. A. Balmford, K. J. Gaston, S. Blyth, A. James, V. Kapos,\\nProc. Natl. Acad. Sci. U.S.A. 100, 1046 (2003).\\n18. A. James, K. J. Gaston, A. Balmford, Bioscience 51,4 3\\n(2001).\\n19. GEF, GEF Biodiversity Strategy in Action (GEF, Washington,\\nDC, 2006).20. GEF, The GEF Project Database (GEF, Washington,\\nDC, 2008); available at http://gefonline.org/home.cfm(accessed March 2008).\\n21. W. D. Newmark, J. L. Hough, Bioscience 50, 585\\n(2000).\\n22. G. W. Luck, B i o l .R e v .C a m b .P h i l o s .S o c . 82, 607\\n(2007).\\n23. K. K. Karanth, L. M. Curran, J. D. Reuning-Scherer,\\nBiol. Conserv. 128, 147 (2006).\\n24. J. S. Brashares, P. Arcese, M. K. Sam, Proc. R. Soc.\\nLondon B. Biol. Sci. 268, 2473 (2001).\\n25. A. T. Hudak, D. H. K. Fairbanks, B. H. Brockett,\\nAgric. Ecosyst. Environ. 101, 307 (2004).\\n26. R. DeFries, A. Hansen, A. C. Newton, M. C. Hansen,\\nEcol. Appl. 15, 19 (2005).\\n27. E. W. Sanderson et al .,Bioscience 52,8 9 1\\n(2002).\\n28. R. DeFries, A. Hansen, B. L. Turner, R. Reid, J. G. Liu,\\nEcol. Appl. 17, 1031 (2007).\\n29. Overlapping buffers in multiple PAs were combined for\\nanalysis, which decreased the sample of 306 PAs to 284buffer areas. Countries and ecoregions with one park\\nwere excluded from statistical analysis of median buffergrowth rates, although results did not vary with their\\ninclusion.\\n30. Supported by NSF OISE-0502340 and the J. S. McDonnell\\nFoundation (G.W.), the Hellman Fund ( J.S.B.) and the\\nUniversity of California at Berkeley. We thank A. Balmford,A. Bruner, P. Coppolillo, C. Golden, C. Kremen, R. Kuriyan,\\nJ. Scharlemann, A. R. E. Sinclair, C. Stoner, two anonymous\\nreviewers, and the Geospatial Imaging and Informatics Facility\\n(GIIF), Museum of Vertebrate Zoology (MVZ), and Department\\nof Environmental Science, Policy, and Management (ESPM)at the University of California at Berkeley. The authors\\ndeclare no competing financial interests.\\nSupporting Online Material\\nwww.sciencemag.org/cgi/content/full/321/5885/123/DC1\\nMaterials and MethodsSOM Text\\nFigs. S1 to S3\\nTable S1References\\n9 April 2008; accepted 4 June 2008\\n10.1126/science.1158900\\nRobust, Tunable Biological\\nOscillations from Interlinked Positive\\nand Negative Feedback Loops\\nTony Yu-Chen Tsai,1*Yoon Sup Choi,1,2*Wenzhe Ma,3,4Joseph R. Pomerening,5\\nChao Tang,3,4James E. Ferrell Jr.1†\\nA simple negative feedback loop of interacting genes or proteins has the potential to generate\\nsustained oscillations. However, many biological oscillators also have a positive feedback loop, raising\\nthe question of what advantages the extra loop imparts. Through computational studies, we show that\\nit is generally difficult to adjust a negative feedback oscillator ’s frequency without compromising its\\namplitude, whereas with positive-plus-negative feedback, one can achieve a widely tunable frequency\\nand near-constant amplitude. This tunability makes the latter design suitable for biological rhythms\\nlike heartbeats and cell cycles that need to provide a constant output over a range of frequencies.\\nPositive-plus-negative oscillators also appear to be more robust and easier to evolve, rationalizing\\nwhy they are found in contexts where an adjustable frequency is unimportant.\\nThe mammalian heart rate is normally es-tablished by the sino-atrial node. The node', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 0}),\n",
       " Document(page_content='why they are found in contexts where an adjustable frequency is unimportant.\\nThe mammalian heart rate is normally es-tablished by the sino-atrial node. The node\\ngenerates constant-amplitude action poten-\\ntials at a tunable frequency of ~50 to 150 action\\npotentials per minute, depending on the body ’s\\noxygen demands. The cell cycle oscillator may\\nalso require this combination of an adjustable\\nfrequency and invariant amplitude. The period of\\nthe cell cycle ranges from about 10 min in rapidly\\ndividing embryos to tens of hours in rapidly di-\\nviding somatic cells (and longer in slowly dividing\\nsomatic cells), but variations in the amplitude[the peak concentration of active cyclin-dependent\\nkinase-1 (CDK1)] of the oscillations seem neither\\nnecessary nor desirable.\\nTwo basic types of circuits have been pro-\\nposed for biological oscillators: (i) those that con-\\ntain both positive and negative feedback loops\\nand (ii) those containing only negative feedback\\n(Table 1) ( 1–6). Both the sino-atrial node oscil-\\nlator and the cell cycle oscillator fall into the\\npositive-plus-negative feedback class, suggesting\\nthat this design might be better suited for gen-\\nerating oscillations with a tunable frequency and\\nconstant amplitude.\\nWe tested this idea through computational\\nstudies, beginning with an ordinary differen-\\ntial equation model of CDK1 oscillations in the\\nXenopus embryonic cell cycle ( 7). The model\\nincludes a negative feedback loop [active CDK1\\nbrings about its inactivation through the anaphase-\\npromoting complex (APC)] and a pair of positive\\nfeedback loops (active CDK1 activates its acti-\\nvator Cdc25 and inactivates its inhibitor Wee1)\\n(Fig. 1A). We specified the strength of the posi-tive feedback through a parameter r,t h er a t i oo f\\nthe activities of Cdc25 and Wee1 in interphase\\nversus M phase. Because the rate of cyclin syn-\\nthesis determines the frequency of CDK1 oscil-\\nlations in Xenopus embryos ( 7,8), we varied the\\ncyclin synthesis rate constant k\\nsynthin the model\\nand determined how the amplitude and frequency\\nof the oscillations were affected by this variation.\\nIn the negative feedback –only version of the\\nmodel ( r= 1 in Fig. 1, B and C), a relatively small\\nrange of ksynthvalues yielded oscillations. Plotting\\nthe amplitude and frequency of the oscillations on\\na log-log plot yielded a tight, inverted U-shaped\\ncurve (Fig. 1B). The range of frequencies over\\nwhich the oscillator functioned was small (1.7-fold),\\nand even within this range, the frequency couldnot be adjusted without compromising the am-\\nplitude substantially.\\nAdding positive feedback markedly changed\\nthe amplitude/frequency relation (Fig. 1, B and\\nC). At a biologically realistic feedback strength\\nofr=1 0( 9–11), the oscillator functioned over a\\n4900-fold range of frequencies (Fig. 1B, green\\npoints). Over much of this range, the frequency\\nof the oscillator was linearly proportional to k\\nsynth,\\nand the amplitude was approximately constant\\n(Fig. 1, B and C). Thus, positive feedback pro-\\nvided a highly tunable frequency and robust\\namplitude.\\nSomething other than the cyclin synthesis rate\\nmay tune the frequencies of some cell cycles. We\\ntherefore asked whether the negative feedback –\\nonly oscillator might operate over a wider range\\nof frequencies if one of the model ’so t h e r2 0\\nparameters were varied. This was not the case;\\ninvariably, the oscillator operated over only a\\nnarrow frequency range. Of course if all of the\\nrate constants were multiplied by the same factor\\n(equivalent to scaling the units of time), the\\noscillator ’s frequency could be varied without\\nchanging the amplitude. However, this type of\\ncoordinated regulation is not relevant to any of\\nthe biological oscillators that we are familiar\\nwith (Table 1).1Department of Chemical and Systems Biology, Stanford Uni-\\nversity School of Medicine, Stanford, CA 94305 –5174, USA.\\n2School of Interdisciplinary Bioscience and Bioengineering,\\nPohang University of Science and Technology, Pohang, 790-', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 0}),\n",
       " Document(page_content='versity School of Medicine, Stanford, CA 94305 –5174, USA.\\n2School of Interdisciplinary Bioscience and Bioengineering,\\nPohang University of Science and Technology, Pohang, 790-\\n784, Republic of Korea.3Center for Theoretical Biology, Peking\\nUniversity, Beijing, 100871, China.4California Institute for\\nQuantitative Biosciences, University of California, San Francisco,CA 94143 –2540, USA.5Department of Biology, Indiana Uni-\\nversity, Bloomington, IN 47405, USA.\\n*These authors contributed equally to this work.\\n†To whom correspondence should be addressed. E-mail:\\njames.ferrell@stanford.edu\\n4 JULY 2008 VOL 321 SCIENCE www.sciencemag.org 126REPORTS\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 0}),\n",
       " Document(page_content='Both the tunable frequency and constant am-\\nplitude of the positive-plus-negative feedback cell\\ncycle model arise because the system behaves like\\na relaxation oscillator ( 12–15). Relaxation oscilla-\\ntors are built on a hysteretic switch, and experi-\\nmental studies have shown that in Xenopus extracts\\nthe response of the CDK1/Cdc25/Wee1/Myt1 posi-\\ntive feedback loop is hysteretic, resembling that\\nshown in Fig. 2A ( 16,17).To see how relaxation oscillations can arise\\nfrom a hysteretic switch and to see why this per-\\nmits a tunable period and constant amplitude, as-\\nsume that a cell cycle begins with no cyclin B and\\nno active CDK1 and that cyclin B synthesis is\\nslow relative to the phosphorylation and dephos-\\nphorylation reactions that allow the hysteretic\\nswitch to approach its steady state (Fig. 2A). As\\ncyclin B accumulates, the system moves up thelower branch of the stimulus/response curve, and\\nthe CDK1 activity slowly rises (Fig. 2, A and B,\\nsegments 0 and 1). Ultimately, the branch termi-\\nnates and the system switches ( “relaxes ”)t ot h e\\nother branch (Fig. 2, A and B, segment 2).\\nNow assume that at this higher, mitotic level\\nof CDK1 activity, the APC is turned on and the\\ncyclin B concentration begins to fall slowly. The\\nsystem progresses down the upper branch of\\nthe stimulus/response curve (Fig. 2, A and B,\\nsegment 3) until the branch terminates and the\\nsystem switches to the lower branch (Fig. 2, A\\nand B, segment 4). The APC turns back off, cyclin\\nB reaccumulates, and the cycle starts over. Thus,\\noscillations in this system essentially represent awalk around the hysteretic steady-state stimulus/\\nresponse loop. The frequency of the oscillator is\\ndetermined by how rapid the walk is, and the\\namplitude (the height of the loop) is constant.\\nIn reality, the rate of cyclin B destruction by\\nthe APC is not slow compared with the phos-\\nphorylation and dephosphorylation reactions ( 7).\\nThis fact is incorporated into the cell cycle model\\nexamined here, and it makes the orbits of the\\noscillator overshoot the hysteretic loop (Fig. 2C).\\nNevertheless, the model still behaves much like a\\nrelaxation oscillator, especially at low k\\nsynthvalues\\n( F i g .2 ,Ca n dD ) .\\nTo test the generality of the idea that positive\\nfeedback enables an oscillator to have a tunable\\nfrequency and constant amplitude, we examined\\nseveral other oscillator models, including five\\nnegative feedback –only models: (i) the Goodwin\\noscillator, a well-studied model relevant to cir-\\ncadian oscillations ( 18,19); (ii) the Repressilator,\\na transcriptional triple-negative feedback loop con-\\nstructed in Escherichia coli (20); (iii) the “Pentilator, ”\\na Repressilator with five (rather than three) re-\\npressors; (iv) the Metabolator ( 21), a synthetic\\nmetabolic oscillator; and (v) the Frzilator, a model\\nof the control of gliding motions in myxobacteria\\n(22). In four of the cases (Goodwin, Repressilator,\\nPentilator, and Metabolator), the amplitude/frequency\\ncurves were inverted U-shaped curves similar to\\nthat seen for the negative feedback –only cell cy-\\ncle model (Figs. 1B and 3A). In the case of theTable 1. Positive feedback loops in biological oscillators.\\nOscillator Period Positive feedback Refs.\\nSino-atrial pacemaker ~1 s Depolarization →Na+channel\\nactivation →depolarization(29)\\nCalcium spikes ~100 s Cytoplasmic Ca2+→PLC→IP3→\\ncytoplasmic Ca2+\\nCytoplasmic Ca2+→IP3R→\\ncytoplasmic Ca2+\\nCytoplasmic Ca2+→IP3R- |\\nER Ca2+-| SOC →cytoplasmic Ca2+(25,30,31)\\nMyxobacterial gliding ~10 min None known ( 22)\\nAnimal cell cycle\\n(Xenopus laevis embryos)~30 min Cdk1 →Cdc25 →Cdk1\\nCdk1 -| Wee1 -| Cdk1\\nCdk1 -| Myt1 -| Cdk1(32,33)\\nSomitogenesis ~30 min DeltaC →Notch→DeltaC ( 34)\\nYeast cell cycle\\n(S. cerevisiae )~2 hours CLN1,2 transcription →CDK1→\\nCLN1,2 transcription\\nCDK1 -| Sic1 -| CDK1\\nCDK1 -| Cdh1 -| CDK1(6,35–39)\\nNF-kB responses ~100 min None known ( 40,41)\\np53 responses ~100 min p53 →PTEN -| Akt →Mdm2 -| p53\\np53→p21 -| Cdk2 -| Rb -| Mdm2 -| p53(42,43)\\nAnimal cell cycle', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 1}),\n",
       " Document(page_content='CDK1 -| Cdh1 -| CDK1(6,35–39)\\nNF-kB responses ~100 min None known ( 40,41)\\np53 responses ~100 min p53 →PTEN -| Akt →Mdm2 -| p53\\np53→p21 -| Cdk2 -| Rb -| Mdm2 -| p53(42,43)\\nAnimal cell cycle\\n(somatic cells)~24 hours CDK2 -| Rb -| E2F →CDK2\\nCdk1→Cdc25 →Cdk1\\nCdk1 -| Wee1 -| Cdk1\\nCdk1 -| Myt1 -| Cdk1(44)\\nCircadian rhythm\\n(mammals)~24 hours BMAL1 →Rora→BMAL1 ( 45)\\nCircadian rhythm\\n(Drosophila )~24 hours CLK →PDP1→CLK ( 45)\\nCircadian rhythm\\n(fungi)~24 hours FRQ →WC-1→FRQ ( 46)\\nCircadian rhythm\\n(cyanobacteria)~24 hours KaiC-SP -| KaiA -| KaiC-SP ( 26)\\nFig. 1. Positive feedback provides an os-\\ncillator with a tunable frequency and nearly\\nconstant amplitude. ( A) Schematic view of the\\nXenopus embryonic cell cycle. ( B) Amplitude/\\nfrequency curves for various strengths ofpositive feedback ( r). The frequency of the\\noscillator was changed by varying the rate constant for cyclin B synthesis, k\\nsynth.(C) Frequency as a function of ksynth for various strengths of positive feedback.\\nwww.sciencemag.org SCIENCE VOL 321 4 JULY 2008 127REPORTS\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 1}),\n",
       " Document(page_content='Frzilator, the legs of the curve were truncated; the\\noscillator had a nonzero minimal amplitude (Fig.\\n3A). For all five of the negative feedback –only\\nmodels, the oscillators functioned over only a\\nnarrow range of frequencies (Fig. 3A).\\nWe also examined four positive-plus-negative\\nfeedback oscillators: (i) the van der Pol oscillator,\\ninspired by studies of vacuum tubes ( 12); (ii) the\\nFitzhugh-Nagumo model of propagating action\\npotentials ( 23,24); (iii) the Meyer-Stryer model of\\ncalcium oscillations ( 25) ;a n d( i v )am o d e lo fc i r -\\ncadian oscillations in the cyanobacterial KaiA/B/C\\nsystem ( 26–28). In each case, we obtained a flat,\\nwide amplitude/frequency curve (Fig. 3B). Thus,\\na tunable frequency plus constant amplitude canbe obtained from many different positive-plus-\\nnegative feedback models; this feature is not peculiar\\nto one particular topology or parameterization.\\nThese findings rationalize why the positive-\\nplus-negative feedback design might have been\\nselected through evolution in cases where a tun-\\nable frequency and constant amplitude are impor-\\ntant, such as heartbeats and cell cycles. However,\\nit is not clear that an adjustable frequency would\\nbe advantageous for circadian oscillations, be-\\ncause frequency is fixed at one cycle per day.\\nNevertheless, the cyanobacterial circadian oscil-\\nlator appears to rely on positive feedback ( 26),\\nand positive feedback loops have been postulated\\nfor other circadian oscillators as well (Table 1).\\nThis raises the question of whether the positive-\\nplus-negative feedback design might offer addi-\\ntional advantages.\\nOne possibility is that the positive-plus-\\nnegative feedback design permits oscillations\\nover a wider range of enzyme concentrations and\\nkinetic constant values, making the oscillator\\neasier to evolve and more robust to variations in\\nits imperfect components. We tested this idea\\nthrough a Monte Carlo approach. We formulated\\nthree simple oscillator models: (i) a three-variable\\ntriple negative feedback loop with no additional\\nfeedback (Fig. 4A), (ii) one with added positive\\nfeedback (Fig. 4B), or (iii) one with added negative\\nfeedback (Fig. 4C). We generated random pa-\\nrameter sets for the models and then for each set\\ndetermined whether the model produced limit cycle\\noscillations. We continued generating parameter setsuntil we had amassed 500 that gave oscillations.\\nFor the negative feedback –only model, 500\\nout of 138,785 parameter sets (0.36%) yielded\\noscillations (Fig. 4D). For the positive-plus-\\nnegative feedback model, oscillatory parameter\\nsets were found at a higher rate: 500 out of 23,848\\nparameter sets (2.1%) if we assumed weak posi-\\ntive feedback and 500 out of 9854 sets (5.1%) for\\nstrong positive feedback (Fig. 4D). The negative-\\nplus-negative feedback model yielded oscillations\\nat a lower rate than even the negative feedback –\\nonly model: 500 out of 264,672 parameter sets\\n(0.19%) for the weaker feedback strength and 500\\nout of 583,263 (0.086%) for the stronger feed-\\nback. This is probably because the short negative\\nfeedback loop stabilizes the output of A,m a k i n g\\nit difficult for changes in C’s activity to be propa-gated onward. Thus, the positive-plus-negative\\nfeedback design was substantially more robust, by\\nthis measure, than either the negative feedback –\\nonly model or the negative-plus-negative feedback\\nmodel.\\nThe random parameter sets also provided a\\nfurther test of the hypothesis that the positive-\\nplus-negative design allows for a tunable frequency.\\nFor each oscillatory set, we varied one parameter\\n(k\\n3) and calculated amplitude/frequency curves\\nand operational frequency ranges. For the nega-\\ntive feedback –only and the negative-plus-negativefeedback models, all of the oscillatory parameter\\nsets yielded narrow, inverted U-shaped amplitude/\\nfrequency curves with small operational frequencyranges (Fig. 4, E and F, and fig. S1). In contrast,\\nmany of the amplitude/frequency curves for the\\npositive-plus-negative feedback model were flat', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 2}),\n",
       " Document(page_content='frequency curves with small operational frequencyranges (Fig. 4, E and F, and fig. S1). In contrast,\\nmany of the amplitude/frequency curves for the\\npositive-plus-negative feedback model were flat\\nand wide, with large operational frequency ranges\\n(Fig. 4, E and F). Thus, the positive-plus-negative\\ndesign provided the possibility of a tunable fre-\\nquency and near-constant amplitude.\\nThe frequent presence of positive feedback\\nloops in natural biological oscillators suggests\\nFig. 2. From a hysteretic switch to a relaxation oscillator. ( A) Hysteretic steady-state response of CDK1 to\\ncyclin B, on the basis of previous experimental studies ( 16,17). (B) CDK1 activation and inactivation in\\nthe limit of slow cyclin B synthesis and degradation. ( CandD) Cell cycle model run with biologically\\nrealistic parameters, showing a looser relation between the oscillations and the hysteretic steady-stateresponse.\\nFig. 3. Amplitude/frequency curves for various legacy oscillators. ( A) Negative feedback –only models.\\n(B) Positive-plus-negative feedback models.\\n4 JULY 2008 VOL 321 SCIENCE www.sciencemag.org 128REPORTS\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 2}),\n",
       " Document(page_content=\"that this type of circuit possesses some perform-\\nance advantages over simple negative feedback\\nloops. The present work demonstrates two such\\nadvantages: (i) the ability to tune the oscillator ’s\\nfrequency without changing its amplitude and (ii)\\na greater robustness and reliability.References and Notes\\n1. C. D. Thron, Biophys. Chem. 57, 239 (1996).\\n2. A. Goldbeter et al .,Chaos 11, 247 (2001).\\n3. J. J. Tyson, K. C. Chen, B. Novak, Curr. Opin. Cell Biol. 15,\\n221 (2003).\\n4. P. Smolen, D. A. Baxter, J. H. Byrne, Am. J. Physiol. 274,\\nC531 (1998).5. A. Goldbeter, Nature 420, 238 (2002).\\n6. F. R. Cross, Dev. Cell 4, 741 (2003).\\n7. J. R. Pomerening, S. Y. Kim, J. E. Ferrell Jr., Cell122,\\n565 (2005).\\n8. R. S. Hartley, R. E. Rempel, J. L. Maller, Dev. Biol. 173,\\n408 (1996).\\n9. A. Kumagai, W. G. Dunphy, Cell70, 139 (1992).\\n10. P. R. Mueller, T. R. Coleman, W. G. Dunphy, Mol. Biol.\\nCell6, 119 (1995).\\n11. S. Y. Kim, E. J. Song, K. J. Lee, J. E. Ferrell Jr., Mol. Cell.\\nBiol. 25, 10580 (2005).\\n12. B. van der Pol, J. van der Mark, Philos. Mag. 6(suppl.),\\n763 (1928).\\n13. H. S. Hahn, A. Nitzan, P. Ortoleva, J. Ross, Proc. Natl.\\nAcad. Sci. U.S.A. 71, 4067 (1974).\\n14. C. R. Nave, Hyperphysics (1995), (http://hyperphysics.\\nphy-astr.gsu.edu/Hbase/electronic/relaxo.html).\\n15. S. H. Strogatz, Nonlinear Dynamics and Chaos: With\\nApplications to Physics, Biology, Chemistry, and\\nEngineering (Westview, Cambridge MA, 1994).\\n16. W. Sha et al.,Proc. Natl. Acad. Sci. U.S.A. 100, 975 (2003).\\n17. J. R. Pomerening, E. D. Sontag, J. E. Ferrell Jr., Nat. Cell\\nBiol. 5, 346 (2003).\\n18. B. C. Goodwin, Ed. Oscillatory Behavior in Enzymatic\\nControl Processes , vol. 3 (Permagon, Oxford, 1965),\\npp. 425 –438.\\n19. P. Ruoff, S. Mohsenzadeh, L. Rensing, Naturwissenschaften\\n83, 514 (1996).\\n20. M. B. Elowitz, S. Leibler, Nature 403, 335 (2000).\\n21. E. Fung et al .,Nature 435, 118 (2005).\\n22. O. A. Igoshin, A. Goldbeter, D. Kaiser, G. Oster, Proc.\\nNatl. Acad. Sci. U.S.A. 101, 15760 (2004).\\n23. R. FitzHugh, Biophys. J. 1, 445 (1961).\\n24. J. Nagumo, S. Arimoto, S. Yoshizawa, Proc. IRE 50, 2061\\n(1964).\\n25. T. Meyer, L. Stryer, Proc. Natl. Acad. Sci. U.S.A. 85, 5051\\n(1988).\\n26. M. J. Rust, J. S. Markson, W. S. Lane, D. S. Fisher,\\nE. K. O'Shea, Science 318, 809 (2007); published\\nonline 4 October 2007, 10.1126/science.1148596.\\n27. M. Nakajima et al .,Science 308, 414 (2005).\\n28. T. Nishiwaki et al .,EMBO J. 26, 4029 (2007).\\n29. A. L. Hodgkin, A. F. Huxley, J. Physiol. 117, 500 (1952).\\n30. R. S. Lewis, Annu. Rev. Immunol. 19, 497 (2001).\\n31. M. J. Berridge, Novartis Found. Symp. 239, 52 (2001).\\n32. B. Novak, J. J. Tyson, J. Cell Sci. 106, 1153 (1993).\\n33. J. J. Tyson, B. Novak, J. Theor. Biol. 210, 249 (2001).\\n34. A. Mara, S. A. Holley, Trends Cell Biol. 17, 593 (2007).\\n35. K. Levine, A. H. Tinkelenberg, F. Cross, Prog. Cell Cycle\\nRes. 1, 101 (1995).\\n36. J. M. Bean, E. D. Siggia, F. R. Cross, Mol. Cell 21, 3 (2006).\\n37. D. Stuart, C. Wittenberg, Genes Dev. 9, 2780 (1995).\\n38. L. Dirick, K. Nasmyth, Nature 351, 754 (1991).\\n39. K. C. Chen et al .,Mol. Biol. Cell 15, 3841 (2004).\\n40. A. Hoffmann, A. Levchenko, M. L. Scott, D. Baltimore,\\nScience 298, 1241 (2002).\\n41. D. E. Nelson et al .,Science 306, 704 (2004).\\n42. R. Lev Bar-Or et al .,Proc. Natl. Acad. Sci. U.S.A. 97,\\n11250 (2000).\\n43. G. Lahav et al .,Nat. Genet. 36, 147 (2004).\\n44. L. A. Buttitta, B. A. Edgar, Curr. Opin. Cell Biol. 19, 697 (2007).\\n45. M. Gallego, D. M. Virshup, Nat. Rev. Mol. Cell Biol. 8,\\n139 (2007).\\n46. K. Lee, J. J. Loros, J. C. Dunlap, Science 289, 107 (2000).\\n47. We thank E. Sontag, A. Millar, and B. Novak for helpful\\ndiscussions; J. Hasty for communicating unpublished results;\\nand J. Ubersax and G. Anderson for comments on themanuscript. This work was supported by grants from NIH\\n(GM61726 and GM77544), by the Li Foundation, and by a\\nKorea Science and Engineering Foundation grant from the\\nKorean government (No. R15-2004-033-05002-0).\\nSupporting Online Material\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 3}),\n",
       " Document(page_content='(GM61726 and GM77544), by the Li Foundation, and by a\\nKorea Science and Engineering Foundation grant from the\\nKorean government (No. R15-2004-033-05002-0).\\nSupporting Online Material\\nwww.sciencemag.org/cgi/content/full/321/5885/126/DC1\\nSOM TextFig. S1\\nTable S1\\nReferences\\n25 February 2008; accepted 9 June 2008\\n10.1126/science.1156951\\nFig. 4. Randomly parameterized oscillator models. ( A) Negative feedback –only. A,B,a n d C, the fractions\\nof proteins A, B, and C that are active; K, median effective concentration values of the Hill functions; n, Hill\\ncoefficients; k,r a t ec o n s t a n t s .( B) Positive-plus-negative feedback. ( C) Negative-plus-negative feedback.\\n(D) Percentage of parameter sets that yielded limit cycle oscillations. For the positive-plus-negative and\\nnegative-plus-negative models, we looked at two ranges of feedback strengths: (i) k7=0t o1 0 0( w e a k )\\nand (ii) k7= 500 to 600 (strong). ( E) Operational frequency ranges for the oscillators. Each point\\nrepresents freq max/freq minfor one of the 2500 parameter sets that produced oscillations, with k3as the\\nbifurcation parameter. Mean operational frequency ranges were 1.6, 370, 63, 1.6, and 1.6. Medians were1.6, 2.2, 3.3, 1.6, and 1.6. ( F) Amplitude/frequency curves for the randomly parameterized models. We\\nshow 300 out of 500 curves for the negative feedback –only model (red) and the positive-plus-negative\\nfeedback model with weak (blue) or strong (green) positive feedback. Curves for the negative-plus-\\nnegative feedback model are shown in fig. S1.\\nwww.sciencemag.org SCIENCE VOL 321 4 JULY 2008\\n129REPORTS\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 3}),\n",
       " Document(page_content='Use of this article is subject to the Terms of service\\nScience (ISSN 1095-9203) is published by the American Association for the Advancement of Science. 1200 New York Avenue NW,\\nWashington, DC 20005. The title Science is a registered trademark of AAAS.\\nAmerican Association for the Advancement of ScienceRobust, Tunable Biological Oscillations from Interlinked Positive and Negative\\nFeedback Loops\\nTony Yu-Chen Tsai, Yoon Sup Choi, Wenzhe Ma, Joseph R. Pomerening, Chao Tang, and James E. Ferrell Jr.\\nScience, 321 (5885), • DOI: 10.1126/science.1156951\\nView the article online\\nhttps://www.science.org/doi/10.1126/science.1156951\\nPermissions\\nhttps://www.science.org/help/reprints-and-permissions\\nDownloaded from https://www.science.org at Johns Hopkins University on October 31, 2022', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/TunableOscillator.pdf', 'page': 4}),\n",
       " Document(page_content='Design Principles of Biochemical Oscillators\\nBéla Novak* and John J. Tyson#\\n*Oxford Centre for Integrative Systems Biology, Department of Biochemistry, University of Oxford,\\nSouth Parks Road, Oxford OX1 3QU, UK\\n#Department of Biological Sciences, Virginia Polytechnic Institute & State University, Blacksburg\\nVA 24061, USA\\nAbstract\\nCellular rhythms are generated by complex interactions among genes, proteins and metabolites.\\nThey are used to control every aspect of cell physiology from signaling, motility and development\\nto growth, division and death. By considering specific examples of oscillatory processes, we pick\\nout three general requirements for biochemical oscillations: delayed negative feedback, sufficient\\n‘nonlinearity’ of the reaction kinetics, and proper balancing of the time-scales of opposing\\nchemical reactions. Positive feedback is one mechanism to delay the negative feedback signal.\\nBiological oscillators can be classified according to the topology of the positive and negative\\nfeedback loops in the underlying regulatory mechanism.\\nBiochemical oscillations occur in many contexts (metabolism, signaling, development, etc.)\\nwhere they control important aspects of cell physiology, such as circadian rhythms, DNA\\nsynthesis and mitosis, and the development of somites in vertebrate embryos (see Table 1).\\nIn the 1950s and 60s, the first clear examples of biochemical oscillations (in metabolic\\nsystems) were recognized in glycolysis1, 2, in cyclic AMP production3, and the horseradish\\nperoxidase reaction4, 5. Soon after these discoveries, theoreticians were thinking about the\\ngeneral requirements for chemical oscillations and the specific mechanisms of these\\nexamples6, 7. After the molecular biology revolution of the 1980s, many new examples of\\noscillations in protein interaction networks and in gene regulatory networks came to light,\\nsuch as the PERIOD proteins in animal circadian control8, the CYCLIN proteins in\\neukaryotic cell cycle control9, 10, and the Repressilator11 in genetically engineered\\nbacteria.\\nUnderstanding the molecular basis of cellular oscillations is more than an exercise in\\nexperimental genetics and biochemistry. Oscillators have systems-level characteristics\\n(periodicity, robustness, entrainment) that transcend the properties of individual molecules\\nor reaction partners and involve the full topology of the reaction network. These properties\\ncan only be fully understood by viewing experimental data from a theoretical perspective,\\nby quantitative mathematical modeling of chemical oscillatory processes. These models\\naddress general concepts of dynamical systems, such as feedback, time delays, bistability\\nand hysteresis.\\nCorrespondence to B.N., bela.novak@bioch.ox.ac.uk, tyson@vt.edu.\\nSupplementary Material:\\n1. Computer Programs\\n2. Negative Feedback with Explicit Time Delay\\n3. Motif G\\n4. Motif H\\nNIH Public Access\\nAuthor Manuscript\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nPublished in final edited form as:\\nNat Rev Mol Cell Biol . 2008 December ; 9(12): 981–991. doi:10.1038/nrm2530.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 0}),\n",
       " Document(page_content=\"In this review, we present a series of examples of increasing complexity, which illustrate the\\nfollowing essential requirements for biochemical oscillators. (1) Negative feedback is\\nnecessary to carry the reaction network back to the ‘starting point’ of its oscillation. (2) The\\nnegative feedback signal must be sufficiently delayed in time so that the chemical reactions\\ndo not home in on a stable steady state. (3) The kinetic rate laws of the reaction mechanism\\nmust be sufficiently ‘nonlinear’ to destabilize the steady state. (4) The reactions that produce\\nand consume the interacting chemical species must occur on appropriate time scales that\\npermit the network to generate oscillations. Time-delay can be created by a physical\\nconstraint (for example, the minimal time necessary to carry out transcription and\\ntranslation, or the time needed to transport chemical species between cellular\\ncompartments), by a long chain of reaction intermediates (as in a metabolic pathway), or by\\ndynamical hysteresis (overshoot and undershoot, as consequences of positive feedback in\\nthe reaction mechanism).\\nTo keep the mathematics of oscillating chemical reactions to a minimum, we will\\ndemonstrate the design principles of biochemical oscillators by rate plots (how do reaction\\nrates depend on chemical concentrations?), signal-response curves (how do oscillations turn\\non and off in response to regulatory signals?) and ‘constraint’ diagrams (how are the kinetic\\nconstants of the reaction mechanism constrained by requirements for periodicity?). The\\nmathematical details are made available to interested readers as online supplementary\\ninformation. For further details on the principles underlying chemical and biochemical\\noscillations, we refer the interested reader to books12-15 and review articles2, 6, 16-18.\\nNegative feedback with time delay\\nIn order to lay bare the ‘design principles’ of biochemical oscillators, it makes no sense to\\nstart with a fully detailed model of a particular cellular rhythm, such as the cell cycle in\\nhuman fibroblasts, which is likely to be so overlain by subtle control signals that the\\nessential features of the oscillator are obscure. Rather, we start from a highly idealized\\nmodel of periodic protein synthesis that illustrates the basic requirements of biochemical\\noscillators in their pristine form.\\nWe have in mind a protein (like PER in the circadian control system of fruit flies19, 20) that\\nrepresses the transcription of its own gene (Fig. 1a). The details of this feedback repression\\nare not important at present. The time-rate of change of protein concentration, d Y/dt, is\\ngiven by a simple kinetic equation,\\n(1)\\nwhere the first term is the rate of synthesis of protein and the second is its rate of\\ndegradation. The synthesis rate is proportional to a ‘signal’ S (which might be the\\nconcentration of a transcription factor that up-regulates the gene) multiplied by a factor,\\n, expressing how gene transcription is down-regulated by Y. In this factor, Kd\\nis the dissociation constant for binding of Y to the up-stream regulatory sequence of the\\ngene, and p is an integer indicating whether Y binds to the DNA sequence as a monomer,\\ndimer, trimer, etc. The rate constant k1 is the rate of synthesis of Y (per unit signal strength)\\nwhen the concentration of Y is small and the gene is fully expressed. In the second term, E\\nis a protease that degrades Y ( ET is the total concentration of enzyme); its turnover rate is k2\\nand its Michaelis constant is Km. (Note: lower case k's are rate constants, unit = time-1;\\nupper case K's are dissociation constants, unit = concentration.)Novak and Tyson Page 2\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 1}),\n",
       " Document(page_content='In Fig. 1b we plot the rates of synthesis and degradation of the protein as functions of its\\nconcentration, Y. From the diagram it is clear that the protein concentration, if indeed it is\\ngoverned by the simple kinetic equation (1), will be drawn towards its steady-state value,\\nYo, without any oscillations, nor with any overshoots or undershoots. If ‘homeostasis’ is\\nwhat we desire, this is great, but if we are looking for ‘oscillations’ we need something\\nmore.\\nExplicit time delay\\nSuppose that the rate of protein synthesis at present (at time t) depends on the concentration\\nof protein at some time in the past (at time t-τ), where τ is the time delay required for\\ntranscription and translation. Then, the governing kinetic equation becomes\\n(2)\\nThis model of protein synthesis and degradation was first studied in detail by Mackey and\\nGlass in 197721. For a proper choice of rate constants and time delay, this equation exhibits\\nperiodic oscillations, as illustrated in Fig. 1c. The time delay causes the negative feedback\\ncontrol repeatedly to overshoot and undershoot the steady state (Fig. 1d). For details on how\\nto simulate Eq. (2) and all other models in this review, see supplementary information S1\\n(box).\\nBy ‘proper choice of rate constants…’ we mean that, in order for Y(t) to oscillate, the\\nkinetic parameters— S (signal strength), p (nonlinearity of feedback), Km (nonlinearity of the\\nremoval step), and τ (duration of time delay)—must satisfy specific constraints, illustrated\\nin Fig. 1e and f. For details on how these constraint curves are calculated, see supplementary\\ninformation S2 (box). The constraints can be summarized in three requirements. (1) The\\ntime delay, τ, must be sufficiently long. (For fixed values of p and Km, there is a minimum\\nvalue of τ —call it τmin—below which oscillations are impossible.) (2) The reaction rate\\nlaws must be sufficiently ‘nonlinear’. (Oscillations become easier—i.e., τmin gets smaller—\\nas either p or Kd/Km increases.) (3) The rates of opposing processes must be appropriately\\nbalanced.\\nTo understand the third requirement, we must look more closely at the axes in Figs. 1e and f.\\nThe value of ‘time delay’ plotted on the vertical axis is really a dimensionless combination\\nof parameters:\\nThe value of ‘Signal strength’ plotted on the horizontal axis is the dimensionless ratio\\nFor fixed values of p and Kd/Km, these ratios must lie above a specific curve plotted in the\\nfigures. For example, for p = 2 and Kd/Km = 10 (‘modest’ nonlinearity of the rate laws for\\nprotein synthesis and degradation), these ratios must (roughly speaking) satisfy theNovak and Tyson Page 3\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 2}),\n",
       " Document(page_content=\"inequalities: τ / Tdegr > 2 and Tdegr / Tsyn > 1 (from the lowest curve in Fig. 1f). Estimating\\nthe time delay for transcription and translation to be about 20 min, we predict that, in order\\nfor the negative feedback loop to oscillate, the time scale for protein degradation must be <\\n10 min and the time scale for protein synthesis must be even shorter. If these conditions are\\nsatisfied, then the period of oscillation is (again, roughly speaking) between 2× and 4× the\\ntime delay, i.e., approximately 40—80 min.\\nIn the remainder of this review, we intend to show that these four requirements (negative\\nfeedback, nonlinearity, time delay, and time-scale constraints) are quite generally true of all\\nbiochemical oscillators, provided the notion of ‘time delay’ is suitably generalized.\\nTime delay by a series of intermediates\\nOur simple model of negative feedback on gene expression, Eq. (2), exhibits sustained\\noscillations if there is a sufficiently long time delay between the action of the protein Y on\\nthe gene and the appearance of new protein molecules in the cytoplasm. We expressed this\\nrequirement as a discrete time delay, τ, in the kinetic equation. Maybe we can dispense with\\nτ if we include the dynamics of mRNA in our model (Fig. 2a). To this end, we write a pair\\nof kinetic equations for X = [mRNA], Y = [Protein]:\\n(3)\\nIn Fig. 2b we plot the ‘nullclines’ of this pair of nonlinear differential equations. The X-\\nnullcline (curve a in Fig. 2b) is the locus of points in the ( X,Y) plane where the rate of\\nmRNA synthesis is exactly balanced by the rate of mRNA degradation, that is, where\\n. Along the X-nullcline, d X/dt = 0 and trajectories move horizontally in the\\n(X,Y) plane because there is no change in the X direction but there may be change in the Y\\ndirection. The Y-nullcline (curve b) is the locus of points where the rate of protein synthesis\\nis balanced by its rate of degradation, that is, where \\n . Along the Y-nullcline,\\ndY/dt = 0 and trajectories move vertically in the ( X,Y) plane. Where the nullclines intersect\\n(the solid circle in Fig. 2b), the trajectory comes to rest at a steady state (both d X/dt = 0 and\\ndY/dt = 0). The sample trajectories in Fig. 2b (the dashed lines) spiral into the stable steady\\nstate. Although the system of reactions may exhibit damped oscillations on the way to the\\nsteady state, sustained oscillations in this simple gene regulatory circuit are impossible22.\\nHence, adding mRNA to the model does not do away with the requirement for an explicit\\ntime delay.\\nBefore giving up the idea of replacing the time delay by a series of intermediates in the\\nnegative feedback loop, let's recognize that (in eukaryotes) the mRNA and protein molecules\\nneed to be transported between nucleus and cytoplasm (Fig. 2c). Equation (3) is readily\\nexpanded to four variables: mRNA and protein in the nucleus ( Xn, Yn) and in the cytoplasm\\n(Xc, Yc). The four-variable negative feedback loop oscillates as naturally as a pendulum\\n(Fig. 2d)!\\nThe constraint diagrams in Fig. 2 underscore the conclusions of the previous section. Figure\\n2e shows that the kinetic rate laws must be sufficiently nonlinear ( p and/or Kd/Km large\\nenough). Figure 2f shows that the turnover rates for mRNA in the nucleus ( kdxn = 0.693/\\nhalf-life of X n) and for protein in the cytoplasm ( kdyc = 0.693/half-life of Y c) must beNovak and Tyson Page 4\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 3}),\n",
       " Document(page_content=\"properly balanced. Furthermore, if either kdxn → ∞ or kdyc → ∞, the negative feedback\\nloop reduces to three components, ( Xc, Yc, Yn) or ( Xn, Xc, Yn) respectively, and yet\\noscillations are still possible. Hence, we conclude that oscillations are impossible in a two-\\ncomponent negative feedback loop (Fig. 2b) but possible in a three-component negative\\nfeedback loop (Fig. 2f)22.\\nThis mechanism (negative feedback on gene expression with three or more components in\\nthe feedback loop), which we have used to illustrate the four basic principles of biochemical\\noscillations, was first put forward by Brian Goodwin23, 24 in mid-1960s as a model for\\nperiodic enzyme synthesis in bacteria. Our calculations in this section show that, with an\\neffective time delay (transcription + translation + reaction intermediates) of about 20 min,\\nthis model gives a period of about 1 h, which is close to the observed periods of such\\nrhythms25. This mechanism has also been a favorite model for circadian rhythms in flies\\nand mammals, governed by the PER protein, which moves into the nucleus and blocks\\nexpression of the PER gene26, 27. Of course, to get a period of 24 h, the time delay for the\\nfeedback signal must be considerably longer than the delay expected for transcription,\\ntranslation and nuclear transport. It is believed that PER undergoes slow post-translational\\nmodifications (phosphorylations) in the cytoplasm before it returns to the nucleus20.\\nThe possibility of sustained oscillations in a three-component negative feedback loop was\\nused by Elowitz and Leibler11 to design the ‘Repressilator’, a synthetic gene regulatory\\nnetwork in E. coli  consisting of three operons, each one expressing a protein that represses\\nthe next operon in the loop. The successful engineering of the Repressilator was a\\nfoundational triumph of the nascent field of ‘synthetic biology’ and a vindication of the\\ntheoretical ideas of Goodwin23, 24, Griffith22, Goldbeter26 and others.\\nTime delay by positive feedback\\nTime delay is a sort of memory: protein synthesis rate at the present time depends on protein\\nconcentration over some time in the past. Memory is a property of biochemical systems with\\nbistability: under identical chemical conditions, the system can be in either of two,\\nalternative, stable steady states28, 29. Which state the system is currently occupying\\ndepends on its recent history (a phenomenon called ‘hysteresis’). Hysteresis can prevent a\\nsystem with negative feedback from finding its homeostatic steady state. To see how this\\nhappens, we add positive feedback to our mRNA-protein example, Eq. (3). In particular, we\\nassume that protein Y, in addition to binding to its own gene regulatory site and down-\\nregulating its own expression, can bind to an allosteric site on protease E and inhibit E's\\nactivity (Fig. 3a). (Note: by inhibiting its own degradation, Y helps itself to accumulate.)\\nThe kinetic equations become:\\n(4)\\nIn these equations, kdy is the rate constant for an alternative pathway of protein degradation\\nand KI is a constant that characterizes the strength of the inhibition of enzyme E by its\\nsubstrate Y. In Eq. (4), as before, S is the concentration of a transcription factor that up-\\nregulates the expression of the mRNA encoding protein Y. We think of S as a signal that\\nmay induce sustained oscillations in protein level as a function of time.\\nIn Fig. 3b we draw the nullclines of Eqs. (4) in the same format as Fig. 2b. The effect of\\npositive feedback is to put a kink in the degradation curve (compare curves b in Figs. 2b and\\n3b), and the kink forces the dynamical system to overshoot and undershoot the steady stateNovak and Tyson Page 5\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 4}),\n",
       " Document(page_content='repeatedly (compare the dashed trajectories in Figs. 2b and 3b). The system executes\\nsustained oscillations (Fig. 3c), provided the signal strength, S, is within certain bounds (Fig.\\n3d).\\nThe constraint diagrams (Figs. 3e and f) show that these oscillations require positive\\nfeedback (they disappear if KI is too small or too large) and proper balancing of time scales:\\nmRNA must be sufficiently stable ( kdx not too large) and signal strength must lie within\\nstrict bounds ( S not too large and not too small).\\nThis mechanism (negative feedback on gene expression + inhibition of protein degradation)\\nhas been suggested by Tyson et al.30 as a possible source of circadian rhythms in the\\nreaction network governing expression of the PER gene in fruit flies. Their idea was that\\nPER protein may form dimers that are less prone to degradation by the protease (‘E’ in the\\nmodel is casein kinase, which phosphorylates PER and labels it for proteolysis). Other\\nexamples of this design for oscillations will be given after we expand our notion of reaction\\nmechanisms from gene regulation to metabolic control systems and protein interaction\\nnetworks.\\nBiochemical interaction networks\\nUsing a simple gene regulatory network (GRN) as an example, we have discovered two\\ndistinct mechanisms for generating oscillations: a negative feedback loop with at least three\\ncomponents, and a combination of short positive and negative feedback loops. We might\\ndepict these mechanisms with the following ‘regulatory motifs’:\\nwhere X → Y means ‘X activates Y’ and X ⊣Y means ‘X inhibits Y’. The general ideas of\\nthe previous section—positive and negative feedback, time delay, nonlinearity—are not\\nlimited to GRNs but apply equally well to metabolic control systems (MCSs) and to protein\\ninteraction networks (PINs). We would like to know whether there is a general theory of\\nregulatory motifs (like the two above) that classifies types of biochemical oscillators.\\nIn this review, we are not so much interested in the precise mechanism of particular\\nbiochemical oscillators as in the patterns of activation and inactivation (that is, the\\nregulatory motifs) that appear repeatedly in all known oscillators. How ‘activation’ and\\n‘inhibition’ is actually carried out, biochemically, varies considerably, depending on the\\ncontext. In GRNs, transcription factors bind to regulatory sequences upstream of genes and\\ncontrol whether the gene is transcribed to mRNA or not. In MCSs, metabolites bind to\\nenzymes and control how quickly or slowly the enzyme catalyzes a particular chemical\\nreaction. In this way, metabolite X can ‘activate’ metabolite Y either by activating the\\nenzyme that produces Y or by inhibiting the enzyme that consumes Y. In PINs, protein X\\nmight activate protein Y because X is a kinase phosphorylating Y, or because X is a binding\\npartner forming an active XY dimer. Protein X might, just as well, inactivate protein Y by\\nbinding to Y or by phosphorylating Y, or because X is a protease that degrades Y. Given all\\nthese possibilities, there are myriad ways to build a PIN or MCS to instantiate a particular\\nregulatory motif. Our previous examples, involving negative feedback on gene expression,\\nsuggest that, besides the regulatory motif itself (the pattern of positive and negativeNovak and Tyson Page 6\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 5}),\n",
       " Document(page_content='interactions), it is important that an oscillatory mechanism have sufficient ‘nonlinearity’ and\\nthat the rates of particular opposing reactions are properly balanced.\\nNonlinearities arise in biochemical reaction networks from many sources (Fig. 4). We have\\nalready used the example of a multimeric transcription factor binding to a genetic regulatory\\nsequence (Fig. 4a), for which the probability of binding is given by a nonlinear ‘Hill’\\nfunction31, \\n . Cooperative binding of substrates and modifiers to multi-subunit,\\nallosteric enzymes (Fig. 4b) also generates sigmoidal nonlinearities, adequately described by\\nHill functions or by more accurate kinetic rate laws31, 32. Reversible phosphorylation and\\ndephosphorylation of target molecules can create a sigmoidal signal-response curve, if the\\ninterconverting enzymes (kinase and phosphatase) have high affinity (low Km) for their\\nabundant substrates (zero-order ultrasensitivity33) or if the target molecule has multiple\\nphosphorylation sites (Fig. 4c)34. Our final example is a stoichiometric inhibitor that binds\\nto a regulatory protein (X) to form an inactive complex (Fig. 4d): as the total amount of X\\nincreases in response to signal (S), the active fraction of X shows a highly nonlinear signal-\\nresponse curve35. A high-affinity substrate can work as a stoichiometric inhibitor of its\\nenzyme, making enzyme activity for other substrates nonlinearly dependent on enzyme\\nlevel36. These sorts of interactions, when introduced into reaction networks of the right\\ntopology, may provide the nonlinearity needed to generate oscillations.\\nIn the next section we present a general classification scheme for simple regulatory motifs\\nthat exhibit sustained oscillations, provided the chemical implementation has sufficient\\nnonlinearity (introduced by reactions like those in Fig. 4) and the time scales of the reactions\\nare properly balanced.\\nClassification of oscillatory motifs\\nSo far, we have shown by examples that biochemical oscillations may be generated by a\\ndelayed negative feedback loop (at least three components in the loop) or by combining\\npositive and negative feedback loops. In this section, we want to put these two examples into\\na general scheme for classifying motifs of biochemical oscillators (Fig. 5).\\nFirst of all, we conjecture that oscillators always involve a negative feedback loop. We\\nknow of no examples of chemical oscillations without negative feedback, and negative\\nfeedback seems necessary to close a sequence of chemical states back on itself. In all our\\nexamples, we will use the letter X to denote the ‘activator’ and Y the ‘inhibitor’ in the\\nnegative feedback loop, that is, X ‘activates’ Y and Y ‘inhibits’ X. We allow that the\\nactivation or inhibition may be indirect, i.e., through an intermediate Z.\\nSecondly, we limit our study to mechanisms that lack autocatalysis (that is, cases where\\ncomponent X directly promotes its own activity). We can think of a few examples (a protein\\nkinase that phosphorylates and activates itself, or a misfolded protein that induces other\\ncopies of itself to misfold), but direct autocatalysis of this sort is rare compared to self-\\npromotion by a positive feedback loop (for example, X activates W and W activates X).\\nDirect autocatalysis can be incorporated into the scheme we are presenting, but it makes the\\nenumeration of cases unnecessarily complex.\\nWe have already shown that the capacity to oscillate (in networks lacking direct\\nautocatalysis) requires at least three chemical species interacting by at least three regulatory\\nlinks (activation or inhibition). Motifs (A) and (B) above show two simple examples: a\\nthree-component negative feedback loop, and a pair of coupled positive- and negative\\nfeedback loops (3 components and 4 links). We have systematically surveyed all three-\\ncomponent regulatory motifs with three or four interaction links and investigated eachNovak and Tyson Page 7', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 6}),\n",
       " Document(page_content='feedback loops (3 components and 4 links). We have systematically surveyed all three-\\ncomponent regulatory motifs with three or four interaction links and investigated eachNovak and Tyson Page 7\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 6}),\n",
       " Document(page_content='topologically distinct motif for the capacity to oscillate. The oscillatory motifs we found can\\nbe divided into three classes, as follows.\\nClass 1: Delayed Negative Feedback Loops\\nBy delayed negative feedback we refer to 3 or more components connected in a single loop\\nby positive and negative links, with an odd number of inhibitory links. Delayed negative\\nfeedback is often used to model oscillatory responses in molecular cell biology. Besides\\ncircadian oscillations of PER protein in fruit flies (mentioned earlier), other examples\\ninclude oscillations of p53 in response to ionizing radiation (p53 → MDM2  mRNA →\\nMDM2 protein ⊣ p53)37-40 and oscillations of NF- κB in response to stimulation by tumor\\nnecrosis factor (NF- κB → IκB mRNA → IκB protein ⊣ NF-κB)40-43.\\nBecause this regulatory motif has 3 or more components, it cannot be adequately\\nrepresented by a two-variable state space, as in Figs. 2b and 3b. Nonetheless, it is instructive\\nto plot trajectories of the basic motif (X → Y → Z ⊣X) in the XY plane (Fig. 5a, left) and\\nin the XZ plane (Fig. 5a, right), and to compare these plots to the case of a two-component\\nnegative feedback loop (X → Y⊣X) in Fig. 2b. In that figure, curves a and b (called\\n‘nullclines’) indicate places where the flow of the reaction system is horizontal (in the Y\\ndirection only) and where it is vertical (in the X direction only). (Please notice that we\\nalways plot the activator X on the vertical axis.) These ‘flow indicators’ force the\\ntrajectories (the dashed curves in Fig. 2b) to spiral into the steady state. For the case of a\\ndelayed feedback loop, the trajectories do not obey the flow indicators (Fig. 5a). If we are\\nplotting X vs. Y (Fig. 5a, left panel), the limit cycle trajectory does not cross curve a in a\\nhorizontal direction, because the rate of synthesis of X depends on Y concentration some\\ntime in the past. If we are plotting X vs. Z (Fig. 5a, right panel), the limit cycle trajectory\\ndoes not cross curve b in a vertical direction, because the rate of synthesis of Z depends on\\nX concentration some time in the past. In either case, the delay allows the trajectory to form\\na closed orbit around the steady state instead of spiraling into it.\\nClass 2: Amplified Negative Feedback Loops\\nIn Fig. 3 we considered a special case where the inhibitor (the protein) is amplified by a\\npositive feedback loop. It should be obvious that activator amplification may be just as\\neffective. For the case of activator amplification (Fig. 5b, left panel), the X-nullcline (curve\\na) is ‘kinked’ by the positive feedback loop. For inhibitor amplification (Fig. 5b, right\\npanel), the Y-nullcline (curve b) is ‘kinked’. In either case, trajectories are now forced to\\nwheel around the steady state onto a closed orbit (curve c, a sustained oscillation).\\nWhere do the ‘kinks’ come from? A two-component positive feedback loop (W → X → W\\nor W ⊣X ⊣W) can respond in a bistable manner to inhibition by Y. (In this case—Fig. 5b,\\nleft—we are thinking of Y as signal strength and X as the response variable.) If Y is large,\\nthen X will be small, for sure, and if Y is small, then X may be large. But for intermediate\\nvalues of Y, the steady state concentration of X can be either large or small, depending upon\\nhow the system got to the intermediate value of Y. This bistability is reflected in the Z-\\nshaped nullcline (curve a) on the left of Fig. 5b. For the case of inhibitor amplification (Fig.\\n5b, right), we think of X as signal strength and Y as response variable. In this case, Y can be\\na multi-valued function of X, and the Y-nullcline (curve b) becomes N-shaped. In either\\ncase, the negative feedback between X and Y forces trajectories to rotate clockwise on our\\nstandard XY plane, and the positive feedback loop puts kinks in one of the nullclines to\\nprevent trajectories from spiraling into the steady state.\\nThis class of oscillators appears commonly in the literature, from the earliest models of', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 7}),\n",
       " Document(page_content='prevent trajectories from spiraling into the steady state.\\nThis class of oscillators appears commonly in the literature, from the earliest models of\\nchemical oscillations 44-46 to recent models of mitosis-promoting factor (MPF) in frog eggNovak and Tyson Page 8\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 7}),\n",
       " Document(page_content=\"extracts47, 48. The latter case is an activator-amplified negative feedback loop, with\\nW=Cdc25, X=MPF and Y=Cdc20. An inhibitor-amplified negative feedback loop\\n(X=Cdc10, Y=Cig2, Z=Rum1) has been used by Novak & Tyson49 to model\\nendoreplication (periodic DNA synthesis in the absence of cell division) in mutant fission\\nyeast cells. Recently, a synthetic oscillator based on an activator-amplified negative\\nfeedback loop has been built in bacteria by Jeff Hasty's group50.\\nClass 3: Incoherently Amplified Negative Feedback Loops\\nBy rewiring an activator-amplified negative feedback loop (D), we create new regulatory\\nmotifs (C and E)\\nthat may also have the potential to oscillate. Each of the motifs (C) and (E) has a two-\\ncomponent positive feedback loop (− −, in both C and E) embedded in a three-component\\nnegative feedback loop (+ + − in C and − − − in E). Motif (C) also has the characteristic that\\nX inhibits W directly and activates W indirectly (through Y). This characteristic is called an\\nincoherent feedforward loop (C ′). Motif (E) can also be redrawn as an incoherent\\nfeedforward loop (E ′).\\nHence, we describe these motifs as incoherently amplified negative feedback loops (NFLs).\\nIn both motifs C ′ and E′, the embedded positive feedback loop is (− −). There are two other\\nincoherently amplified NFLs based on an embedded (+ +) feedback loop. The four cases are\\nshown in Fig. 5c. In each case, the NFL may, of course, oscillate in its own right, but the\\nadditional positive feedback loop adds bistability and robustness to the mechanism51.\\nThe earliest models of glycolytic oscillations 52-54 belong to this class of oscillators (Fig.\\n5c, fourth motif). In this motif, Z → X refers to the biochemical reaction that converts\\nfructose-6-phosphate + ATP (Z) into fructose-1,6-bisphosphate + ADP (X). The enzyme\\nthat catalyzes this reaction, phosphofructokinase (species Y in the motif), is activated by\\nADP (X → Y), and the active form of Y promotes both the removal of Z (Y ⊣ Z) and the\\nproduction of X (Y → X).\\nThe Martiel-Goldbeter55 model of cyclic AMP oscillations in slime mold cells is another\\nexample of the fourth motif in Fig. 5c. In this case, intracellular cAMP is the activator (X)\\nand extracellular cAMP is the incoherent signaler (Y). Extracellular cAMP binds to a\\nmembrane receptor that quickly activates the enzyme that synthesizes cAMP from ATP\\ninside the cell. This is the fast activatory signal from Y to X. Furthermore, extracellular\\ncAMP pushes the membrane receptor (Z) into an inactive state, which only slowly recovers\\nactivity after cAMP is destroyed by extracellular phosphodiesterase. This is the slow\\ninhibitory signal from Y to X. Tyson & Novak56, 57 use incoherently amplified(- -)NFLsNovak and Tyson Page 9\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 8}),\n",
       " Document(page_content='(the first and second motifs in Fig. 5c) to model cell cycle transitions (M/G1 and G1/S,\\nrespectively). To model oscillations in p53 (a transcription factor that coordinates\\nintracellular responses to DNA damage), Ciliberto et al.58 used an incoherently amplified(−\\n−)NFL and Zhang et al.59 suggested an alternative mechanism based on an incoherently\\namplified(+ +)NFL (the first model in their Fig. 1).\\nOther possibilities?\\nThere are three other regulatory motifs with 3 components, 4 links and a topology similar to\\nmotifs (C ′) and (E′):\\nIn these motifs, we use circles to indicate interactions of either sign (+ or -); two white\\ncircles must have the same sign, whereas black and white circles have opposite signs.\\nMotif (F) is a ‘coherently repressed’ NFL, consisting of two negative feedback loops, one\\nwith three components and the other with two components. Clearly, if the link X → Y is\\nweak enough, the three-component NFL might oscillate on its own. Addition of the two-\\ncomponent NFL dampens the propensity of the three-component NFL to oscillate.\\nMotif (G) is a different sort of incoherently amplified NFL; it differs from the Class 3 motifs\\nin Fig. 5c in that the positive loop has three components (rather than two) and the negative\\nloop has two components (rather than three). Everything we have said so far might lead us to\\nexpect oscillations in this motif under the right choice of nonlinearities and rate constants.\\nHowever, it is possible to show that motif (G) cannot generate oscillations for any choice of\\nnonlinear rate equations or any parameter settings (see supplementary information S3\\n(box)).\\nMotif (H) has two positive feedback loops, one with two components and one with three.\\nWe might expect this regulatory motif to exhibit bistability but not oscillations, and indeed\\nthis is the case (see supplementary information S4 (box)).\\nMore complex topologies and oscillatory behaviours\\nTo this point we have exhausted all possible oscillatory motifs with 3 components and at\\nmost 4 links. Topologies with 3 components and 5 or 6 links are so densely connected that it\\nis difficult to think of them as regulatory ‘motifs’. For example, motif (I) combines a Class 2\\ninhibitor-amplified negative feedback loop with a Class 1 three-component negative\\nfeedback loop. Either sub-motif may oscillate in its own right. Motif (I) has recently been\\nused by Rust et al.60 to model circadian oscillations in phosphorylation state of the KaiC\\nprotein in cyanobacteria. In this case, X is KaiC phosphorylated on threonine alone or on\\nthreonine + serine, Y is KaiC phosphorylated on serine alone, and Z is KaiA.Novak and Tyson Page 10\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 9}),\n",
       " Document(page_content=\"Allowing for four components and 5 or 6 links opens new possibilities that we have not\\nexplored systematically. Some oscillatory motifs are simple generalizations of topologies we\\nhave seen before. For example, motif (J) is just an activator-amplified negative feedback\\nloop (Class 2); the only difference from Fig. 5b is that the negative feedback loop has been\\nextended to three components. This motif is capable of oscillating61.\\nBy combining two or more oscillatory motifs in a common mechanism it is possible to\\ncreate quite exotic behavior including chaos, as illustrated in Fig. 6. Figure 6a, adapted from\\nRössler's classic paper62. on chemical chaos, presents two overlapping ‘activator\\namplification’ oscillators (W-X-Y 1 and W-X-Y 2) that share a common activator. As Fig. 6b\\nshows, the time course of X fluctuates up and down but never repeats itself. Viewing a\\ntrajectory in (X-Y 1-Y2) space (Fig. 6c), we see that the curve never closes on itself (it is not\\nperiodic) and seems to sweep out a surface of complex topology.\\nIt would be beyond the scope of this review to go further into the subtleties of deterministic\\nchaos, except to point out that the requirements for chaos seem to be quite undemanding.\\nChaotic trajectories readily arise in systems of coupled oscillators. Since multiple, coupled\\noscillators are likely to be common companions in the complex reaction networks\\nunderlying cell physiology, it is surprising that deterministic chaos has not been identified\\nmore often in experimental data5. Perhaps the chaotic trajectories of individual cells are\\naveraged out when large populations of cells are monitored. When the behavior of single\\ncells is monitored with fluorescent proteins, the possibility of deterministic chaos is likely to\\nbe swamped by the white noise of molecular fluctuations in small volumes (a single cell).\\nNonetheless, experimentalists and theoreticians should be open to the possibility of\\ndeterministic chaos in their data and models of complex reaction networks with multiple\\nsources of oscillation.\\nSumming up\\nBy modelling specific examples of oscillatory processes, we have drawn a number of\\ngeneral conclusions about the design principles of biochemical oscillators. All biochemical\\noscillators are built around some sort of negative feedback signal (X → Y → … ⊣X),\\nwhich insures that, if the concentration of X gets too large, it will eventually decrease, and if\\nit gets too small it will eventually increase 63. Negative feedback is often used in\\nbiochemistry to achieve homeostasis (a stable steady state of intermediate X level), but\\nunder certain conditions the steady state may lose stability and be replaced by spontaneous\\noscillations of X level (high → low → high → low → …). The conditions for oscillations\\nare: sufficient nonlinearity in the reaction kinetics, sufficient ‘memory’ in the negative\\nfeedback loop, and proper balancing of the time scales of components within the loop. In\\nbiochemical reaction kinetics there are many sources of nonlinearity that are conducive to\\noscillations (Fig. 4). ‘Memory’ may be a simple consequence of a long negative feedback\\nloop, but more likely it stems from positive feedback loops in the biochemical reaction\\nmechanism. When positive feedback creates two, alternative, stable steady states in the\\nreaction dynamics, then the system can ‘remember’ its recent history and thereby overshoot\\nand undershoot the homeostatic tendencies of the negative feedback loop.Novak and Tyson Page 11\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 10}),\n",
       " Document(page_content='With these ideas in mind we classify biochemical oscillators according to the topology of\\nthe positive and negative feedback loops in the reaction mechanism. For systems with three\\ncomponents and three-or-four links (no self-activation links), we identify three classes of\\noscillators: delayed negative feedback loops, amplified negative feedback loops, and\\nincoherently amplified negative feedback loops. Our classification scheme is by no means\\ncomplete, and oscillator motifs may be more complicated than any of our classes. Also, we\\nhave neglected mixed-mode effects, e.g., Y activates the synthesis of X at low concentration\\nand inactivates it at high concentration. Mixed-mode interactions may easily generate\\ncomplex oscillations and chaos.\\nIf a reaction mechanism contains one of the regulatory motifs we have identified, then it\\nmay exhibit oscillatory behavior, provided the rate constants are properly tuned. Because\\ninteraction motifs and reaction rate constants are under genetic control, it is possible for\\nbiochemical oscillators to evolve. Indeed, it is likely that biochemical oscillations have\\narisen repeatedly from basic, homeostatic, negative feedback loops by serendipitous genetic\\nchanges that destabilized the steady state and generated sustained oscillations. Maladaptive\\noscillations would have been quickly weeded out by selection, but weakly deleterious or\\nadventitious oscillations may have been co-opted by evolving populations for beneficial\\nphysiological purposes. Almost surely, mechanisms of circadian rhythms evolved many\\ntimes independently by this scenario. On the other hand, bistability and oscillations that\\ngovern cell cycle events are much more highly constrained, and the underlying mechanism\\n(which seems to be universal across all eukaryotic cells) must have been derived from a\\nsingle common ancestor.\\nAnother consequence of the fact that a given interaction motif may or may not oscillate,\\ndepending on subtle balancing of reaction rates, is the possibility of a class of ‘dynamical\\ndiseases’ as distinct from ‘genetic diseases’. In a genetic disease, like sickle cell anemia, a\\nmutant gene encodes a defective protein that cannot do its essential job in some important\\naspect of physiology. In a dynamical disease, a mutant gene encodes a modified protein that\\nstill does its job but at a different rate, thereby causing a homeostatic control mechanism to\\nbreak out into pathological oscillations, or an oscillatory control system to spiral into a\\nstable steady state. Some cyclic blood disorders may be dynamical diseases of the first kind,\\nand some sleep disorders may be dynamical diseases of the second kind.\\nFinally, we hope that this review will help biochemists and molecular biologists to\\nunderstand better the mechanisms underlying cellular oscillations and to recognize the\\nimportance of quantitative modelling in studying these oscillations.\\nSupplementary Material\\nRefer to Web version on PubMed Central for supplementary material.\\nAcknowledgments\\nJJT acknowledges financial support from the National Institutes of Health (R01-GM07898 and R01-GM079207)\\nand the hospitality of Merton College, Oxford, during the writing of this review. B.N. acknowledges support from\\nthe BBSRC (UK) and from the EC FP7 (201142). Our understanding of biochemical oscillations has developed\\nover many years of delightful conversations with Albert Goldbeter, Lee Segel, Art Winfree, Michael Mackey and\\nLeon Glass.\\nReferences\\n1. Pye K, Chance B. Sustained sinusoidal oscillations of reduced pyridine nucleotide in a cell-free\\nextract of Saccharomyces carlsbergensis. Proc Natl Acad Sci U S A. 1966; 55:888–894. [PubMed:\\n4286762]Novak and Tyson Page 12\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 11}),\n",
       " Document(page_content=\"2. Hess B, Boiteux A. Oscillatory phenomena in biochemistry. Annu Rev Biochem. 1971; 40:237–258.\\n[PubMed: 4330578]\\n3. Gerisch G, Fromm H, Huesgen A, Wick U. Control of cell-contact sites by cyclic AMP pulses in\\ndifferentiating Dictyostelium cells. Nature. 1975; 255:547–549. [PubMed: 167285]\\n4. Olsen LF, Degn H. Oscillatory kinetics of the peroxidase-oxidase reaction in an open system.\\nExperimental and theoretical studies. Biochim Biophys Acta. 1978; 523:321–334. [PubMed:\\n207332]\\n5. Olsen LF, Degn H. Chaos in an enzyme reaction. Nature. 1977; 267:177–178. [PubMed: 16073439]\\n6. Higgins J. Theory of oscillating reactions. Ind Eng Chem. 1967; 59:19–62.\\n7. Prigogine I, Lefever R, Goldbeter A, Herschkowitz-Kaufman M. Symmetry breaking instabilities in\\nbiological systems. Nature. 1969; 223:913–916. [PubMed: 5803393]\\n8. Dunlap JC. Molecular bases for circadian clocks. Cell. 1999; 96:271–290. [PubMed: 9988221]\\n9. Evans T, Rosenthal ET, Youngblom J, Distel D, Hunt T. Cyclin: a protein specified by maternal\\nmRNA in sea urchin eggs that is destroyed at each cleavage division. Cell. 1983; 33:389–396.\\n[PubMed: 6134587]\\n10. Gerhart J, Wu M, Kirschner M. Cell cycle dynamics of an M-phase-specific cytoplasmic factor in\\nXenopus laevis oocytes and eggs. J Cell Biol. 1984; 98:1247–1255. [PubMed: 6425302]\\n11. Elowitz MB, Leibler S. A synthetic oscillatory network of transcriptional regulators. Nature. 2000;\\n403:335–338. [PubMed: 10659856]\\n12. Chance, B.; Pye, EK.; Ghosh, AK.; Hess, B. Biological and biochemical oscillators. Academic\\nPress; New York: 1973.\\n13. Gray, P.; Scott, SK. Chemical oscillations and Instabilities. Clarendon Press; Oxford: 1994.\\n14. Goldbeter, A. Biochemical oscillations and cellular rhythms. Cambridge University Press;\\nCambridge: 1996.\\n15. Epstein, IR.; Pojman, JA. An introduction to nonlinear chemical dynamics. Oxford University\\nPress; Oxford: 1998.\\n16. Berridge MJ, Rapp PE. A comparative survey of the function, mechanism and control of cellular\\noscillators. J Exp Biol. 1979; 81:217–279. [PubMed: 390080]\\n17. Goldbeter A. Computational approaches to cellular rhythms. Nature. 2002; 420:238–245.\\n[PubMed: 12432409]\\n18. Kholodenko BN. Cell-signalling dynamics in time and space. Nat Rev Mol Cell Biol. 2006; 7:165–\\n176. [PubMed: 16482094]\\n19. Hardin PE, Hall JC, Rosbash M. Feedback of the Drosophila period gene product on circadian\\ncycling of its messenger RNA levels. Nature. 1990; 343:536–540. [PubMed: 2105471]\\n20. Gallego M, Virshup DM. Post-translational modifications regulate the ticking of the circadian\\nclock. Nat Rev Mol Cell Biol. 2007; 8:139–148. [PubMed: 17245414]\\n21. Mackey MC, Glass L. Oscillation and chaos in physiological control systems. Science. 1977;\\n197:287–289. [PubMed: 267326]\\n22. Griffith JS. Mathematics of cellular control processes. I. Negative feedback to one gene. J Theor\\nBiol. 1968; 20:202–208. [PubMed: 5727239]\\n23. Goodwin BC. Oscillatory behavior in enzymatic control processes. Adv Enzyme Regul. 1965;\\n3:425–438. [PubMed: 5861813]\\n24. Goodwin BC. An entrainment model for timed enzyme syntheses in bacteria. Nature. 1966;\\n209:479–481. [PubMed: 5919577]\\n25. Masters M, Donachie WD. Repression and the control of cyclic enzyme synthesis in Bacillus\\nsubtilis. Nature. 1966; 209:476–479. [PubMed: 4224087]\\n26. Goldbeter A. A model for circadian oscillations in the Drosophila period protein (PER). Proc Biol\\nSci. 1995; 261:319–324. [PubMed: 8587874]\\n27. Leloup JC, Goldbeter A. Modeling the circadian clock: from molecular mechanism to\\nphysiological disorders. Bioessays. 2008; 30:590–600. [PubMed: 18478538]\\n28. Thomas, R.; D'Ari, R. Biological feedback. CRC Press; Boca Raton, Florida: 1990.Novak and Tyson Page 13\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 12}),\n",
       " Document(page_content=\"29. Laurent M, Kellershohn N. Multistability: a major means of differentiation and evolution in\\nbiological systems. Trends Biochem Sci. 1999; 24:418–422. [PubMed: 10542403]\\n30. Tyson JJ, Hong CI, Thron CD, Novak B. A simple model of circadian rhythms based on\\ndimerization and proteolysis of PER and TIM. Biophys J. 1999; 77:2411–2417. [PubMed:\\n10545344]\\n31. Segel, LA. Biological kinetics. Cambridge University Press; Cambridge: 1991.\\n32. Monod J, Wyman J, Changeux JP. On the nature of allosteric transitions: a plausible model. J Mol\\nBiol. 1965; 12:88–118. [PubMed: 14343300]\\n33. Goldbeter A, Koshland DE Jr. An amplified sensitivity arising from covalent modification in\\nbiological systems. Proc Natl Acad Sci U S A. 1981; 78:6840–6844. [PubMed: 6947258]\\n34. Gunawardena J. Multisite protein phosphorylation makes a good threshold but can be a poor\\nswitch. Proc Natl Acad Sci U S A. 2005; 102:14617–14622. [PubMed: 16195377]\\n35. Thron CD. Mathematical analysis of binary activation of a cell cycle kinase which down-regulates\\nits own inhibitor. Biophys Chem. 1999; 79:95–106. [PubMed: 10389236]\\n36. Kim SY, Ferrell JE Jr. Substrate competition as a source of ultrasensitivity in the inactivation of\\nWee1. Cell. 2007; 128:1133–1145. [PubMed: 17382882]\\n37. Lahav G, et al. Dynamics of the p53-Mdm2 feedback loop in individual cells. Nat Genet. 2004;\\n36:147–150. [PubMed: 14730303]\\n38. Lev Bar-Or R, et al. Generation of oscillations by the p53-Mdm2 feedback loop: a theoretical and\\nexperimental study. Proc Natl Acad Sci U S A. 2000; 97:11250–11255. [PubMed: 11016968]\\n39. Ma L, et al. A plausible model for the digital response of p53 to DNA damage. Proc Natl Acad Sci\\nU S A. 2005; 102:14266–14271. [PubMed: 16186499]\\n40. Monk NA. Oscillatory expression of Hes1, p53, and NF-kappaB driven by transcriptional time\\ndelays. Curr Biol. 2003; 13:1409–1413. [PubMed: 12932324]\\n41. Nelson DE, et al. Oscillations in NF-kappaB signaling control the dynamics of gene expression.\\nScience. 2004; 306:704–708. [PubMed: 15499023]\\n42. Cheong R, Hoffmann A, Levchenko A. Understanding NF-kappaB signaling via mathematical\\nmodeling. Mol Syst Biol. 2008; 4:192. [PubMed: 18463616]\\n43. Hoffmann A, Levchenko A, Scott ML, Baltimore D. The IkappaB-NF-kappaB signaling module:\\ntemporal control and selective gene activation. Science. 2002; 298:1241–1245. [PubMed:\\n12424381]\\n44. Franck UF. Kinetic feedback processes in physico-chemical oscillatory systems. Faraday Symp\\nChem Soc. 1974; 9:137–149.\\n45. Rossler OE. A principle for chemical multivibration. J Theor Biol. 1972; 36:413–417. [PubMed:\\n5073926]\\n46. Gierer A, Meinhardt H. A theory of biological pattern formation. Kybernetik. 1972; 12:30–39.\\n[PubMed: 4663624]\\n47. Novak B, Tyson JJ. Numerical analysis of a comprehensive model of M-phase control in Xenopus\\noocyte extracts and intact embryos. J Cell Sci. 1993; 106:1153–1168. [PubMed: 8126097]\\n48. Pomerening JR, Kim SY, Ferrell JE Jr. Systems-level dissection of the cell-cycle oscillator:\\nbypassing positive feedback produces damped oscillations. Cell. 2005; 122:565–578. [PubMed:\\n16122424]\\n49. Novak B, Tyson JJ. Modeling the control of DNA replication in fission yeast. Proc Natl Acad Sci\\nU S A. 1997; 94:9147–9152. [PubMed: 9256450]\\n50. Stricker J, et al. A fast, robust, and tunable synthetic gene oscillator. Nature. 2008 in press.\\n51. Tsai TY, et al. Robust, tunable biological oscillations from interlinked positive and negative\\nfeedback loops. Science. 2008; 321:126–129. [PubMed: 18599789]\\n52. Goldbeter A, Lefever R. Dissipative structures for an allosteric model. Application to glycolytic\\noscillations. Biophys J. 1972; 12:1302–1315. [PubMed: 4263005]\\n53. Higgins J. A chemical mechanism for oscillation of glycolytic intermediates in yeast cells. Proc\\nNatl Acad Sci USA. 1964; 51:989–994. [PubMed: 14215656]\\n54. Sel'kov. Self-oscillation in glycolysis. 1. A simple kinetic model. Eur J Biochem. 1968; 4:79–86.\\n[PubMed: 4230812]Novak and Tyson Page 14\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 13}),\n",
       " Document(page_content=\"Natl Acad Sci USA. 1964; 51:989–994. [PubMed: 14215656]\\n54. Sel'kov. Self-oscillation in glycolysis. 1. A simple kinetic model. Eur J Biochem. 1968; 4:79–86.\\n[PubMed: 4230812]Novak and Tyson Page 14\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 13}),\n",
       " Document(page_content=\"55. Martiel JL, Goldbeter A. A model based on receptor desensitization for cyclic AMP signaling in\\nDictyostelium cells. Biophys J. 1987; 52\\n56. Tyson JJ, Novak B. Regulation of the eukaryotic cell cycle: molecular antagonism, hysteresis, and\\nirreversible transitions. J Theor Biol. 2001; 210:249–263. [PubMed: 11371178]\\n57. Tyson JJ, Novak B. Temporal organization of the cell cycle. Current Biology. 2008; 18:R759–768.\\n[PubMed: 18786381]\\n58. Ciliberto A, Novak B, Tyson JJ. Steady states and oscillations in the p53/Mdm2 network. Cell\\nCycle. 2005; 4:488–493. [PubMed: 15725723]\\n59. Zhang T, Brazhnik P, Tyson JJ. Exploring mechanisms of the DNA-damage response: p53 pulses\\nand their possible relevance to apoptosis. Cell Cycle. 2007; 6:85–94. [PubMed: 17245126]\\n60. Rust MJ, Markson JS, Lane WS, Fisher DS, O'Shea EK. Ordered phosphorylation governs\\noscillation of a three-protein circadian clock. Science. 2007; 318:809–812. [PubMed: 17916691]\\n61. Borisuk MT, Tyson JJ. Bifurcation analysis of a model of mitotic control in frog eggs. J Theor\\nBiol. 1998; 195:69–85. [PubMed: 9802951]\\n62. Rossler OE. Chaos in abstract kinetics: two prototypes. Bull Math Biol. 1977; 39:275–289.\\n[PubMed: 851667]\\n63. Snoussi EH. Necessary conditions for multistationarity and stable periodicity. Journal of Biological\\nSystems. 1998; 6:3–9.\\n64. Goldbeter A. Mechanism for oscillatory synthesis of cyclic AMP in Dictyostelium discoideum.\\nNature. 1975; 253:540–542. [PubMed: 163974]\\n65. Meyer T, Stryer L. Molecular model for receptor-stimulated calcium spiking. Proc Natl Acad Sci\\nU S A. 1988; 85:5051–5055. [PubMed: 2455890]\\n66. Garmendia-Torres C, Goldbeter A, Jacquet M. Nucleocytoplasmic oscillations of the yeast\\ntranscription factor Msn2: evidence for periodic PKA activation. Curr Biol. 2007; 17:1044–1049.\\n[PubMed: 17570669]\\n67. Jacquet M, Renault G, Lallet S, De Mey J, Goldbeter A. Oscillatory nucleocytoplasmic shuttling of\\nthe general stress response transcriptional activators Msn2 and Msn4 in Saccharomyces cerevisiae.\\nJ Cell Biol. 2003; 161:497–505. [PubMed: 12732613]\\n68. Lewis J. Autoinhibition with transcriptional delay: a simple mechanism for the zebrafish\\nsomitogenesis oscillator. Curr Biol. 2003; 13:1398–1408. [PubMed: 12932323]Novak and Tyson Page 15\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 14}),\n",
       " Document(page_content='Figure 1. Time-delayed negative feedback oscillator\\na| Protein level is determined by opposing processes of synthesis and degradation. Protein\\nsynthesis is down-regulated by the protein itself. b| Curves a and b are the rates of protein\\nsynthesis and degradation, respectively. The arrows indicate the direction of change of\\nprotein concentration, which is always towards Yo, the steady state concentration of protein,\\nwhere the rate of synthesis equals the rate of degradation. c| Sustained oscillations for Eq.\\n(2), with p = 2, Km/Kd=1, S/Kd =1, k1 = k2ET/Kd = 1 min-1, and τ = 10 min. The period of\\noscillation is Tc = 27.2 min. d| In curve c we plot the time-delayed rate of protein synthesis,\\n1/(1+ Y(t−τ)p), as a function of the present protein concentration, Y(t). The dashed portion ofNovak and Tyson Page 16\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 15}),\n",
       " Document(page_content='curve c corresponds to the dashed portion of the oscillation in panel c; it is τ time units in\\nduration, and it extends from the maximum value of Y (at t = 20 min) to the minimum value\\nof the rate of production of Y (at t = 30 min). The time-delayed loop repeatedly overshoots\\nand undershoots the steady state because the protein synthesis rate is no longer given by\\ncurve a at Y(t) but by curve a at Y(t - τ). e| Constraint curves for p = 1. Each curve is drawn\\nfor a specific value of Kd/Km. For each case, Eq. (2) exhibits sustained oscillations in the\\nregion above the curve. f| Constraint curves for p = 2. Notice that the oscillatory domain\\nbecomes larger as p increases and as Kd/Km increases, i.e., as the kinetic rate laws become\\nmore nonlinear.Novak and Tyson Page 17\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 16}),\n",
       " Document(page_content='Figure 2. Multi-component, negative feedback oscillator\\na| Negative feedback between mRNA and protein, as described by kinetic equations (3). b|\\nRepresentative solutions (dashed curves) of the kinetic equations (3), for parameter values: p\\n= 2, Km/Kd =1, S/Kd =1, k1 = kdx = 0.1 min-1, ksy = k2ET/Kd = 1 min-1. Notice that every\\ntrajectory spirals into the stable steady state located at the black circle. Curves a and b are\\n‘nullclines’ for differential equations (3), as explained in the text. The small arrows indicate\\nthe direction of motion of trajectories as they cross the nullclines. Notice that the nullclines\\nin this figure are identical to the rate curves in Fig. 1b. c| The negative feedback loop taking\\ninto account transport of macromolecules between nucleus and cytoplasm. d| SustainedNovak and Tyson Page 18\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 17}),\n",
       " Document(page_content='oscillations for the four-component loop in panel c. See supplementary information S1 (box)\\nfor details. e| Nonlinearity constraint. For the negative feedback loop to oscillate, p and Kd /\\nKm must be large enough. f| Time-scale balancing constraint. The half-lives of mRNA in the\\nnucleus and of protein in the cytoplasm must lie in the shaded band in order for the negative\\nfeedback loop to oscillate.Novak and Tyson Page 19\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 18}),\n",
       " Document(page_content='Figure 3. Hysteresis-driven, negative feedback oscillator\\na| mRNA and protein in a negative feedback loop, as in Figure 2, and the protein inhibits its\\nown degradation. This mechanism is described by kinetic equations (4). b| Limit cycle\\nsolution (curve c) of Eqs. (4) for parameter values: p = 4, Km/Kd =0.1, Kd·KI = 2, S =1, k1 =\\nkdx = kdy = 0.05 min-1, ksy = k2ET/Kd = 1 min-1. Curves a and b as in Fig. 2b, except that\\ncurve b is given by X = k dyY+Y/(Km+Y+KIY2). c| Sustained oscillations of mRNA and\\nprotein, corresponding to curve c in panel b. d| Signal-response curve. Solid lines: stable\\nsteady states; dashed lines: unstable steady states; black circles: maximum and minimum\\nexcursions of Y(t) during a limit cycle oscillation. The oscillation in panel c is indicated byNovak and Tyson Page 20\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 19}),\n",
       " Document(page_content='the double-headed arrow at S = 1. Notice that oscillations are possible only for a restricted\\nrange of signal strengths, S. e| Nonlinearity constraint. For this mechanism to oscillate, the\\npositive feedback loop must be strong enough ( KI sufficiently large) and the negative\\nfeedback loop must be sufficiently nonlinear ( p must be large enough). f| Time-scale\\nbalancing constraint. The turnover rate of mRNA ( kdx) cannot be too large, and the signal\\nstrength ( S) must be within specific bounds for this system to oscillate.Novak and Tyson Page 21\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 20}),\n",
       " Document(page_content='Figure 4. Sources of nonlinearity\\na| Oligomer binding. Left: a transcription factor (blue ball) forms an n-component homo-\\noligomer, which then binds upstream of a structural gene and either activates or represses\\nmRNA synthesis. Right: rate of mRNA synthesis as a function of transcription factor\\nconcentration, for an activator (solid line) or a repressor (dashed line). b| Cooperativity and\\nallostery. Left: an enzyme, consisting of two catalytic subunits (spheres) and two regulatory\\nsubunits (cubes), catalyzes the conversion of substrate into product. Activators and\\ninhibitors bind to specific sites on the regulatory subunits. Right: if the binding of substrate\\nto the catalytic subunits is cooperative, then the rate of reaction as a function of substrateNovak and Tyson Page 22\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 21}),\n",
       " Document(page_content='concentration is sigmoidal (solid line). The rate curve can be shifted to the left or to the right\\nby increasing concentrations of activator or inhibitor, respectively. c| Multisite\\nphosphorylation. Left: a regulatory protein, X, is phosphorylated on multiple sites by a\\nprotein kinase and dephosphorylated by a protein phosphatase. Right: Concentration of the\\nunphosphorylated form of X as a function of the ratio of activities of kinase and\\nphosphatase. d| Stoichiometric inhibition. Left: a regulatory protein, X, is synthesized in\\nresponse to a signal, S. X binds strongly to an inhibitor to form an inactive complex. Right:\\nthe concentration of total X increases hyperbolically with S (dashed line) but the\\nconcentration of ‘free’ X is a sigmoidal function of S (solid line).Novak and Tyson Page 23\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 22}),\n",
       " Document(page_content='Figure 5. A classification scheme for biochemical oscillators\\nWe classify oscillators by their interaction motifs, where X → Y means ‘X activates Y’, Y\\n⊣X means ‘Y inhibits X’, and W ⊸ X means ‘W may either activate or inhibit X’. If two\\nwhite circles appear in the same regulatory motif, they must have the same sign (either ++ or\\n−−). We assume that all interactions are positive or negative (not mixed-mode) and all self-\\ninteractions are negative. a| Class 1: delayed negative feedback loops. Below each feedback\\nloop, we present a state-space diagram in the style of Fig. 1d. We plot ‘activator’ X vs\\n‘inhibitor’ Y (left) or Z (right). Curve c is a projection of the limit cycle oscillation onto the\\nXY plane. b| Class 2: amplified negative feedback loops. Either the activator X may beNovak and Tyson Page 24\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 23}),\n",
       " Document(page_content='amplified by positive feedback with W (left), or the inhibitor Y may be amplified by\\npositive feedback with Z (right). For each motif, we plot the limit cycle oscillation (curve c)\\non the XY plane. c| Class 3: incoherently amplified negative feedback loops. Each motif\\nconsists of a three-component negative feedback loop (‘oscillatory’) and a two-component\\npositive feedback loop (‘amplifying’). Each motif also contains an incoherent feed-forward\\nloop that may originate from either X or Y. To the left and right of each motif we indicate\\nhow the state-space diagram will appear, depending upon which variable is plotted on the\\nabscissa and which on the ordinate.Novak and Tyson Page 25\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 24}),\n",
       " Document(page_content='Figure 6. Chaotic oscillators\\na| Activator-amplification with two negative feedback loops in parallel. b| Chaotic trajectory\\nfor the mechanism in panel a. See supplementary information S1 (box) for details. c|\\nProjection of the chaotic trajectory into the three-dimensional state space ( X, Y1, Y2). The\\nchaotic trajectory was recomputed from the equations and parameter values in Rössler62.Novak and Tyson Page 26\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.\\nNIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author Manuscript', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 25}),\n",
       " Document(page_content='NIH-PA Author Manuscript NIH-PA Author Manuscript NIH-PA Author ManuscriptNovak and Tyson Page 27\\nTable 1\\nSurvey of Biochemical Oscillators\\nFunction Components PeriodClass*References\\nMetabolism Glucose, ATP, phosphofructokinase 2 min 3 52-54\\nSignalling cAMP, receptor, adenylate cyclase 5 min 3 55, 64\\nSignalling Ca2+, IP 3> 1 s 3 65\\nSignalling NFκB, IκB, IKK ∼2 h 1 41, 43\\nSignalling p53, Mdm2 5 h 1 39, 40\\n3 58, 59\\nSignalling Msn2, AC, cAMP, PKA ∼10 min 1 66, 67\\nSomitogenesis Her1/her7, Notch 30-90 min 1 40, 68\\nYeast endoreplication cycles Cig2, Cdc10, Rum1 1 – 2 h 2 49\\nFrog egg cycles CycB, Wee1, Cdc25, Cdc20 30 min 2 47, 48\\nCircadian rhythm PER, TIM, CLOCK, CYC 24 h 1 26\\n2 30\\n*See Figure 5: Class 1 = delayed negative feedback loop, Class 2 = amplified negative feedback loop, Class 3 = incoherently amplified negative\\nfeedback loop.\\nNat Rev Mol Cell Biol . Author manuscript; available in PMC 2009 December 21.', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/DesignPrinciplesOscillators.pdf', 'page': 26})]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "docs = load_docs(pdfdirname)\n",
    "texts = split_docs(docs)\n",
    "texts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "OpenAIEmbeddings(client=<class 'openai.api_resources.embedding.Embedding'>, model='text-embedding-ada-002', deployment='text-embedding-ada-002', openai_api_version='2022-12-01', openai_api_base=None, openai_api_type=None, embedding_ctx_length=8191, openai_api_key=None, openai_organization=None, allowed_special=set(), disallowed_special='all', chunk_size=1000, max_retries=6)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "embeddings = OpenAIEmbeddings()\n",
    "embeddings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain.vectorstores import Chroma\n",
    "from langchain.chains import RetrievalQA\n",
    "from langchain.chat_models import ChatOpenAI\n",
    "from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create Chroma vectorstore for pdfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using embedded DuckDB without persistence: data will be transient\n"
     ]
    }
   ],
   "source": [
    "doc_retriever = Chroma.from_documents(texts, embeddings).as_retriever() #initialize vectorstore into retriever"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain.retrievers import ContextualCompressionRetriever\n",
    "from langchain.retrievers.document_compressors import LLMChainExtractor\n",
    "\n",
    "compressor = LLMChainExtractor()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load chat model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "chat = ChatOpenAI(model_name=\"gpt-4\", streaming=True, callbacks=[StreamingStdOutCallbackHandler()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "qa = RetrievalQA.from_chain_type(llm=chat, chain_type=\"stuff\", retriever=doc_retriever, return_source_documents=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Membrane localization affects biochemical kinetics in several ways. One significant influence is on the binding affinities and association/dissociation rates of proteins. As shown in the provided context, protein-lipid affinities (KaPM) are often critical in controlling the overall time-scales of complexation, even driving slowdowns in speeds relative to solution binding. The change in binding strengths, rather than the slower 2D diffusion on the membrane, primarily determines assembly speeds in physiological rate regimes.\n",
      "\n",
      "Moreover, the change from 3D to 2D interactions on the membrane surface reduces the search space for protein-protein interactions, which can lead to enhanced complex formation and faster assembly. However, the influence of diffusion on reaction rates is rarely a dominant factor in physiological rate regimes.\n",
      "\n",
      "It is important to note that as spatial dimensions increase, times to diffuse and reach the membrane will influence the overall equilibration times. Additionally, crowding would lower effective diffusion constants of proteins, although the decrease in time-scales to equilibrate would be negligible unless binding rates were strongly diffusion-influenced."
     ]
    }
   ],
   "source": [
    "query = \"How does membrane localization effect biochemical kinetics?\"\n",
    "result = qa({\"query\": query})\n",
    "# qa.run(query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Membrane localization affects biochemical kinetics in several ways. One significant influence is on the binding affinities and association/dissociation rates of proteins. As shown in the provided context, protein-lipid affinities (KaPM) are often critical in controlling the overall time-scales of complexation, even driving slowdowns in speeds relative to solution binding. The change in binding strengths, rather than the slower 2D diffusion on the membrane, primarily determines assembly speeds in physiological rate regimes.\\n\\nMoreover, the change from 3D to 2D interactions on the membrane surface reduces the search space for protein-protein interactions, which can lead to enhanced complex formation and faster assembly. However, the influence of diffusion on reaction rates is rarely a dominant factor in physiological rate regimes.\\n\\nIt is important to note that as spatial dimensions increase, times to diffuse and reach the membrane will influence the overall equilibration times. Additionally, crowding would lower effective diffusion constants of proteins, although the decrease in time-scales to equilibrate would be negligible unless binding rates were strongly diffusion-influenced.'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(result[\"result\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[Document(page_content='Weapply thetheory here tocharacterizing, within aquantitative framework, therole of\\nmembrane localization forenhancing 55binding interactions involving 33distinct protein\\npair interaction sets(S1Table). Through simulation, wealso move beyond themodel illus-\\ntrated inFig1ofonly pairs ofsoluble binding partners toshow how complexation involving\\nnon-membrane binding scaffold proteins such asclathrin, orhow formation ofhigher-order\\noligomers, which isfunctionally important fordriving membrane remodeling [11, 17,18], can\\nalso beregulated bymembrane localization (13additional interaction setsinS2Table). Our\\ntheory only applies tothepair interactions illustrated inFig1.Weinclude 22proteins involved\\ninCME inboth human and yeast cells, aswell as15proteins involved inlipid regulation, vesi-\\ncleformation onendosomes, budding, and morphogenesis inyeast cells (Table 1).Wecol-\\nlected concentration and cellular geometry data based oninvivo values tobetter connect to\\nphysiologic regimes (S3Table and S4Table). Although ourtheory represents anapproximate\\nsolution tothefullmodel shown inFig1,weshow through extensive simulations using both\\nsystems ofordinary differential equations and single-particle reaction-diffusion, [19, 20]that it\\nishighly accurate. Through simulation, weadditionally find that membrane localization alters\\nthetimescales ofprotein-protein assembly, butthat theresult isnotdominated bychanges in\\nprotein diffusion between solution and themembrane. Rather, forphysiologic binding\\nstrengths, therate-limiting step isthespeed ofbinding tothemembrane surface from solution.\\nFinally, apractical application ofoursimple formula isthat itcanbeused toexperimentally fit\\nprotein-protein binding affinities onsurfaces (Ka2D),which arerarely measured [21, 22]. The\\nadvantage oftheformula isthat itapplies toinvitro experiments where apair ofproteins can\\nreversibly bind tothemembrane, thus avoiding theneed torestrict proteins tothesurface.\\nResults\\nModel and theory\\nInourprimary model, weconsider twoproteins P1and P2,that bind insolution with equilib-\\nrium constantKPP;3D\\na\\x88\\x89P1P2\\x8aeq\\n\\x89P1\\x8aeq\\x89P2\\x8aeq\\x88\\x89P1P2\\x8aeq\\n\\x85\\x89P1\\x8a0\\x00\\x89P1P2\\x8aeq\\x86\\x85\\x89P2\\x8a0\\x00\\x89P1P2\\x8aeq\\x86,where total concentrations ofthe\\nproteins are[P1]0=[P1P2]+[P1]and same for[P2]0.Ifthese proteins canalso reversibly bind\\ntomembranes viatargeting aspecific lipidM,and continue tobind oneanother, their binding\\nequilibrium willshift asatotal ofnine distinct species canform (Fig 1,Methods). The bound\\nprotein-protein complexes caneither beinsolution oronthemembrane, [P1P2]sol+mem=\\n[P1P2]sol+[P1P2]mem=[P1P2]+[P 1P2M]+[MP 1P2][MP 1P2M],and unbound species aresimi-\\nlarly defined [P1]sol+[P1]mem=[P1]+[MP 1],and [P2]sol+[P2]mem=[P2]+[P 2M],whereMindi-\\ncates acopy ofatarget membrane lipid bound toP1orP2.The model thus assumes each\\nprotein binds membrane viatargeting asingle copy ofaspecific lipid type. Proteins onthe\\nmembrane must beable todiffuse tobind oneanother, which isconsistent with experimental\\nobservations [23] even ofRNA-protein complexes (>6000kDa) that areanchored viamultiple\\nlipid binding sites along with myristoyl groups [24]. Each ofthenine distinct species willbe\\nconstrained topreserve detailed balance atequilibrium, asdefined bythe10pairwise binding\\ninteractions ofFig1(see Methods), and thetotal concentrations ofproteins isfixed atthe\\nsame values asabove, butnow [P1]0=[P1P2]sol+mem+[P 1]sol+mem,and thesame for[P2]0.Simi-\\nlarly, [M] 0=[M]+[MP 1]+[P 2M]+[P 1P2M]+[MP 1P2]+2[MP 1P2M].Wenote that species onthe\\nmembrane have concentrations normally ofμm-2,matching theunits ofequilibrium constants\\nin2D(Ka2D)-1.Allspecies copy numbers, whether onoroffthemembrane, however, canbe\\nsolved forinvolume units when theappropriate Solution volume/Membrane surface Area (V/\\nA)conversion factor scales the2Dbinding constants, sowealways report volume units for\\nconcentrations. Toquantify thechange inbound protein-protein complexes asafunction of\\nMembrane localizatio ncantrigger protein assembly', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 3}),\n",
       " Document(page_content=\"novel and valuable metric forinterpreting how important membrane localization canbegiven\\ntheconcentrations and binding properties ofcomponent proteins, and itisolates therole of\\nmembrane localization from other factors. Since even mostinvitro experiments contain more\\ncomponents and complexity than iscaptured inoursimple model, wediscuss how itcanstill\\nbeused asaquantitative guide forestimating how membrane heterogeneity, competition for\\nbinding, and mutations would influence theparameters ofthemodel (volume, surface area,\\nbinding affinities and concentrations) and thus theproteins' subsequent response tolocaliza-\\ntion. Wespecifically address inourresults how lipids such asPI(4,5)P 2canbetargeted by\\nmany proteins atanytime [14, 15], how some membrane binding domains such asBAR\\ndomains bind membranes with widely varying lipid composition and inacurvature dependent\\nmanner [16, 17], and how mutations and multiple protein binding partners would alter pro-\\ntein complex formation. Despite thelimitations ofapplying anequilibrium theory tounder-\\nstand complexation that ultimately occurs inthenonequilibrium cell, webelieve thetheory\\nrepresents awell-defined starting point from which toprobe more complex systems, justas\\nusinginvitro studies provide auseful guide forinterpreting behavior inthecell. Itisalso aref-\\nerence point forstudying thetime-dependence ofassembly through computer simulation, as\\nwedohere, and astarting point from which tobuild further complexity into themodel.\\nFig1.Quantifyin ghow protein binding partners insolution can increase complex formation bybinding tolipids onmembrane surfaces. This model contains two\\ntypes ofproteins (P1and P2)and onetype oflipid (M). Weshow alltenpossible bindin ginteractions that canoccur between thispair ofproteins that bind each other\\n(P1+P2\\x84P 1P2)and also bind specific lipids (M), producing asystem ofnine total distinct species: P1,P2,M,P1P2,MP 1,P2M,MP 1P2,P1P2M,and MP 1P2M.a)Solution\\n(3D) binding. b)Interactio nsinsolution (3D) that pull proteins tothemembran esurface through protein orlipid binding. In(c-e) thebinding interacti onsarein2D\\n(species concentrat ioninArea-1)and canexploit thelower search space. Conversio nfrom 3Dto2Dequilibriu mconstant isdefined bythevariableσ,where onlyσ=σPP\\nappears inEq3.Tosolve forallspecies inconsistent units (i.e.Volume-1),Ka2Dvalues must bemultiplie dbyV/A. The sizeofsolution volume Vvs.membrane surface\\narea Aisthus acritical parameter controlling binding enhanceme nt.The membran esurface canbetheplasma membran e,forexample, butalso liposome ssuspended in\\nsolution. d,e) Proteins localized atthesurface willalso bind lipids ina2Dsearch. There areover 100functionally diverse peripheral membrane proteins inyeast alone\\n[14] whose binding interactions with oneanother could strengthen substa ntially viabinding tomembranes. Wesimulated thismodel foracompreh ensive range of\\nconditions using mainly systems ofordinary different ialequations (ODE), butalso single-part iclereaction-di ffusion (RD) simulations [19, 20](Methods) .\\nhttps://do i.org/10.1371/j ournal.pc bi.1006031. g001\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 3/28\", metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 2}),\n",
       " Document(page_content='formation, decreases inKaPPcanbemuch more significant, acting toincrease thesensitivity of\\ncomplex formation tolocalization, such that itismore likely toactasatrigger forassembly\\n(S3E Fig). Mutations that asymmetrically affect theKa2Dvalues would result inchanges inσ\\nvalues, with smaller values always driving larger enhancement and stronger complexation on\\nthemembrane (S3F Fig).\\nTimescales ofassembly vary strongly with binding affinity\\nOur theory (Eq3)only describes theequilibrium state ofthemodel. However, wecandeter-\\nmine speeds ofassembly viasimulation. Now, thebinding rates and theabsolute values ofV\\nand A(not justtheratio) willinfluence thekinetics (allsimulation inputs inS2,S3,and S4\\nDatasets). Forthese time-scales, wefind that protein-lipid affinities KaPMaremost often\\nshown tobecritical incontrolling theoverall time-scales ofcomplexation, even driving slow-\\ndowns inspeeds relative tosolution binding (Fig 3E,S4Fig). Changes indiffusion from solu-\\ntion tothemembrane (about 100times slower) affect themagnitude ofassociation and disso-\\nciation rates and arecaptured implicitly inourODE simulations (Methods), and explicitly in\\nourspatially resolved reaction-diffusion simulations [19, 20]. However, theinfluence ofdiffu-\\nsion onthereaction rates israrely adominant factor inphysiological rate regimes (S4Fig),\\nindicating itisthebinding strengths rather than slow 2Ddiffusion that determine assembly\\nspeeds. However, wenote that ourcomparison ofODE and RDkinetics wasperformed inrel-\\natively small RDsystems duetosimulation costs, and itistrue that asspatial dimensions\\nincrease, times todiffuse toreach themembrane willinfluence theoverall equilibration times.\\nThe timescales wecalculated forprotein pairs and scaffold mediated systems (S4B and S4C\\nFig) were performed using ODEs attheir corresponding cellular dimensions (S3Table):\\nV=1200μm3(human) and V=37.2μm3(yeast). Performing RDsimulations atthese dimen-\\nsions would produce slower relaxation times, particularly forhuman cells, duetothetime\\nrequired toreach thesurface. Crowding would also lower effective diffusion constants ofpro-\\nteins, although thedecrease intime-scales toequilibrate would benegligible unless binding\\nrates were strongly diffusion-influenc ed(Methods).\\nBiological relevance forproteins inCME\\nTotestthebiological relevance ofmembrane localization fordriving complex formation and\\nassembly, wecollected biochemical (Table 1,S1Table, S2Table), concentration (Table 1,S4\\nTable), and cellular geometry data (S3Table) forinteractions among 37membrane targeting\\nproteins inyeast and human cells, including 22proteins involved inclathrin-mediated endo-\\ncytosis (CME). Wefirst study only individual protein pairs that canbind according toour\\nmodel ofFig1,(S1Table) shown inFig4A:themembrane binding proteins AP-2, DAB2,\\nARH, FCHo1, FCHo2, HIP1, HIP1R, PICALM, SH3GL2, EPN, AP180, SLA2, and SYP1. In\\nFig4Band 4Cweshow results ofbinding between specific pairs. Weused cytosplasmic con-\\ncentrations oftheproteins (Table 1)and thetargeted lipids (S4Table), and therelative Volume\\nand Area from their respective celltypes (S3Table). Binding constants arecollected from pre-\\nvious experimental studies (Table 1,S1Table, S3Dataset), and for2Dbinding constants we\\ntestvalues ofσ=1nm (Fig 4)and 10nm (S5Fig). Our results thus provide quantitative insight\\ninto how these pairs inisolation would usemembrane localization atphysiologic conditions to\\ndrive their protein-protein interactions. Forsome proteins, such asAP-2, solutionKPP\\navalues\\nhave been measured with partners (Fig 4A,S1Table), butfurther experiments indicate that the\\nproteins undergo minimal binding insolution duetoconformational regulation [1].Despite\\nthisadditional regulation, membrane localization willstillincrease complex formation relative\\ntowhat isobserved insolution (γ>1), sotheeffect isquantified here using themeasured KPP\\na\\nMembrane localizatio ncantrigger protein assembly', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 10}),\n",
       " Document(page_content='concentrations. Toquantify thechange inbound protein-protein complexes asafunction of\\nMembrane localizatio ncantrigger protein assembly\\nPLOS Computationa lBiology |https:/ /doi.org/10.13 71/journal.p cbi.1006031 March 5,2018 4/28', metadata={'source': '/home/jfisch27/Desktop/AI_CodingFun/LangChainQueryTexts/ThesisPapers/Osman.pdf', 'page': 3})]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result[\"source_documents\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain import PromptTemplate, LLMChain\n",
    "from langchain.chains.mapreduce import MapReduceChain\n",
    "from langchain.prompts import PromptTemplate"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ai",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.3"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
